// oci-databasetools/lib/model/database-tools-identity-lifecycle-state.js
"use strict";
/**
 * Database Tools
 * Use the Database Tools API to manage connections, private endpoints, and work requests in the Database Tools service.
 * OpenAPI spec version: 20201005
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DatabaseToolsIdentityLifecycleState = void 0;
/**
 * The lifecycle state.
 **/
var DatabaseToolsIdentityLifecycleState;
(function (DatabaseToolsIdentityLifecycleState) {
    DatabaseToolsIdentityLifecycleState["Creating"] = "CREATING";
    DatabaseToolsIdentityLifecycleState["Updating"] = "UPDATING";
    DatabaseToolsIdentityLifecycleState["Active"] = "ACTIVE";
    DatabaseToolsIdentityLifecycleState["Deleting"] = "DELETING";
    DatabaseToolsIdentityLifecycleState["Deleted"] = "DELETED";
    DatabaseToolsIdentityLifecycleState["Failed"] = "FAILED";
    DatabaseToolsIdentityLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DatabaseToolsIdentityLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(DatabaseToolsIdentityLifecycleState = exports.DatabaseToolsIdentityLifecycleState || (exports.DatabaseToolsIdentityLifecycleState = {}));
(function (DatabaseToolsIdentityLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    DatabaseToolsIdentityLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DatabaseToolsIdentityLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(DatabaseToolsIdentityLifecycleState = exports.DatabaseToolsIdentityLifecycleState || (exports.DatabaseToolsIdentityLifecycleState = {}));
