// oci-disasterrecovery/lib/model/automatic-dr-configuration-lifecycle-state.js
"use strict";
/**
 * Full Stack Disaster Recovery API
 * Use the Full Stack Disaster Recovery (DR) API to manage disaster recovery for business applications.
Full Stack DR is an OCI disaster recovery orchestration and management service that provides comprehensive disaster
recovery capabilities for all layers of an application stack, including infrastructure, middleware, database,
and application.

 * OpenAPI spec version: 20220125
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutomaticDrConfigurationLifecycleState = void 0;
/**
 * The lifecycle states of an Automatic DR configuration.
 * - CREATING - Automatic DR configuration is being created.
 * - UPDATING - Automatic DR configuration is being updated.
 * - ACTIVE - Automatic DR configuration is active.
 * - INACTIVE - Automatic DR configuration is inactive and is not available for operations.
 * - NEEDS_ATTENTION - Automatic DR configuration needs user attention.
 * - DELETING - Automatic DR configuration is being deleted.
 * - DELETED - Automatic DR configuration has been deleted.
 * - FAILED - Automatic DR configuration creation failed.
 *
 **/
var AutomaticDrConfigurationLifecycleState;
(function (AutomaticDrConfigurationLifecycleState) {
    AutomaticDrConfigurationLifecycleState["Creating"] = "CREATING";
    AutomaticDrConfigurationLifecycleState["Updating"] = "UPDATING";
    AutomaticDrConfigurationLifecycleState["Active"] = "ACTIVE";
    AutomaticDrConfigurationLifecycleState["Inactive"] = "INACTIVE";
    AutomaticDrConfigurationLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    AutomaticDrConfigurationLifecycleState["Deleting"] = "DELETING";
    AutomaticDrConfigurationLifecycleState["Deleted"] = "DELETED";
    AutomaticDrConfigurationLifecycleState["Failed"] = "FAILED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AutomaticDrConfigurationLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(AutomaticDrConfigurationLifecycleState = exports.AutomaticDrConfigurationLifecycleState || (exports.AutomaticDrConfigurationLifecycleState = {}));
(function (AutomaticDrConfigurationLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    AutomaticDrConfigurationLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AutomaticDrConfigurationLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(AutomaticDrConfigurationLifecycleState = exports.AutomaticDrConfigurationLifecycleState || (exports.AutomaticDrConfigurationLifecycleState = {}));
