// oci-disasterrecovery/lib/model/automatic-dr-configuration-member-type.js
"use strict";
/**
 * Full Stack Disaster Recovery API
 * Use the Full Stack Disaster Recovery (DR) API to manage disaster recovery for business applications.
Full Stack DR is an OCI disaster recovery orchestration and management service that provides comprehensive disaster
recovery capabilities for all layers of an application stack, including infrastructure, middleware, database,
and application.

 * OpenAPI spec version: 20220125
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AutomaticDrConfigurationMemberType = void 0;
/**
 * Enumeration of different types of Oracle Cloud Infrastructure (OCI) resources that are members in Automatic DR Configuration.
 * - DATABASE
 * - AUTONOMOUS_DATABASE
 * - AUTONOMOUS_CONTAINER_DATABASE
 *
 **/
var AutomaticDrConfigurationMemberType;
(function (AutomaticDrConfigurationMemberType) {
    AutomaticDrConfigurationMemberType["Database"] = "DATABASE";
    AutomaticDrConfigurationMemberType["AutonomousDatabase"] = "AUTONOMOUS_DATABASE";
    AutomaticDrConfigurationMemberType["AutonomousContainerDatabase"] = "AUTONOMOUS_CONTAINER_DATABASE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    AutomaticDrConfigurationMemberType["UnknownValue"] = "UNKNOWN_VALUE";
})(AutomaticDrConfigurationMemberType = exports.AutomaticDrConfigurationMemberType || (exports.AutomaticDrConfigurationMemberType = {}));
(function (AutomaticDrConfigurationMemberType) {
    function getJsonObj(obj) {
        return obj;
    }
    AutomaticDrConfigurationMemberType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    AutomaticDrConfigurationMemberType.getDeserializedJsonObj = getDeserializedJsonObj;
})(AutomaticDrConfigurationMemberType = exports.AutomaticDrConfigurationMemberType || (exports.AutomaticDrConfigurationMemberType = {}));
