// oci-fleetappsmanagement/lib/model/index.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompliancePolicyRule = exports.CompliancePolicyCollection = exports.CompliancePolicy = exports.CompliancePatchDetail = exports.ComplianceLevel = exports.ComplianceDetailTarget = exports.ComplianceDetailResource = exports.ComplianceDetailProduct = exports.ComplianceDetailPolicy = exports.Compliance = exports.ColumnMetadata = exports.CloneCatalogItemDetails = exports.Choice = exports.CheckResourceTaggingDetails = exports.ChangeTaskRecordCompartmentDetails = exports.ChangeRunbookCompartmentDetails = exports.ChangeProvisionCompartmentDetails = exports.ChangePropertyCompartmentDetails = exports.ChangePlatformConfigurationCompartmentDetails = exports.ChangePatchCompartmentDetails = exports.ChangeFleetCompartmentDetails = exports.ChangeCatalogItemCompartmentDetails = exports.CatalogSourcePayload = exports.CatalogResultPayload = exports.CatalogListingVersionCriteriaEnum = exports.CatalogItemVariablesDefinition = exports.CatalogItemSummary = exports.CatalogItemSortBy = exports.CatalogItemCollection = exports.CatalogItem = exports.BaseVariable = exports.BaseOutput = exports.AvailabilityDomainVariableDependsOn = exports.AutonomousDatabaseVersionVariableDependsOn = exports.AutonomousDataBaseVariableDependsOn = exports.AutonomousContainerDbVariableDependsOn = exports.Association = exports.AssociatedTaskDetails = exports.AssociatedSchedulerDefinition = exports.AssociatedFleetResourceDetails = exports.AssociatedFleetPropertyDetails = exports.AssociatedFleetCredentialDetails = exports.ArtifactDetails = exports.AnnouncementSummary = exports.AnnouncementSortBy = exports.AnnouncementCollection = exports.ActivityResourceTarget = exports.ActionType = exports.ActionGroupDetails = exports.ActionGroup = void 0;
exports.DbSystemVariableDependsOn = exports.DbHomeVersionVariableDependsOn = exports.DbHomeVariableDependsOn = exports.DataBaseVariableDependsOn = exports.CredentialEntitySpecificDetails = exports.CredentialDetails = exports.CreateTaskRecordDetails = exports.CreateSchedulerDefinitionDetails = exports.CreateRunbookVersionDetails = exports.CreateRunbookDetails = exports.CreateProvisionDetails = exports.CreatePropertyDetails = exports.CreatePlatformConfigurationDetails = exports.CreatePatchDetails = exports.CreateOnboardingDetails = exports.CreateMaintenanceWindowDetails = exports.CreateFleetResourceDetails = exports.CreateFleetPropertyDetails = exports.CreateFleetDetails = exports.CreateFleetCredentialDetails = exports.CreateCompliancePolicyRuleDetails = exports.CreateCatalogItemDetails = exports.ContentDetails = exports.ContainerClusterVariableDependsOn = exports.ConfirmTargetsDetails = exports.ConfigureCatalogItemDetails = exports.ConfigFileDetails = exports.ConfigCategoryDetails = exports.ConfigAssociationDetails = exports.Condition = exports.ComponentProperties = exports.ComplianceState = exports.ComplianceRuleSeverity = exports.ComplianceReportTarget = exports.ComplianceReportResource = exports.ComplianceReportProduct = exports.ComplianceReportPatchDetail = exports.ComplianceReport = exports.ComplianceRecordSummary = exports.ComplianceRecordSortBy = exports.ComplianceRecordDimension = exports.ComplianceRecordCollection = exports.ComplianceRecordAggregationCollection = exports.ComplianceRecordAggregation = exports.ComplianceRecord = exports.CompliancePolicySummary = exports.CompliancePolicySortBy = exports.CompliancePolicyRuleSummary = exports.CompliancePolicyRuleSortBy = exports.CompliancePolicyRuleCollection = void 0;
exports.ImportRunbookPrecheckRequiredValue = exports.ImportRunbookPrecheckDetails = exports.ImportRunbookDetails = exports.ImageVariableDependsOn = exports.IdentityDomainVariableDependsOn = exports.GroupsVariableDependsOn = exports.Group = exports.GenericArtifact = exports.GenerateComplianceReportDetails = exports.FleetTargetSummary = exports.FleetTargetCollection = exports.FleetTarget = exports.FleetSummary = exports.FleetSortBy = exports.FleetResourceSummary = exports.FleetResourceSortBy = exports.FleetResourceCollection = exports.FleetResource = exports.FleetPropertySummary = exports.FleetPropertySortBy = exports.FleetPropertyCollection = exports.FleetProperty = exports.FleetProductSummary = exports.FleetProductCollection = exports.FleetDetails = exports.FleetCredentialSummary = exports.FleetCredentialSortBy = exports.FleetCredentialCollection = exports.FleetCredential = exports.FleetCollection = exports.Fleet = exports.FindRunbookImportDependencyDetails = exports.FindRunbookExportDependencyDetails = exports.Filter = exports.FaultDomainVariableDependsOn = exports.ExportRunbookVersionDetails = exports.ExportRunbookDetails = exports.ExportComplianceReportDetails = exports.ExecutionWorkflowDetails = exports.ExecutionSummary = exports.ExecutionDetails = exports.ExecutionCollection = exports.Execution = exports.EntityExecutionDetails = exports.EnableLatestPolicyDetails = exports.DynamicGroupsVariableDependsOn = exports.DiscoveredTarget = exports.Details = exports.DeployedResourceDetails = exports.DependentPatchDetails = void 0;
exports.OnboardingCollection = exports.Onboarding = exports.OdsProjectVariableDependsOn = exports.NsgVariableDependsOn = exports.NotificationPreference = exports.NatGatewayVariableDependsOn = exports.MountTargetsVariableDependsOn = exports.ManagementAgentDependsOn = exports.ManagedEntityDimension = exports.ManagedEntityAggregationCollection = exports.ManagedEntityAggregation = exports.ManagedEntity = exports.ManageSettingsDetails = exports.ManageJobExecutionDetails = exports.MaintenanceWindowSummary = exports.MaintenanceWindowSortBy = exports.MaintenanceWindowCollection = exports.MaintenanceWindow = exports.LogAnalyticsSourcesDependsOn = exports.LogAnalyticsScheduledTasksDependsOn = exports.LogAnalyticsLogGroupDependsOn = exports.LogAnalyticsLogEntitiesDependsOn = exports.LoadBalancerVariableDependsOn = exports.KubernetesVersionsVariableDependsOn = exports.KmsVaultVariableDependsOn = exports.KmsSecretVariableDependsOn = exports.KmsKeyVariableDependsOn = exports.KeyValueProperty = exports.JobStatus = exports.JobExecutionDetails = exports.JobActivity = exports.InventoryResourceSummary = exports.InventoryResourceSortBy = exports.InventoryResourceCollection = exports.InventoryRecordSummary = exports.InventoryRecordProperty = exports.InventoryRecordPatchDetails = exports.InventoryRecordComponent = exports.InventoryRecordCollection = exports.InventoryRecord = exports.InstanceVariableDependsOn = exports.InstanceSummary = exports.InstanceShapeVariableWithFlexDependsOn = exports.InstanceShapeVariableDependsOn = exports.InstalledPatchSummary = exports.InstalledPatchCollection = exports.InputParameter = exports.InputFileContentDetails = exports.InputArgument = exports.ImportRunbookVersionDetails = void 0;
exports.ReportMetadataCollection = exports.ReportFilter = exports.ReportDetails = exports.ReportCollection = exports.RecommendedPatchSummary = exports.RecommendedPatchCollection = exports.PublishRunbookDetails = exports.ProvisionSummary = exports.ProvisionCollection = exports.Provision = exports.PropertySummary = exports.PropertySortBy = exports.PropertyCollection = exports.Property = exports.Properties = exports.Prompt = exports.ProductVersionDetails = exports.ProductStackSubCategoryDetails = exports.ProductSortBy = exports.PrivateEndpointVariableDependsOn = exports.PreviousTaskInstanceDetails = exports.Preferences = exports.PlatformSpecificArtifact = exports.PlatformConfigurationSummary = exports.PlatformConfigurationSortBy = exports.PlatformConfigurationCollection = exports.PlatformConfiguration = exports.Placeholder = exports.PauseDetails = exports.PatchType = exports.PatchSummary = exports.PatchSortBy = exports.PatchSeverity = exports.PatchSelectionDetails = exports.PatchProduct = exports.PatchFileContentDetails = exports.PatchCollection = exports.Patch = exports.OutputVariableMapping = exports.OutputVariableDetails = exports.OutputGroup = exports.Outcome = exports.OsType = exports.OrderClause = exports.OperationType = exports.OperationStatus = exports.OperationRunbook = exports.OnboardingSummary = exports.OnboardingPolicySummary = exports.OnboardingPolicyCollection = void 0;
exports.Scope = exports.SchemaDocument = exports.SchedulerJobSummary = exports.SchedulerJobDimension = exports.SchedulerJobCollection = exports.SchedulerJobAggregationCollection = exports.SchedulerJobAggregation = exports.SchedulerJob = exports.SchedulerExecutionSummary = exports.SchedulerExecutionCollection = exports.SchedulerDefinitionSummary = exports.SchedulerDefinitionCollection = exports.SchedulerDefinition = exports.ScheduledFleetSummary = exports.ScheduledFleetCollection = exports.Schedule = exports.RunbookVersionSummary = exports.RunbookVersionCollection = exports.RunbookVersion = exports.RunbookTransferDependencySummary = exports.RunbookTransferDependency = exports.RunbookSummary = exports.RunbookSortBy = exports.RunbookImportStatusSummary = exports.RunbookImportStatusCollection = exports.RunbookImportRequiredValue = exports.RunbookImportDependencyFinderRequiredValue = exports.RunbookImportDependencyCollection = exports.RunbookImportAs = exports.RunbookImport = exports.RunbookExportStatusSummary = exports.RunbookExportStatusCollection = exports.RunbookExportDependencyCollection = exports.RunbookExport = exports.RunbookCollection = exports.Runbook = exports.RunOnDetails = exports.Rule = exports.RollbackWorkflowDetails = exports.ResourceTagEnablementInfo = exports.ResourceTagCheckDetails = exports.ResourceSummary = exports.ResourceSelection = exports.ResourceCollection = exports.RequiredValue = exports.RequestTargetDiscoveryDetails = exports.RequestResourceValidationDetails = exports.ReportTimeRange = exports.ReportSummary = exports.ReportMetadataSummary = void 0;
exports.VariableGroup = exports.Variable = exports.ValueType = exports.UserActionDetails = exports.UpdateTaskRecordDetails = exports.UpdateSchedulerJobDetails = exports.UpdateSchedulerDefinitionDetails = exports.UpdateRunbookVersionDetails = exports.UpdateRunbookDetails = exports.UpdateProvisionDetails = exports.UpdatePropertyDetails = exports.UpdatePlatformConfigurationDetails = exports.UpdatePatchDetails = exports.UpdateOnboardingDetails = exports.UpdateMaintenanceWindowDetails = exports.UpdateFleetResourceDetails = exports.UpdateFleetPropertyDetails = exports.UpdateFleetDetails = exports.UpdateFleetCredentialDetails = exports.UpdateCompliancePolicyRuleDetails = exports.UpdateCatalogItemDetails = exports.UpcomingSchedule = exports.TransferRunbookContentDetails = exports.TaskVariable = exports.TaskScope = exports.TaskRecordSummary = exports.TaskRecordSortBy = exports.TaskRecordCollection = exports.TaskRecord = exports.TaskNotificationPreferences = exports.TaskExecutionType = exports.TaskArgument = exports.Task = exports.TargetSortBy = exports.TargetResource = exports.TargetPropertySummary = exports.TargetPropertyCollection = exports.TargetComponentSummary = exports.TargetComponentCollection = exports.TagVariableDependsOn = exports.SubnetVariableDependsOn = exports.StepSummary = exports.StepCollection = exports.StackSource = exports.SortOrder = exports.SortBy = exports.SetDefaultRunbookDetails = exports.ServiceGatewayVariableDependsOn = exports.SelectionCriteria = exports.Selection = void 0;
exports.DbHomeVariable = exports.DatetimeVariable = exports.DatabaseExadataId = exports.DatabaseDbNodeId = exports.DatabaseDataguardAssociationId = exports.DatabaseDataGuardId = exports.DatabaseCdbId = exports.DataBaseVariable = exports.CustomSchedule = exports.CsvOutput = exports.CredentialConfigCategoryDetails = exports.CopyableStringOutput = exports.ContainerClusterVariable = exports.CompartmentVariable = exports.CloudVmClusterId = exports.CatalogTemplateResultConfig = exports.CatalogSourceTemplateConfig = exports.CatalogParSourceConfig = exports.CatalogParResultConfig = exports.CatalogMarketplaceSourceConfig = exports.CatalogGitSourceConfig = exports.CatalogGitResultConfig = exports.CatalogContentDetails = exports.BooleanVariable = exports.BoolOutput = exports.AvailabilityDomainVariable = exports.AutonomousDatabaseVersionVariable = exports.AutonomousDataBaseVariable = exports.AutonomousContainerDbVariable = exports.AssociatedSharedTaskDetails = exports.AssociatedLocalTaskDetails = exports.ArrayVariable = exports.ApmDomainVariable = exports.ApiBasedExecutionDetails = exports.ActionGroupBasedUserActionDetails = exports.WorkflowGroup = exports.WorkflowComponent = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.VolumeBackupPoliciesVariableDependsOn = exports.Version = exports.VcnVariableDependsOn = exports.VariableGroups = void 0;
exports.MultiValueFilter = exports.MountTargetsVariable = exports.MapOutput = exports.ManualResourceSelection = exports.ManagementAgents = exports.MaintenanceWindowSchedule = exports.LogAnalyticsSources = exports.LogAnalyticsScheduledTasks = exports.LogAnalyticsLogGroup = exports.LogAnalyticsLogEntities = exports.LogAnalyticsEntityTypes = exports.LoadBalancerVariable = exports.LoadBalancerReservedIps = exports.ListOutput = exports.LinkOutput = exports.LifecycleOperationConfigCategoryDetails = exports.KubernetesVersionsVariable = exports.KmsVaultVariable = exports.KmsSecretVariable = exports.KmsKeyVariable = exports.KeyEncryptionCredentialDetails = exports.JsonOutput = exports.IntegerVariable = exports.InstanceVariable = exports.InstanceShapeVariableWithFlex = exports.InstanceShapeVariable = exports.InputFileObjectStorageBucketContentDetails = exports.ImageVariable = exports.IdentityDomainVariable = exports.GroupsVariable = exports.GroupFleetDetails = exports.GenericFleetDetails = exports.GenericArtifactDetails = exports.FleetCredentialEntitySpecificDetails = exports.FleetBasedActionGroupDetails = exports.FleetBasedActionGroup = exports.FileVariable = exports.FileTaskArgument = exports.FileStorageMountTargetId = exports.FileStorageFilesystemId = exports.FileStorageExportSetId = exports.FileInputArgument = exports.FaultDomainVariable = exports.EnvironmentFleetDetails = exports.EnvironmentConfigCategoryDetails = exports.EnumVariable = exports.DynamicResourceSelection = exports.DynamicGroupsVariable = exports.DbSystemVariable = exports.DbHomeVersionVariable = void 0;
exports.TerraformBasedExecutionDetails = exports.TargetCredentialEntitySpecificDetails = exports.TagVariable = exports.SubnetVariable = exports.StringVariable = exports.StringTaskArgument = exports.StringOutput = exports.StringInputArgument = exports.StepBasedUserActionDetails = exports.SshPublicKeyVariable = exports.SingleValueFilter = exports.ServiceGatewayVariable = exports.SelfHostedInstanceRunOnDetails = exports.SelfHostedInstanceConfigCategoryDetails = exports.ScriptBasedExecutionDetails = exports.ScheduleInstanceTargetRunOnDetails = exports.ScheduleInstanceRunOnDetails = exports.RunbookImportAsVersion = exports.RunbookImportAsRunbook = exports.ResourceCredentialEntitySpecificDetails = exports.RegionVariable = exports.ProductStackGenericSubCategoryDetails = exports.ProductStackConfigCategoryDetails = exports.ProductStackAsProductSubCategoryDetails = exports.ProductFleetDetails = exports.ProductConfigCategoryDetails = exports.PrivateEndpointVariable = exports.PreviousTaskInstanceRunOnDetails = exports.PlatformSpecificArtifactDetails = exports.PlainTextCredentialDetails = exports.PatchTypeConfigCategoryDetails = exports.PatchReleaseDateSelectionDetails = exports.PatchNameSelectionDetails = exports.PatchLevelSelectionDetails = exports.PatchFileObjectStorageBucketContentDetails = exports.PasswordVariable = exports.OutputVariableInputArgument = exports.OdsProjectVariable = exports.OcidOutput = exports.ObjectStorageBucketName = exports.ObjectStorageBucketContentDetails = exports.ObjectStorageBucketConfigFileDetails = exports.NumberVariable = exports.NumberOutput = exports.NsgVariable = exports.NetworkLoadBalancerId = exports.NatGatewayVariable = exports.MysqlShapeId = exports.MysqlConfigurationId = exports.MultilineVariable = void 0;
exports.WorkflowTaskComponent = exports.WorkflowGroupComponent = exports.VolumeId = exports.VolumeBackupPoliciesVariable = exports.VcnVariable = exports.VcnSecListId = exports.VcnNetworkSecurityGroupId = exports.VaultSecretCredentialDetails = exports.UserActionBasedPauseDetails = exports.TransferRunbookParUrlContentDetails = exports.TransferRunbookObjectStorageBucketContentDetails = exports.TimeBasedPauseDetails = void 0;
const ActionGroup = __importStar(require("./action-group"));
exports.ActionGroup = ActionGroup.ActionGroup;
const ActionGroupDetails = __importStar(require("./action-group-details"));
exports.ActionGroupDetails = ActionGroupDetails.ActionGroupDetails;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ActivityResourceTarget = __importStar(require("./activity-resource-target"));
exports.ActivityResourceTarget = ActivityResourceTarget.ActivityResourceTarget;
const AnnouncementCollection = __importStar(require("./announcement-collection"));
exports.AnnouncementCollection = AnnouncementCollection.AnnouncementCollection;
const AnnouncementSortBy = __importStar(require("./announcement-sort-by"));
exports.AnnouncementSortBy = AnnouncementSortBy.AnnouncementSortBy;
const AnnouncementSummary = __importStar(require("./announcement-summary"));
exports.AnnouncementSummary = AnnouncementSummary.AnnouncementSummary;
const ArtifactDetails = __importStar(require("./artifact-details"));
exports.ArtifactDetails = ArtifactDetails.ArtifactDetails;
const AssociatedFleetCredentialDetails = __importStar(require("./associated-fleet-credential-details"));
exports.AssociatedFleetCredentialDetails = AssociatedFleetCredentialDetails.AssociatedFleetCredentialDetails;
const AssociatedFleetPropertyDetails = __importStar(require("./associated-fleet-property-details"));
exports.AssociatedFleetPropertyDetails = AssociatedFleetPropertyDetails.AssociatedFleetPropertyDetails;
const AssociatedFleetResourceDetails = __importStar(require("./associated-fleet-resource-details"));
exports.AssociatedFleetResourceDetails = AssociatedFleetResourceDetails.AssociatedFleetResourceDetails;
const AssociatedSchedulerDefinition = __importStar(require("./associated-scheduler-definition"));
exports.AssociatedSchedulerDefinition = AssociatedSchedulerDefinition.AssociatedSchedulerDefinition;
const AssociatedTaskDetails = __importStar(require("./associated-task-details"));
exports.AssociatedTaskDetails = AssociatedTaskDetails.AssociatedTaskDetails;
const Association = __importStar(require("./association"));
exports.Association = Association.Association;
const AutonomousContainerDbVariableDependsOn = __importStar(require("./autonomous-container-db-variable-depends-on"));
exports.AutonomousContainerDbVariableDependsOn = AutonomousContainerDbVariableDependsOn.AutonomousContainerDbVariableDependsOn;
const AutonomousDataBaseVariableDependsOn = __importStar(require("./autonomous-data-base-variable-depends-on"));
exports.AutonomousDataBaseVariableDependsOn = AutonomousDataBaseVariableDependsOn.AutonomousDataBaseVariableDependsOn;
const AutonomousDatabaseVersionVariableDependsOn = __importStar(require("./autonomous-database-version-variable-depends-on"));
exports.AutonomousDatabaseVersionVariableDependsOn = AutonomousDatabaseVersionVariableDependsOn.AutonomousDatabaseVersionVariableDependsOn;
const AvailabilityDomainVariableDependsOn = __importStar(require("./availability-domain-variable-depends-on"));
exports.AvailabilityDomainVariableDependsOn = AvailabilityDomainVariableDependsOn.AvailabilityDomainVariableDependsOn;
const BaseOutput = __importStar(require("./base-output"));
exports.BaseOutput = BaseOutput.BaseOutput;
const BaseVariable = __importStar(require("./base-variable"));
exports.BaseVariable = BaseVariable.BaseVariable;
const CatalogItem = __importStar(require("./catalog-item"));
exports.CatalogItem = CatalogItem.CatalogItem;
const CatalogItemCollection = __importStar(require("./catalog-item-collection"));
exports.CatalogItemCollection = CatalogItemCollection.CatalogItemCollection;
const CatalogItemSortBy = __importStar(require("./catalog-item-sort-by"));
exports.CatalogItemSortBy = CatalogItemSortBy.CatalogItemSortBy;
const CatalogItemSummary = __importStar(require("./catalog-item-summary"));
exports.CatalogItemSummary = CatalogItemSummary.CatalogItemSummary;
const CatalogItemVariablesDefinition = __importStar(require("./catalog-item-variables-definition"));
exports.CatalogItemVariablesDefinition = CatalogItemVariablesDefinition.CatalogItemVariablesDefinition;
const CatalogListingVersionCriteriaEnum = __importStar(require("./catalog-listing-version-criteria-enum"));
exports.CatalogListingVersionCriteriaEnum = CatalogListingVersionCriteriaEnum.CatalogListingVersionCriteriaEnum;
const CatalogResultPayload = __importStar(require("./catalog-result-payload"));
exports.CatalogResultPayload = CatalogResultPayload.CatalogResultPayload;
const CatalogSourcePayload = __importStar(require("./catalog-source-payload"));
exports.CatalogSourcePayload = CatalogSourcePayload.CatalogSourcePayload;
const ChangeCatalogItemCompartmentDetails = __importStar(require("./change-catalog-item-compartment-details"));
exports.ChangeCatalogItemCompartmentDetails = ChangeCatalogItemCompartmentDetails.ChangeCatalogItemCompartmentDetails;
const ChangeFleetCompartmentDetails = __importStar(require("./change-fleet-compartment-details"));
exports.ChangeFleetCompartmentDetails = ChangeFleetCompartmentDetails.ChangeFleetCompartmentDetails;
const ChangePatchCompartmentDetails = __importStar(require("./change-patch-compartment-details"));
exports.ChangePatchCompartmentDetails = ChangePatchCompartmentDetails.ChangePatchCompartmentDetails;
const ChangePlatformConfigurationCompartmentDetails = __importStar(require("./change-platform-configuration-compartment-details"));
exports.ChangePlatformConfigurationCompartmentDetails = ChangePlatformConfigurationCompartmentDetails.ChangePlatformConfigurationCompartmentDetails;
const ChangePropertyCompartmentDetails = __importStar(require("./change-property-compartment-details"));
exports.ChangePropertyCompartmentDetails = ChangePropertyCompartmentDetails.ChangePropertyCompartmentDetails;
const ChangeProvisionCompartmentDetails = __importStar(require("./change-provision-compartment-details"));
exports.ChangeProvisionCompartmentDetails = ChangeProvisionCompartmentDetails.ChangeProvisionCompartmentDetails;
const ChangeRunbookCompartmentDetails = __importStar(require("./change-runbook-compartment-details"));
exports.ChangeRunbookCompartmentDetails = ChangeRunbookCompartmentDetails.ChangeRunbookCompartmentDetails;
const ChangeTaskRecordCompartmentDetails = __importStar(require("./change-task-record-compartment-details"));
exports.ChangeTaskRecordCompartmentDetails = ChangeTaskRecordCompartmentDetails.ChangeTaskRecordCompartmentDetails;
const CheckResourceTaggingDetails = __importStar(require("./check-resource-tagging-details"));
exports.CheckResourceTaggingDetails = CheckResourceTaggingDetails.CheckResourceTaggingDetails;
const Choice = __importStar(require("./choice"));
exports.Choice = Choice.Choice;
const CloneCatalogItemDetails = __importStar(require("./clone-catalog-item-details"));
exports.CloneCatalogItemDetails = CloneCatalogItemDetails.CloneCatalogItemDetails;
const ColumnMetadata = __importStar(require("./column-metadata"));
exports.ColumnMetadata = ColumnMetadata.ColumnMetadata;
const Compliance = __importStar(require("./compliance"));
exports.Compliance = Compliance.Compliance;
const ComplianceDetailPolicy = __importStar(require("./compliance-detail-policy"));
exports.ComplianceDetailPolicy = ComplianceDetailPolicy.ComplianceDetailPolicy;
const ComplianceDetailProduct = __importStar(require("./compliance-detail-product"));
exports.ComplianceDetailProduct = ComplianceDetailProduct.ComplianceDetailProduct;
const ComplianceDetailResource = __importStar(require("./compliance-detail-resource"));
exports.ComplianceDetailResource = ComplianceDetailResource.ComplianceDetailResource;
const ComplianceDetailTarget = __importStar(require("./compliance-detail-target"));
exports.ComplianceDetailTarget = ComplianceDetailTarget.ComplianceDetailTarget;
const ComplianceLevel = __importStar(require("./compliance-level"));
exports.ComplianceLevel = ComplianceLevel.ComplianceLevel;
const CompliancePatchDetail = __importStar(require("./compliance-patch-detail"));
exports.CompliancePatchDetail = CompliancePatchDetail.CompliancePatchDetail;
const CompliancePolicy = __importStar(require("./compliance-policy"));
exports.CompliancePolicy = CompliancePolicy.CompliancePolicy;
const CompliancePolicyCollection = __importStar(require("./compliance-policy-collection"));
exports.CompliancePolicyCollection = CompliancePolicyCollection.CompliancePolicyCollection;
const CompliancePolicyRule = __importStar(require("./compliance-policy-rule"));
exports.CompliancePolicyRule = CompliancePolicyRule.CompliancePolicyRule;
const CompliancePolicyRuleCollection = __importStar(require("./compliance-policy-rule-collection"));
exports.CompliancePolicyRuleCollection = CompliancePolicyRuleCollection.CompliancePolicyRuleCollection;
const CompliancePolicyRuleSortBy = __importStar(require("./compliance-policy-rule-sort-by"));
exports.CompliancePolicyRuleSortBy = CompliancePolicyRuleSortBy.CompliancePolicyRuleSortBy;
const CompliancePolicyRuleSummary = __importStar(require("./compliance-policy-rule-summary"));
exports.CompliancePolicyRuleSummary = CompliancePolicyRuleSummary.CompliancePolicyRuleSummary;
const CompliancePolicySortBy = __importStar(require("./compliance-policy-sort-by"));
exports.CompliancePolicySortBy = CompliancePolicySortBy.CompliancePolicySortBy;
const CompliancePolicySummary = __importStar(require("./compliance-policy-summary"));
exports.CompliancePolicySummary = CompliancePolicySummary.CompliancePolicySummary;
const ComplianceRecord = __importStar(require("./compliance-record"));
exports.ComplianceRecord = ComplianceRecord.ComplianceRecord;
const ComplianceRecordAggregation = __importStar(require("./compliance-record-aggregation"));
exports.ComplianceRecordAggregation = ComplianceRecordAggregation.ComplianceRecordAggregation;
const ComplianceRecordAggregationCollection = __importStar(require("./compliance-record-aggregation-collection"));
exports.ComplianceRecordAggregationCollection = ComplianceRecordAggregationCollection.ComplianceRecordAggregationCollection;
const ComplianceRecordCollection = __importStar(require("./compliance-record-collection"));
exports.ComplianceRecordCollection = ComplianceRecordCollection.ComplianceRecordCollection;
const ComplianceRecordDimension = __importStar(require("./compliance-record-dimension"));
exports.ComplianceRecordDimension = ComplianceRecordDimension.ComplianceRecordDimension;
const ComplianceRecordSortBy = __importStar(require("./compliance-record-sort-by"));
exports.ComplianceRecordSortBy = ComplianceRecordSortBy.ComplianceRecordSortBy;
const ComplianceRecordSummary = __importStar(require("./compliance-record-summary"));
exports.ComplianceRecordSummary = ComplianceRecordSummary.ComplianceRecordSummary;
const ComplianceReport = __importStar(require("./compliance-report"));
exports.ComplianceReport = ComplianceReport.ComplianceReport;
const ComplianceReportPatchDetail = __importStar(require("./compliance-report-patch-detail"));
exports.ComplianceReportPatchDetail = ComplianceReportPatchDetail.ComplianceReportPatchDetail;
const ComplianceReportProduct = __importStar(require("./compliance-report-product"));
exports.ComplianceReportProduct = ComplianceReportProduct.ComplianceReportProduct;
const ComplianceReportResource = __importStar(require("./compliance-report-resource"));
exports.ComplianceReportResource = ComplianceReportResource.ComplianceReportResource;
const ComplianceReportTarget = __importStar(require("./compliance-report-target"));
exports.ComplianceReportTarget = ComplianceReportTarget.ComplianceReportTarget;
const ComplianceRuleSeverity = __importStar(require("./compliance-rule-severity"));
exports.ComplianceRuleSeverity = ComplianceRuleSeverity.ComplianceRuleSeverity;
const ComplianceState = __importStar(require("./compliance-state"));
exports.ComplianceState = ComplianceState.ComplianceState;
const ComponentProperties = __importStar(require("./component-properties"));
exports.ComponentProperties = ComponentProperties.ComponentProperties;
const Condition = __importStar(require("./condition"));
exports.Condition = Condition.Condition;
const ConfigAssociationDetails = __importStar(require("./config-association-details"));
exports.ConfigAssociationDetails = ConfigAssociationDetails.ConfigAssociationDetails;
const ConfigCategoryDetails = __importStar(require("./config-category-details"));
exports.ConfigCategoryDetails = ConfigCategoryDetails.ConfigCategoryDetails;
const ConfigFileDetails = __importStar(require("./config-file-details"));
exports.ConfigFileDetails = ConfigFileDetails.ConfigFileDetails;
const ConfigureCatalogItemDetails = __importStar(require("./configure-catalog-item-details"));
exports.ConfigureCatalogItemDetails = ConfigureCatalogItemDetails.ConfigureCatalogItemDetails;
const ConfirmTargetsDetails = __importStar(require("./confirm-targets-details"));
exports.ConfirmTargetsDetails = ConfirmTargetsDetails.ConfirmTargetsDetails;
const ContainerClusterVariableDependsOn = __importStar(require("./container-cluster-variable-depends-on"));
exports.ContainerClusterVariableDependsOn = ContainerClusterVariableDependsOn.ContainerClusterVariableDependsOn;
const ContentDetails = __importStar(require("./content-details"));
exports.ContentDetails = ContentDetails.ContentDetails;
const CreateCatalogItemDetails = __importStar(require("./create-catalog-item-details"));
exports.CreateCatalogItemDetails = CreateCatalogItemDetails.CreateCatalogItemDetails;
const CreateCompliancePolicyRuleDetails = __importStar(require("./create-compliance-policy-rule-details"));
exports.CreateCompliancePolicyRuleDetails = CreateCompliancePolicyRuleDetails.CreateCompliancePolicyRuleDetails;
const CreateFleetCredentialDetails = __importStar(require("./create-fleet-credential-details"));
exports.CreateFleetCredentialDetails = CreateFleetCredentialDetails.CreateFleetCredentialDetails;
const CreateFleetDetails = __importStar(require("./create-fleet-details"));
exports.CreateFleetDetails = CreateFleetDetails.CreateFleetDetails;
const CreateFleetPropertyDetails = __importStar(require("./create-fleet-property-details"));
exports.CreateFleetPropertyDetails = CreateFleetPropertyDetails.CreateFleetPropertyDetails;
const CreateFleetResourceDetails = __importStar(require("./create-fleet-resource-details"));
exports.CreateFleetResourceDetails = CreateFleetResourceDetails.CreateFleetResourceDetails;
const CreateMaintenanceWindowDetails = __importStar(require("./create-maintenance-window-details"));
exports.CreateMaintenanceWindowDetails = CreateMaintenanceWindowDetails.CreateMaintenanceWindowDetails;
const CreateOnboardingDetails = __importStar(require("./create-onboarding-details"));
exports.CreateOnboardingDetails = CreateOnboardingDetails.CreateOnboardingDetails;
const CreatePatchDetails = __importStar(require("./create-patch-details"));
exports.CreatePatchDetails = CreatePatchDetails.CreatePatchDetails;
const CreatePlatformConfigurationDetails = __importStar(require("./create-platform-configuration-details"));
exports.CreatePlatformConfigurationDetails = CreatePlatformConfigurationDetails.CreatePlatformConfigurationDetails;
const CreatePropertyDetails = __importStar(require("./create-property-details"));
exports.CreatePropertyDetails = CreatePropertyDetails.CreatePropertyDetails;
const CreateProvisionDetails = __importStar(require("./create-provision-details"));
exports.CreateProvisionDetails = CreateProvisionDetails.CreateProvisionDetails;
const CreateRunbookDetails = __importStar(require("./create-runbook-details"));
exports.CreateRunbookDetails = CreateRunbookDetails.CreateRunbookDetails;
const CreateRunbookVersionDetails = __importStar(require("./create-runbook-version-details"));
exports.CreateRunbookVersionDetails = CreateRunbookVersionDetails.CreateRunbookVersionDetails;
const CreateSchedulerDefinitionDetails = __importStar(require("./create-scheduler-definition-details"));
exports.CreateSchedulerDefinitionDetails = CreateSchedulerDefinitionDetails.CreateSchedulerDefinitionDetails;
const CreateTaskRecordDetails = __importStar(require("./create-task-record-details"));
exports.CreateTaskRecordDetails = CreateTaskRecordDetails.CreateTaskRecordDetails;
const CredentialDetails = __importStar(require("./credential-details"));
exports.CredentialDetails = CredentialDetails.CredentialDetails;
const CredentialEntitySpecificDetails = __importStar(require("./credential-entity-specific-details"));
exports.CredentialEntitySpecificDetails = CredentialEntitySpecificDetails.CredentialEntitySpecificDetails;
const DataBaseVariableDependsOn = __importStar(require("./data-base-variable-depends-on"));
exports.DataBaseVariableDependsOn = DataBaseVariableDependsOn.DataBaseVariableDependsOn;
const DbHomeVariableDependsOn = __importStar(require("./db-home-variable-depends-on"));
exports.DbHomeVariableDependsOn = DbHomeVariableDependsOn.DbHomeVariableDependsOn;
const DbHomeVersionVariableDependsOn = __importStar(require("./db-home-version-variable-depends-on"));
exports.DbHomeVersionVariableDependsOn = DbHomeVersionVariableDependsOn.DbHomeVersionVariableDependsOn;
const DbSystemVariableDependsOn = __importStar(require("./db-system-variable-depends-on"));
exports.DbSystemVariableDependsOn = DbSystemVariableDependsOn.DbSystemVariableDependsOn;
const DependentPatchDetails = __importStar(require("./dependent-patch-details"));
exports.DependentPatchDetails = DependentPatchDetails.DependentPatchDetails;
const DeployedResourceDetails = __importStar(require("./deployed-resource-details"));
exports.DeployedResourceDetails = DeployedResourceDetails.DeployedResourceDetails;
const Details = __importStar(require("./details"));
exports.Details = Details.Details;
const DiscoveredTarget = __importStar(require("./discovered-target"));
exports.DiscoveredTarget = DiscoveredTarget.DiscoveredTarget;
const DynamicGroupsVariableDependsOn = __importStar(require("./dynamic-groups-variable-depends-on"));
exports.DynamicGroupsVariableDependsOn = DynamicGroupsVariableDependsOn.DynamicGroupsVariableDependsOn;
const EnableLatestPolicyDetails = __importStar(require("./enable-latest-policy-details"));
exports.EnableLatestPolicyDetails = EnableLatestPolicyDetails.EnableLatestPolicyDetails;
const EntityExecutionDetails = __importStar(require("./entity-execution-details"));
exports.EntityExecutionDetails = EntityExecutionDetails.EntityExecutionDetails;
const Execution = __importStar(require("./execution"));
exports.Execution = Execution.Execution;
const ExecutionCollection = __importStar(require("./execution-collection"));
exports.ExecutionCollection = ExecutionCollection.ExecutionCollection;
const ExecutionDetails = __importStar(require("./execution-details"));
exports.ExecutionDetails = ExecutionDetails.ExecutionDetails;
const ExecutionSummary = __importStar(require("./execution-summary"));
exports.ExecutionSummary = ExecutionSummary.ExecutionSummary;
const ExecutionWorkflowDetails = __importStar(require("./execution-workflow-details"));
exports.ExecutionWorkflowDetails = ExecutionWorkflowDetails.ExecutionWorkflowDetails;
const ExportComplianceReportDetails = __importStar(require("./export-compliance-report-details"));
exports.ExportComplianceReportDetails = ExportComplianceReportDetails.ExportComplianceReportDetails;
const ExportRunbookDetails = __importStar(require("./export-runbook-details"));
exports.ExportRunbookDetails = ExportRunbookDetails.ExportRunbookDetails;
const ExportRunbookVersionDetails = __importStar(require("./export-runbook-version-details"));
exports.ExportRunbookVersionDetails = ExportRunbookVersionDetails.ExportRunbookVersionDetails;
const FaultDomainVariableDependsOn = __importStar(require("./fault-domain-variable-depends-on"));
exports.FaultDomainVariableDependsOn = FaultDomainVariableDependsOn.FaultDomainVariableDependsOn;
const Filter = __importStar(require("./filter"));
exports.Filter = Filter.Filter;
const FindRunbookExportDependencyDetails = __importStar(require("./find-runbook-export-dependency-details"));
exports.FindRunbookExportDependencyDetails = FindRunbookExportDependencyDetails.FindRunbookExportDependencyDetails;
const FindRunbookImportDependencyDetails = __importStar(require("./find-runbook-import-dependency-details"));
exports.FindRunbookImportDependencyDetails = FindRunbookImportDependencyDetails.FindRunbookImportDependencyDetails;
const Fleet = __importStar(require("./fleet"));
exports.Fleet = Fleet.Fleet;
const FleetCollection = __importStar(require("./fleet-collection"));
exports.FleetCollection = FleetCollection.FleetCollection;
const FleetCredential = __importStar(require("./fleet-credential"));
exports.FleetCredential = FleetCredential.FleetCredential;
const FleetCredentialCollection = __importStar(require("./fleet-credential-collection"));
exports.FleetCredentialCollection = FleetCredentialCollection.FleetCredentialCollection;
const FleetCredentialSortBy = __importStar(require("./fleet-credential-sort-by"));
exports.FleetCredentialSortBy = FleetCredentialSortBy.FleetCredentialSortBy;
const FleetCredentialSummary = __importStar(require("./fleet-credential-summary"));
exports.FleetCredentialSummary = FleetCredentialSummary.FleetCredentialSummary;
const FleetDetails = __importStar(require("./fleet-details"));
exports.FleetDetails = FleetDetails.FleetDetails;
const FleetProductCollection = __importStar(require("./fleet-product-collection"));
exports.FleetProductCollection = FleetProductCollection.FleetProductCollection;
const FleetProductSummary = __importStar(require("./fleet-product-summary"));
exports.FleetProductSummary = FleetProductSummary.FleetProductSummary;
const FleetProperty = __importStar(require("./fleet-property"));
exports.FleetProperty = FleetProperty.FleetProperty;
const FleetPropertyCollection = __importStar(require("./fleet-property-collection"));
exports.FleetPropertyCollection = FleetPropertyCollection.FleetPropertyCollection;
const FleetPropertySortBy = __importStar(require("./fleet-property-sort-by"));
exports.FleetPropertySortBy = FleetPropertySortBy.FleetPropertySortBy;
const FleetPropertySummary = __importStar(require("./fleet-property-summary"));
exports.FleetPropertySummary = FleetPropertySummary.FleetPropertySummary;
const FleetResource = __importStar(require("./fleet-resource"));
exports.FleetResource = FleetResource.FleetResource;
const FleetResourceCollection = __importStar(require("./fleet-resource-collection"));
exports.FleetResourceCollection = FleetResourceCollection.FleetResourceCollection;
const FleetResourceSortBy = __importStar(require("./fleet-resource-sort-by"));
exports.FleetResourceSortBy = FleetResourceSortBy.FleetResourceSortBy;
const FleetResourceSummary = __importStar(require("./fleet-resource-summary"));
exports.FleetResourceSummary = FleetResourceSummary.FleetResourceSummary;
const FleetSortBy = __importStar(require("./fleet-sort-by"));
exports.FleetSortBy = FleetSortBy.FleetSortBy;
const FleetSummary = __importStar(require("./fleet-summary"));
exports.FleetSummary = FleetSummary.FleetSummary;
const FleetTarget = __importStar(require("./fleet-target"));
exports.FleetTarget = FleetTarget.FleetTarget;
const FleetTargetCollection = __importStar(require("./fleet-target-collection"));
exports.FleetTargetCollection = FleetTargetCollection.FleetTargetCollection;
const FleetTargetSummary = __importStar(require("./fleet-target-summary"));
exports.FleetTargetSummary = FleetTargetSummary.FleetTargetSummary;
const GenerateComplianceReportDetails = __importStar(require("./generate-compliance-report-details"));
exports.GenerateComplianceReportDetails = GenerateComplianceReportDetails.GenerateComplianceReportDetails;
const GenericArtifact = __importStar(require("./generic-artifact"));
exports.GenericArtifact = GenericArtifact.GenericArtifact;
const Group = __importStar(require("./group"));
exports.Group = Group.Group;
const GroupsVariableDependsOn = __importStar(require("./groups-variable-depends-on"));
exports.GroupsVariableDependsOn = GroupsVariableDependsOn.GroupsVariableDependsOn;
const IdentityDomainVariableDependsOn = __importStar(require("./identity-domain-variable-depends-on"));
exports.IdentityDomainVariableDependsOn = IdentityDomainVariableDependsOn.IdentityDomainVariableDependsOn;
const ImageVariableDependsOn = __importStar(require("./image-variable-depends-on"));
exports.ImageVariableDependsOn = ImageVariableDependsOn.ImageVariableDependsOn;
const ImportRunbookDetails = __importStar(require("./import-runbook-details"));
exports.ImportRunbookDetails = ImportRunbookDetails.ImportRunbookDetails;
const ImportRunbookPrecheckDetails = __importStar(require("./import-runbook-precheck-details"));
exports.ImportRunbookPrecheckDetails = ImportRunbookPrecheckDetails.ImportRunbookPrecheckDetails;
const ImportRunbookPrecheckRequiredValue = __importStar(require("./import-runbook-precheck-required-value"));
exports.ImportRunbookPrecheckRequiredValue = ImportRunbookPrecheckRequiredValue.ImportRunbookPrecheckRequiredValue;
const ImportRunbookVersionDetails = __importStar(require("./import-runbook-version-details"));
exports.ImportRunbookVersionDetails = ImportRunbookVersionDetails.ImportRunbookVersionDetails;
const InputArgument = __importStar(require("./input-argument"));
exports.InputArgument = InputArgument.InputArgument;
const InputFileContentDetails = __importStar(require("./input-file-content-details"));
exports.InputFileContentDetails = InputFileContentDetails.InputFileContentDetails;
const InputParameter = __importStar(require("./input-parameter"));
exports.InputParameter = InputParameter.InputParameter;
const InstalledPatchCollection = __importStar(require("./installed-patch-collection"));
exports.InstalledPatchCollection = InstalledPatchCollection.InstalledPatchCollection;
const InstalledPatchSummary = __importStar(require("./installed-patch-summary"));
exports.InstalledPatchSummary = InstalledPatchSummary.InstalledPatchSummary;
const InstanceShapeVariableDependsOn = __importStar(require("./instance-shape-variable-depends-on"));
exports.InstanceShapeVariableDependsOn = InstanceShapeVariableDependsOn.InstanceShapeVariableDependsOn;
const InstanceShapeVariableWithFlexDependsOn = __importStar(require("./instance-shape-variable-with-flex-depends-on"));
exports.InstanceShapeVariableWithFlexDependsOn = InstanceShapeVariableWithFlexDependsOn.InstanceShapeVariableWithFlexDependsOn;
const InstanceSummary = __importStar(require("./instance-summary"));
exports.InstanceSummary = InstanceSummary.InstanceSummary;
const InstanceVariableDependsOn = __importStar(require("./instance-variable-depends-on"));
exports.InstanceVariableDependsOn = InstanceVariableDependsOn.InstanceVariableDependsOn;
const InventoryRecord = __importStar(require("./inventory-record"));
exports.InventoryRecord = InventoryRecord.InventoryRecord;
const InventoryRecordCollection = __importStar(require("./inventory-record-collection"));
exports.InventoryRecordCollection = InventoryRecordCollection.InventoryRecordCollection;
const InventoryRecordComponent = __importStar(require("./inventory-record-component"));
exports.InventoryRecordComponent = InventoryRecordComponent.InventoryRecordComponent;
const InventoryRecordPatchDetails = __importStar(require("./inventory-record-patch-details"));
exports.InventoryRecordPatchDetails = InventoryRecordPatchDetails.InventoryRecordPatchDetails;
const InventoryRecordProperty = __importStar(require("./inventory-record-property"));
exports.InventoryRecordProperty = InventoryRecordProperty.InventoryRecordProperty;
const InventoryRecordSummary = __importStar(require("./inventory-record-summary"));
exports.InventoryRecordSummary = InventoryRecordSummary.InventoryRecordSummary;
const InventoryResourceCollection = __importStar(require("./inventory-resource-collection"));
exports.InventoryResourceCollection = InventoryResourceCollection.InventoryResourceCollection;
const InventoryResourceSortBy = __importStar(require("./inventory-resource-sort-by"));
exports.InventoryResourceSortBy = InventoryResourceSortBy.InventoryResourceSortBy;
const InventoryResourceSummary = __importStar(require("./inventory-resource-summary"));
exports.InventoryResourceSummary = InventoryResourceSummary.InventoryResourceSummary;
const JobActivity = __importStar(require("./job-activity"));
exports.JobActivity = JobActivity.JobActivity;
const JobExecutionDetails = __importStar(require("./job-execution-details"));
exports.JobExecutionDetails = JobExecutionDetails.JobExecutionDetails;
const JobStatus = __importStar(require("./job-status"));
exports.JobStatus = JobStatus.JobStatus;
const KeyValueProperty = __importStar(require("./key-value-property"));
exports.KeyValueProperty = KeyValueProperty.KeyValueProperty;
const KmsKeyVariableDependsOn = __importStar(require("./kms-key-variable-depends-on"));
exports.KmsKeyVariableDependsOn = KmsKeyVariableDependsOn.KmsKeyVariableDependsOn;
const KmsSecretVariableDependsOn = __importStar(require("./kms-secret-variable-depends-on"));
exports.KmsSecretVariableDependsOn = KmsSecretVariableDependsOn.KmsSecretVariableDependsOn;
const KmsVaultVariableDependsOn = __importStar(require("./kms-vault-variable-depends-on"));
exports.KmsVaultVariableDependsOn = KmsVaultVariableDependsOn.KmsVaultVariableDependsOn;
const KubernetesVersionsVariableDependsOn = __importStar(require("./kubernetes-versions-variable-depends-on"));
exports.KubernetesVersionsVariableDependsOn = KubernetesVersionsVariableDependsOn.KubernetesVersionsVariableDependsOn;
const LoadBalancerVariableDependsOn = __importStar(require("./load-balancer-variable-depends-on"));
exports.LoadBalancerVariableDependsOn = LoadBalancerVariableDependsOn.LoadBalancerVariableDependsOn;
const LogAnalyticsLogEntitiesDependsOn = __importStar(require("./log-analytics-log-entities-depends-on"));
exports.LogAnalyticsLogEntitiesDependsOn = LogAnalyticsLogEntitiesDependsOn.LogAnalyticsLogEntitiesDependsOn;
const LogAnalyticsLogGroupDependsOn = __importStar(require("./log-analytics-log-group-depends-on"));
exports.LogAnalyticsLogGroupDependsOn = LogAnalyticsLogGroupDependsOn.LogAnalyticsLogGroupDependsOn;
const LogAnalyticsScheduledTasksDependsOn = __importStar(require("./log-analytics-scheduled-tasks-depends-on"));
exports.LogAnalyticsScheduledTasksDependsOn = LogAnalyticsScheduledTasksDependsOn.LogAnalyticsScheduledTasksDependsOn;
const LogAnalyticsSourcesDependsOn = __importStar(require("./log-analytics-sources-depends-on"));
exports.LogAnalyticsSourcesDependsOn = LogAnalyticsSourcesDependsOn.LogAnalyticsSourcesDependsOn;
const MaintenanceWindow = __importStar(require("./maintenance-window"));
exports.MaintenanceWindow = MaintenanceWindow.MaintenanceWindow;
const MaintenanceWindowCollection = __importStar(require("./maintenance-window-collection"));
exports.MaintenanceWindowCollection = MaintenanceWindowCollection.MaintenanceWindowCollection;
const MaintenanceWindowSortBy = __importStar(require("./maintenance-window-sort-by"));
exports.MaintenanceWindowSortBy = MaintenanceWindowSortBy.MaintenanceWindowSortBy;
const MaintenanceWindowSummary = __importStar(require("./maintenance-window-summary"));
exports.MaintenanceWindowSummary = MaintenanceWindowSummary.MaintenanceWindowSummary;
const ManageJobExecutionDetails = __importStar(require("./manage-job-execution-details"));
exports.ManageJobExecutionDetails = ManageJobExecutionDetails.ManageJobExecutionDetails;
const ManageSettingsDetails = __importStar(require("./manage-settings-details"));
exports.ManageSettingsDetails = ManageSettingsDetails.ManageSettingsDetails;
const ManagedEntity = __importStar(require("./managed-entity"));
exports.ManagedEntity = ManagedEntity.ManagedEntity;
const ManagedEntityAggregation = __importStar(require("./managed-entity-aggregation"));
exports.ManagedEntityAggregation = ManagedEntityAggregation.ManagedEntityAggregation;
const ManagedEntityAggregationCollection = __importStar(require("./managed-entity-aggregation-collection"));
exports.ManagedEntityAggregationCollection = ManagedEntityAggregationCollection.ManagedEntityAggregationCollection;
const ManagedEntityDimension = __importStar(require("./managed-entity-dimension"));
exports.ManagedEntityDimension = ManagedEntityDimension.ManagedEntityDimension;
const ManagementAgentDependsOn = __importStar(require("./management-agent-depends-on"));
exports.ManagementAgentDependsOn = ManagementAgentDependsOn.ManagementAgentDependsOn;
const MountTargetsVariableDependsOn = __importStar(require("./mount-targets-variable-depends-on"));
exports.MountTargetsVariableDependsOn = MountTargetsVariableDependsOn.MountTargetsVariableDependsOn;
const NatGatewayVariableDependsOn = __importStar(require("./nat-gateway-variable-depends-on"));
exports.NatGatewayVariableDependsOn = NatGatewayVariableDependsOn.NatGatewayVariableDependsOn;
const NotificationPreference = __importStar(require("./notification-preference"));
exports.NotificationPreference = NotificationPreference.NotificationPreference;
const NsgVariableDependsOn = __importStar(require("./nsg-variable-depends-on"));
exports.NsgVariableDependsOn = NsgVariableDependsOn.NsgVariableDependsOn;
const OdsProjectVariableDependsOn = __importStar(require("./ods-project-variable-depends-on"));
exports.OdsProjectVariableDependsOn = OdsProjectVariableDependsOn.OdsProjectVariableDependsOn;
const Onboarding = __importStar(require("./onboarding"));
exports.Onboarding = Onboarding.Onboarding;
const OnboardingCollection = __importStar(require("./onboarding-collection"));
exports.OnboardingCollection = OnboardingCollection.OnboardingCollection;
const OnboardingPolicyCollection = __importStar(require("./onboarding-policy-collection"));
exports.OnboardingPolicyCollection = OnboardingPolicyCollection.OnboardingPolicyCollection;
const OnboardingPolicySummary = __importStar(require("./onboarding-policy-summary"));
exports.OnboardingPolicySummary = OnboardingPolicySummary.OnboardingPolicySummary;
const OnboardingSummary = __importStar(require("./onboarding-summary"));
exports.OnboardingSummary = OnboardingSummary.OnboardingSummary;
const OperationRunbook = __importStar(require("./operation-runbook"));
exports.OperationRunbook = OperationRunbook.OperationRunbook;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const OrderClause = __importStar(require("./order-clause"));
exports.OrderClause = OrderClause.OrderClause;
const OsType = __importStar(require("./os-type"));
exports.OsType = OsType.OsType;
const Outcome = __importStar(require("./outcome"));
exports.Outcome = Outcome.Outcome;
const OutputGroup = __importStar(require("./output-group"));
exports.OutputGroup = OutputGroup.OutputGroup;
const OutputVariableDetails = __importStar(require("./output-variable-details"));
exports.OutputVariableDetails = OutputVariableDetails.OutputVariableDetails;
const OutputVariableMapping = __importStar(require("./output-variable-mapping"));
exports.OutputVariableMapping = OutputVariableMapping.OutputVariableMapping;
const Patch = __importStar(require("./patch"));
exports.Patch = Patch.Patch;
const PatchCollection = __importStar(require("./patch-collection"));
exports.PatchCollection = PatchCollection.PatchCollection;
const PatchFileContentDetails = __importStar(require("./patch-file-content-details"));
exports.PatchFileContentDetails = PatchFileContentDetails.PatchFileContentDetails;
const PatchProduct = __importStar(require("./patch-product"));
exports.PatchProduct = PatchProduct.PatchProduct;
const PatchSelectionDetails = __importStar(require("./patch-selection-details"));
exports.PatchSelectionDetails = PatchSelectionDetails.PatchSelectionDetails;
const PatchSeverity = __importStar(require("./patch-severity"));
exports.PatchSeverity = PatchSeverity.PatchSeverity;
const PatchSortBy = __importStar(require("./patch-sort-by"));
exports.PatchSortBy = PatchSortBy.PatchSortBy;
const PatchSummary = __importStar(require("./patch-summary"));
exports.PatchSummary = PatchSummary.PatchSummary;
const PatchType = __importStar(require("./patch-type"));
exports.PatchType = PatchType.PatchType;
const PauseDetails = __importStar(require("./pause-details"));
exports.PauseDetails = PauseDetails.PauseDetails;
const Placeholder = __importStar(require("./placeholder"));
exports.Placeholder = Placeholder.Placeholder;
const PlatformConfiguration = __importStar(require("./platform-configuration"));
exports.PlatformConfiguration = PlatformConfiguration.PlatformConfiguration;
const PlatformConfigurationCollection = __importStar(require("./platform-configuration-collection"));
exports.PlatformConfigurationCollection = PlatformConfigurationCollection.PlatformConfigurationCollection;
const PlatformConfigurationSortBy = __importStar(require("./platform-configuration-sort-by"));
exports.PlatformConfigurationSortBy = PlatformConfigurationSortBy.PlatformConfigurationSortBy;
const PlatformConfigurationSummary = __importStar(require("./platform-configuration-summary"));
exports.PlatformConfigurationSummary = PlatformConfigurationSummary.PlatformConfigurationSummary;
const PlatformSpecificArtifact = __importStar(require("./platform-specific-artifact"));
exports.PlatformSpecificArtifact = PlatformSpecificArtifact.PlatformSpecificArtifact;
const Preferences = __importStar(require("./preferences"));
exports.Preferences = Preferences.Preferences;
const PreviousTaskInstanceDetails = __importStar(require("./previous-task-instance-details"));
exports.PreviousTaskInstanceDetails = PreviousTaskInstanceDetails.PreviousTaskInstanceDetails;
const PrivateEndpointVariableDependsOn = __importStar(require("./private-endpoint-variable-depends-on"));
exports.PrivateEndpointVariableDependsOn = PrivateEndpointVariableDependsOn.PrivateEndpointVariableDependsOn;
const ProductSortBy = __importStar(require("./product-sort-by"));
exports.ProductSortBy = ProductSortBy.ProductSortBy;
const ProductStackSubCategoryDetails = __importStar(require("./product-stack-sub-category-details"));
exports.ProductStackSubCategoryDetails = ProductStackSubCategoryDetails.ProductStackSubCategoryDetails;
const ProductVersionDetails = __importStar(require("./product-version-details"));
exports.ProductVersionDetails = ProductVersionDetails.ProductVersionDetails;
const Prompt = __importStar(require("./prompt"));
exports.Prompt = Prompt.Prompt;
const Properties = __importStar(require("./properties"));
exports.Properties = Properties.Properties;
const Property = __importStar(require("./property"));
exports.Property = Property.Property;
const PropertyCollection = __importStar(require("./property-collection"));
exports.PropertyCollection = PropertyCollection.PropertyCollection;
const PropertySortBy = __importStar(require("./property-sort-by"));
exports.PropertySortBy = PropertySortBy.PropertySortBy;
const PropertySummary = __importStar(require("./property-summary"));
exports.PropertySummary = PropertySummary.PropertySummary;
const Provision = __importStar(require("./provision"));
exports.Provision = Provision.Provision;
const ProvisionCollection = __importStar(require("./provision-collection"));
exports.ProvisionCollection = ProvisionCollection.ProvisionCollection;
const ProvisionSummary = __importStar(require("./provision-summary"));
exports.ProvisionSummary = ProvisionSummary.ProvisionSummary;
const PublishRunbookDetails = __importStar(require("./publish-runbook-details"));
exports.PublishRunbookDetails = PublishRunbookDetails.PublishRunbookDetails;
const RecommendedPatchCollection = __importStar(require("./recommended-patch-collection"));
exports.RecommendedPatchCollection = RecommendedPatchCollection.RecommendedPatchCollection;
const RecommendedPatchSummary = __importStar(require("./recommended-patch-summary"));
exports.RecommendedPatchSummary = RecommendedPatchSummary.RecommendedPatchSummary;
const ReportCollection = __importStar(require("./report-collection"));
exports.ReportCollection = ReportCollection.ReportCollection;
const ReportDetails = __importStar(require("./report-details"));
exports.ReportDetails = ReportDetails.ReportDetails;
const ReportFilter = __importStar(require("./report-filter"));
exports.ReportFilter = ReportFilter.ReportFilter;
const ReportMetadataCollection = __importStar(require("./report-metadata-collection"));
exports.ReportMetadataCollection = ReportMetadataCollection.ReportMetadataCollection;
const ReportMetadataSummary = __importStar(require("./report-metadata-summary"));
exports.ReportMetadataSummary = ReportMetadataSummary.ReportMetadataSummary;
const ReportSummary = __importStar(require("./report-summary"));
exports.ReportSummary = ReportSummary.ReportSummary;
const ReportTimeRange = __importStar(require("./report-time-range"));
exports.ReportTimeRange = ReportTimeRange.ReportTimeRange;
const RequestResourceValidationDetails = __importStar(require("./request-resource-validation-details"));
exports.RequestResourceValidationDetails = RequestResourceValidationDetails.RequestResourceValidationDetails;
const RequestTargetDiscoveryDetails = __importStar(require("./request-target-discovery-details"));
exports.RequestTargetDiscoveryDetails = RequestTargetDiscoveryDetails.RequestTargetDiscoveryDetails;
const RequiredValue = __importStar(require("./required-value"));
exports.RequiredValue = RequiredValue.RequiredValue;
const ResourceCollection = __importStar(require("./resource-collection"));
exports.ResourceCollection = ResourceCollection.ResourceCollection;
const ResourceSelection = __importStar(require("./resource-selection"));
exports.ResourceSelection = ResourceSelection.ResourceSelection;
const ResourceSummary = __importStar(require("./resource-summary"));
exports.ResourceSummary = ResourceSummary.ResourceSummary;
const ResourceTagCheckDetails = __importStar(require("./resource-tag-check-details"));
exports.ResourceTagCheckDetails = ResourceTagCheckDetails.ResourceTagCheckDetails;
const ResourceTagEnablementInfo = __importStar(require("./resource-tag-enablement-info"));
exports.ResourceTagEnablementInfo = ResourceTagEnablementInfo.ResourceTagEnablementInfo;
const RollbackWorkflowDetails = __importStar(require("./rollback-workflow-details"));
exports.RollbackWorkflowDetails = RollbackWorkflowDetails.RollbackWorkflowDetails;
const Rule = __importStar(require("./rule"));
exports.Rule = Rule.Rule;
const RunOnDetails = __importStar(require("./run-on-details"));
exports.RunOnDetails = RunOnDetails.RunOnDetails;
const Runbook = __importStar(require("./runbook"));
exports.Runbook = Runbook.Runbook;
const RunbookCollection = __importStar(require("./runbook-collection"));
exports.RunbookCollection = RunbookCollection.RunbookCollection;
const RunbookExport = __importStar(require("./runbook-export"));
exports.RunbookExport = RunbookExport.RunbookExport;
const RunbookExportDependencyCollection = __importStar(require("./runbook-export-dependency-collection"));
exports.RunbookExportDependencyCollection = RunbookExportDependencyCollection.RunbookExportDependencyCollection;
const RunbookExportStatusCollection = __importStar(require("./runbook-export-status-collection"));
exports.RunbookExportStatusCollection = RunbookExportStatusCollection.RunbookExportStatusCollection;
const RunbookExportStatusSummary = __importStar(require("./runbook-export-status-summary"));
exports.RunbookExportStatusSummary = RunbookExportStatusSummary.RunbookExportStatusSummary;
const RunbookImport = __importStar(require("./runbook-import"));
exports.RunbookImport = RunbookImport.RunbookImport;
const RunbookImportAs = __importStar(require("./runbook-import-as"));
exports.RunbookImportAs = RunbookImportAs.RunbookImportAs;
const RunbookImportDependencyCollection = __importStar(require("./runbook-import-dependency-collection"));
exports.RunbookImportDependencyCollection = RunbookImportDependencyCollection.RunbookImportDependencyCollection;
const RunbookImportDependencyFinderRequiredValue = __importStar(require("./runbook-import-dependency-finder-required-value"));
exports.RunbookImportDependencyFinderRequiredValue = RunbookImportDependencyFinderRequiredValue.RunbookImportDependencyFinderRequiredValue;
const RunbookImportRequiredValue = __importStar(require("./runbook-import-required-value"));
exports.RunbookImportRequiredValue = RunbookImportRequiredValue.RunbookImportRequiredValue;
const RunbookImportStatusCollection = __importStar(require("./runbook-import-status-collection"));
exports.RunbookImportStatusCollection = RunbookImportStatusCollection.RunbookImportStatusCollection;
const RunbookImportStatusSummary = __importStar(require("./runbook-import-status-summary"));
exports.RunbookImportStatusSummary = RunbookImportStatusSummary.RunbookImportStatusSummary;
const RunbookSortBy = __importStar(require("./runbook-sort-by"));
exports.RunbookSortBy = RunbookSortBy.RunbookSortBy;
const RunbookSummary = __importStar(require("./runbook-summary"));
exports.RunbookSummary = RunbookSummary.RunbookSummary;
const RunbookTransferDependency = __importStar(require("./runbook-transfer-dependency"));
exports.RunbookTransferDependency = RunbookTransferDependency.RunbookTransferDependency;
const RunbookTransferDependencySummary = __importStar(require("./runbook-transfer-dependency-summary"));
exports.RunbookTransferDependencySummary = RunbookTransferDependencySummary.RunbookTransferDependencySummary;
const RunbookVersion = __importStar(require("./runbook-version"));
exports.RunbookVersion = RunbookVersion.RunbookVersion;
const RunbookVersionCollection = __importStar(require("./runbook-version-collection"));
exports.RunbookVersionCollection = RunbookVersionCollection.RunbookVersionCollection;
const RunbookVersionSummary = __importStar(require("./runbook-version-summary"));
exports.RunbookVersionSummary = RunbookVersionSummary.RunbookVersionSummary;
const Schedule = __importStar(require("./schedule"));
exports.Schedule = Schedule.Schedule;
const ScheduledFleetCollection = __importStar(require("./scheduled-fleet-collection"));
exports.ScheduledFleetCollection = ScheduledFleetCollection.ScheduledFleetCollection;
const ScheduledFleetSummary = __importStar(require("./scheduled-fleet-summary"));
exports.ScheduledFleetSummary = ScheduledFleetSummary.ScheduledFleetSummary;
const SchedulerDefinition = __importStar(require("./scheduler-definition"));
exports.SchedulerDefinition = SchedulerDefinition.SchedulerDefinition;
const SchedulerDefinitionCollection = __importStar(require("./scheduler-definition-collection"));
exports.SchedulerDefinitionCollection = SchedulerDefinitionCollection.SchedulerDefinitionCollection;
const SchedulerDefinitionSummary = __importStar(require("./scheduler-definition-summary"));
exports.SchedulerDefinitionSummary = SchedulerDefinitionSummary.SchedulerDefinitionSummary;
const SchedulerExecutionCollection = __importStar(require("./scheduler-execution-collection"));
exports.SchedulerExecutionCollection = SchedulerExecutionCollection.SchedulerExecutionCollection;
const SchedulerExecutionSummary = __importStar(require("./scheduler-execution-summary"));
exports.SchedulerExecutionSummary = SchedulerExecutionSummary.SchedulerExecutionSummary;
const SchedulerJob = __importStar(require("./scheduler-job"));
exports.SchedulerJob = SchedulerJob.SchedulerJob;
const SchedulerJobAggregation = __importStar(require("./scheduler-job-aggregation"));
exports.SchedulerJobAggregation = SchedulerJobAggregation.SchedulerJobAggregation;
const SchedulerJobAggregationCollection = __importStar(require("./scheduler-job-aggregation-collection"));
exports.SchedulerJobAggregationCollection = SchedulerJobAggregationCollection.SchedulerJobAggregationCollection;
const SchedulerJobCollection = __importStar(require("./scheduler-job-collection"));
exports.SchedulerJobCollection = SchedulerJobCollection.SchedulerJobCollection;
const SchedulerJobDimension = __importStar(require("./scheduler-job-dimension"));
exports.SchedulerJobDimension = SchedulerJobDimension.SchedulerJobDimension;
const SchedulerJobSummary = __importStar(require("./scheduler-job-summary"));
exports.SchedulerJobSummary = SchedulerJobSummary.SchedulerJobSummary;
const SchemaDocument = __importStar(require("./schema-document"));
exports.SchemaDocument = SchemaDocument.SchemaDocument;
const Scope = __importStar(require("./scope"));
exports.Scope = Scope.Scope;
const Selection = __importStar(require("./selection"));
exports.Selection = Selection.Selection;
const SelectionCriteria = __importStar(require("./selection-criteria"));
exports.SelectionCriteria = SelectionCriteria.SelectionCriteria;
const ServiceGatewayVariableDependsOn = __importStar(require("./service-gateway-variable-depends-on"));
exports.ServiceGatewayVariableDependsOn = ServiceGatewayVariableDependsOn.ServiceGatewayVariableDependsOn;
const SetDefaultRunbookDetails = __importStar(require("./set-default-runbook-details"));
exports.SetDefaultRunbookDetails = SetDefaultRunbookDetails.SetDefaultRunbookDetails;
const SortBy = __importStar(require("./sort-by"));
exports.SortBy = SortBy.SortBy;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const StackSource = __importStar(require("./stack-source"));
exports.StackSource = StackSource.StackSource;
const StepCollection = __importStar(require("./step-collection"));
exports.StepCollection = StepCollection.StepCollection;
const StepSummary = __importStar(require("./step-summary"));
exports.StepSummary = StepSummary.StepSummary;
const SubnetVariableDependsOn = __importStar(require("./subnet-variable-depends-on"));
exports.SubnetVariableDependsOn = SubnetVariableDependsOn.SubnetVariableDependsOn;
const TagVariableDependsOn = __importStar(require("./tag-variable-depends-on"));
exports.TagVariableDependsOn = TagVariableDependsOn.TagVariableDependsOn;
const TargetComponentCollection = __importStar(require("./target-component-collection"));
exports.TargetComponentCollection = TargetComponentCollection.TargetComponentCollection;
const TargetComponentSummary = __importStar(require("./target-component-summary"));
exports.TargetComponentSummary = TargetComponentSummary.TargetComponentSummary;
const TargetPropertyCollection = __importStar(require("./target-property-collection"));
exports.TargetPropertyCollection = TargetPropertyCollection.TargetPropertyCollection;
const TargetPropertySummary = __importStar(require("./target-property-summary"));
exports.TargetPropertySummary = TargetPropertySummary.TargetPropertySummary;
const TargetResource = __importStar(require("./target-resource"));
exports.TargetResource = TargetResource.TargetResource;
const TargetSortBy = __importStar(require("./target-sort-by"));
exports.TargetSortBy = TargetSortBy.TargetSortBy;
const Task = __importStar(require("./task"));
exports.Task = Task.Task;
const TaskArgument = __importStar(require("./task-argument"));
exports.TaskArgument = TaskArgument.TaskArgument;
const TaskExecutionType = __importStar(require("./task-execution-type"));
exports.TaskExecutionType = TaskExecutionType.TaskExecutionType;
const TaskNotificationPreferences = __importStar(require("./task-notification-preferences"));
exports.TaskNotificationPreferences = TaskNotificationPreferences.TaskNotificationPreferences;
const TaskRecord = __importStar(require("./task-record"));
exports.TaskRecord = TaskRecord.TaskRecord;
const TaskRecordCollection = __importStar(require("./task-record-collection"));
exports.TaskRecordCollection = TaskRecordCollection.TaskRecordCollection;
const TaskRecordSortBy = __importStar(require("./task-record-sort-by"));
exports.TaskRecordSortBy = TaskRecordSortBy.TaskRecordSortBy;
const TaskRecordSummary = __importStar(require("./task-record-summary"));
exports.TaskRecordSummary = TaskRecordSummary.TaskRecordSummary;
const TaskScope = __importStar(require("./task-scope"));
exports.TaskScope = TaskScope.TaskScope;
const TaskVariable = __importStar(require("./task-variable"));
exports.TaskVariable = TaskVariable.TaskVariable;
const TransferRunbookContentDetails = __importStar(require("./transfer-runbook-content-details"));
exports.TransferRunbookContentDetails = TransferRunbookContentDetails.TransferRunbookContentDetails;
const UpcomingSchedule = __importStar(require("./upcoming-schedule"));
exports.UpcomingSchedule = UpcomingSchedule.UpcomingSchedule;
const UpdateCatalogItemDetails = __importStar(require("./update-catalog-item-details"));
exports.UpdateCatalogItemDetails = UpdateCatalogItemDetails.UpdateCatalogItemDetails;
const UpdateCompliancePolicyRuleDetails = __importStar(require("./update-compliance-policy-rule-details"));
exports.UpdateCompliancePolicyRuleDetails = UpdateCompliancePolicyRuleDetails.UpdateCompliancePolicyRuleDetails;
const UpdateFleetCredentialDetails = __importStar(require("./update-fleet-credential-details"));
exports.UpdateFleetCredentialDetails = UpdateFleetCredentialDetails.UpdateFleetCredentialDetails;
const UpdateFleetDetails = __importStar(require("./update-fleet-details"));
exports.UpdateFleetDetails = UpdateFleetDetails.UpdateFleetDetails;
const UpdateFleetPropertyDetails = __importStar(require("./update-fleet-property-details"));
exports.UpdateFleetPropertyDetails = UpdateFleetPropertyDetails.UpdateFleetPropertyDetails;
const UpdateFleetResourceDetails = __importStar(require("./update-fleet-resource-details"));
exports.UpdateFleetResourceDetails = UpdateFleetResourceDetails.UpdateFleetResourceDetails;
const UpdateMaintenanceWindowDetails = __importStar(require("./update-maintenance-window-details"));
exports.UpdateMaintenanceWindowDetails = UpdateMaintenanceWindowDetails.UpdateMaintenanceWindowDetails;
const UpdateOnboardingDetails = __importStar(require("./update-onboarding-details"));
exports.UpdateOnboardingDetails = UpdateOnboardingDetails.UpdateOnboardingDetails;
const UpdatePatchDetails = __importStar(require("./update-patch-details"));
exports.UpdatePatchDetails = UpdatePatchDetails.UpdatePatchDetails;
const UpdatePlatformConfigurationDetails = __importStar(require("./update-platform-configuration-details"));
exports.UpdatePlatformConfigurationDetails = UpdatePlatformConfigurationDetails.UpdatePlatformConfigurationDetails;
const UpdatePropertyDetails = __importStar(require("./update-property-details"));
exports.UpdatePropertyDetails = UpdatePropertyDetails.UpdatePropertyDetails;
const UpdateProvisionDetails = __importStar(require("./update-provision-details"));
exports.UpdateProvisionDetails = UpdateProvisionDetails.UpdateProvisionDetails;
const UpdateRunbookDetails = __importStar(require("./update-runbook-details"));
exports.UpdateRunbookDetails = UpdateRunbookDetails.UpdateRunbookDetails;
const UpdateRunbookVersionDetails = __importStar(require("./update-runbook-version-details"));
exports.UpdateRunbookVersionDetails = UpdateRunbookVersionDetails.UpdateRunbookVersionDetails;
const UpdateSchedulerDefinitionDetails = __importStar(require("./update-scheduler-definition-details"));
exports.UpdateSchedulerDefinitionDetails = UpdateSchedulerDefinitionDetails.UpdateSchedulerDefinitionDetails;
const UpdateSchedulerJobDetails = __importStar(require("./update-scheduler-job-details"));
exports.UpdateSchedulerJobDetails = UpdateSchedulerJobDetails.UpdateSchedulerJobDetails;
const UpdateTaskRecordDetails = __importStar(require("./update-task-record-details"));
exports.UpdateTaskRecordDetails = UpdateTaskRecordDetails.UpdateTaskRecordDetails;
const UserActionDetails = __importStar(require("./user-action-details"));
exports.UserActionDetails = UserActionDetails.UserActionDetails;
const ValueType = __importStar(require("./value-type"));
exports.ValueType = ValueType.ValueType;
const Variable = __importStar(require("./variable"));
exports.Variable = Variable.Variable;
const VariableGroup = __importStar(require("./variable-group"));
exports.VariableGroup = VariableGroup.VariableGroup;
const VariableGroups = __importStar(require("./variable-groups"));
exports.VariableGroups = VariableGroups.VariableGroups;
const VcnVariableDependsOn = __importStar(require("./vcn-variable-depends-on"));
exports.VcnVariableDependsOn = VcnVariableDependsOn.VcnVariableDependsOn;
const Version = __importStar(require("./version"));
exports.Version = Version.Version;
const VolumeBackupPoliciesVariableDependsOn = __importStar(require("./volume-backup-policies-variable-depends-on"));
exports.VolumeBackupPoliciesVariableDependsOn = VolumeBackupPoliciesVariableDependsOn.VolumeBackupPoliciesVariableDependsOn;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const WorkflowComponent = __importStar(require("./workflow-component"));
exports.WorkflowComponent = WorkflowComponent.WorkflowComponent;
const WorkflowGroup = __importStar(require("./workflow-group"));
exports.WorkflowGroup = WorkflowGroup.WorkflowGroup;
const ActionGroupBasedUserActionDetails = __importStar(require("./action-group-based-user-action-details"));
exports.ActionGroupBasedUserActionDetails = ActionGroupBasedUserActionDetails.ActionGroupBasedUserActionDetails;
const ApiBasedExecutionDetails = __importStar(require("./api-based-execution-details"));
exports.ApiBasedExecutionDetails = ApiBasedExecutionDetails.ApiBasedExecutionDetails;
const ApmDomainVariable = __importStar(require("./apm-domain-variable"));
exports.ApmDomainVariable = ApmDomainVariable.ApmDomainVariable;
const ArrayVariable = __importStar(require("./array-variable"));
exports.ArrayVariable = ArrayVariable.ArrayVariable;
const AssociatedLocalTaskDetails = __importStar(require("./associated-local-task-details"));
exports.AssociatedLocalTaskDetails = AssociatedLocalTaskDetails.AssociatedLocalTaskDetails;
const AssociatedSharedTaskDetails = __importStar(require("./associated-shared-task-details"));
exports.AssociatedSharedTaskDetails = AssociatedSharedTaskDetails.AssociatedSharedTaskDetails;
const AutonomousContainerDbVariable = __importStar(require("./autonomous-container-db-variable"));
exports.AutonomousContainerDbVariable = AutonomousContainerDbVariable.AutonomousContainerDbVariable;
const AutonomousDataBaseVariable = __importStar(require("./autonomous-data-base-variable"));
exports.AutonomousDataBaseVariable = AutonomousDataBaseVariable.AutonomousDataBaseVariable;
const AutonomousDatabaseVersionVariable = __importStar(require("./autonomous-database-version-variable"));
exports.AutonomousDatabaseVersionVariable = AutonomousDatabaseVersionVariable.AutonomousDatabaseVersionVariable;
const AvailabilityDomainVariable = __importStar(require("./availability-domain-variable"));
exports.AvailabilityDomainVariable = AvailabilityDomainVariable.AvailabilityDomainVariable;
const BoolOutput = __importStar(require("./bool-output"));
exports.BoolOutput = BoolOutput.BoolOutput;
const BooleanVariable = __importStar(require("./boolean-variable"));
exports.BooleanVariable = BooleanVariable.BooleanVariable;
const CatalogContentDetails = __importStar(require("./catalog-content-details"));
exports.CatalogContentDetails = CatalogContentDetails.CatalogContentDetails;
const CatalogGitResultConfig = __importStar(require("./catalog-git-result-config"));
exports.CatalogGitResultConfig = CatalogGitResultConfig.CatalogGitResultConfig;
const CatalogGitSourceConfig = __importStar(require("./catalog-git-source-config"));
exports.CatalogGitSourceConfig = CatalogGitSourceConfig.CatalogGitSourceConfig;
const CatalogMarketplaceSourceConfig = __importStar(require("./catalog-marketplace-source-config"));
exports.CatalogMarketplaceSourceConfig = CatalogMarketplaceSourceConfig.CatalogMarketplaceSourceConfig;
const CatalogParResultConfig = __importStar(require("./catalog-par-result-config"));
exports.CatalogParResultConfig = CatalogParResultConfig.CatalogParResultConfig;
const CatalogParSourceConfig = __importStar(require("./catalog-par-source-config"));
exports.CatalogParSourceConfig = CatalogParSourceConfig.CatalogParSourceConfig;
const CatalogSourceTemplateConfig = __importStar(require("./catalog-source-template-config"));
exports.CatalogSourceTemplateConfig = CatalogSourceTemplateConfig.CatalogSourceTemplateConfig;
const CatalogTemplateResultConfig = __importStar(require("./catalog-template-result-config"));
exports.CatalogTemplateResultConfig = CatalogTemplateResultConfig.CatalogTemplateResultConfig;
const CloudVmClusterId = __importStar(require("./cloud-vm-cluster-id"));
exports.CloudVmClusterId = CloudVmClusterId.CloudVmClusterId;
const CompartmentVariable = __importStar(require("./compartment-variable"));
exports.CompartmentVariable = CompartmentVariable.CompartmentVariable;
const ContainerClusterVariable = __importStar(require("./container-cluster-variable"));
exports.ContainerClusterVariable = ContainerClusterVariable.ContainerClusterVariable;
const CopyableStringOutput = __importStar(require("./copyable-string-output"));
exports.CopyableStringOutput = CopyableStringOutput.CopyableStringOutput;
const CredentialConfigCategoryDetails = __importStar(require("./credential-config-category-details"));
exports.CredentialConfigCategoryDetails = CredentialConfigCategoryDetails.CredentialConfigCategoryDetails;
const CsvOutput = __importStar(require("./csv-output"));
exports.CsvOutput = CsvOutput.CsvOutput;
const CustomSchedule = __importStar(require("./custom-schedule"));
exports.CustomSchedule = CustomSchedule.CustomSchedule;
const DataBaseVariable = __importStar(require("./data-base-variable"));
exports.DataBaseVariable = DataBaseVariable.DataBaseVariable;
const DatabaseCdbId = __importStar(require("./database-cdb-id"));
exports.DatabaseCdbId = DatabaseCdbId.DatabaseCdbId;
const DatabaseDataGuardId = __importStar(require("./database-data-guard-id"));
exports.DatabaseDataGuardId = DatabaseDataGuardId.DatabaseDataGuardId;
const DatabaseDataguardAssociationId = __importStar(require("./database-dataguard-association-id"));
exports.DatabaseDataguardAssociationId = DatabaseDataguardAssociationId.DatabaseDataguardAssociationId;
const DatabaseDbNodeId = __importStar(require("./database-db-node-id"));
exports.DatabaseDbNodeId = DatabaseDbNodeId.DatabaseDbNodeId;
const DatabaseExadataId = __importStar(require("./database-exadata-id"));
exports.DatabaseExadataId = DatabaseExadataId.DatabaseExadataId;
const DatetimeVariable = __importStar(require("./datetime-variable"));
exports.DatetimeVariable = DatetimeVariable.DatetimeVariable;
const DbHomeVariable = __importStar(require("./db-home-variable"));
exports.DbHomeVariable = DbHomeVariable.DbHomeVariable;
const DbHomeVersionVariable = __importStar(require("./db-home-version-variable"));
exports.DbHomeVersionVariable = DbHomeVersionVariable.DbHomeVersionVariable;
const DbSystemVariable = __importStar(require("./db-system-variable"));
exports.DbSystemVariable = DbSystemVariable.DbSystemVariable;
const DynamicGroupsVariable = __importStar(require("./dynamic-groups-variable"));
exports.DynamicGroupsVariable = DynamicGroupsVariable.DynamicGroupsVariable;
const DynamicResourceSelection = __importStar(require("./dynamic-resource-selection"));
exports.DynamicResourceSelection = DynamicResourceSelection.DynamicResourceSelection;
const EnumVariable = __importStar(require("./enum-variable"));
exports.EnumVariable = EnumVariable.EnumVariable;
const EnvironmentConfigCategoryDetails = __importStar(require("./environment-config-category-details"));
exports.EnvironmentConfigCategoryDetails = EnvironmentConfigCategoryDetails.EnvironmentConfigCategoryDetails;
const EnvironmentFleetDetails = __importStar(require("./environment-fleet-details"));
exports.EnvironmentFleetDetails = EnvironmentFleetDetails.EnvironmentFleetDetails;
const FaultDomainVariable = __importStar(require("./fault-domain-variable"));
exports.FaultDomainVariable = FaultDomainVariable.FaultDomainVariable;
const FileInputArgument = __importStar(require("./file-input-argument"));
exports.FileInputArgument = FileInputArgument.FileInputArgument;
const FileStorageExportSetId = __importStar(require("./file-storage-export-set-id"));
exports.FileStorageExportSetId = FileStorageExportSetId.FileStorageExportSetId;
const FileStorageFilesystemId = __importStar(require("./file-storage-filesystem-id"));
exports.FileStorageFilesystemId = FileStorageFilesystemId.FileStorageFilesystemId;
const FileStorageMountTargetId = __importStar(require("./file-storage-mount-target-id"));
exports.FileStorageMountTargetId = FileStorageMountTargetId.FileStorageMountTargetId;
const FileTaskArgument = __importStar(require("./file-task-argument"));
exports.FileTaskArgument = FileTaskArgument.FileTaskArgument;
const FileVariable = __importStar(require("./file-variable"));
exports.FileVariable = FileVariable.FileVariable;
const FleetBasedActionGroup = __importStar(require("./fleet-based-action-group"));
exports.FleetBasedActionGroup = FleetBasedActionGroup.FleetBasedActionGroup;
const FleetBasedActionGroupDetails = __importStar(require("./fleet-based-action-group-details"));
exports.FleetBasedActionGroupDetails = FleetBasedActionGroupDetails.FleetBasedActionGroupDetails;
const FleetCredentialEntitySpecificDetails = __importStar(require("./fleet-credential-entity-specific-details"));
exports.FleetCredentialEntitySpecificDetails = FleetCredentialEntitySpecificDetails.FleetCredentialEntitySpecificDetails;
const GenericArtifactDetails = __importStar(require("./generic-artifact-details"));
exports.GenericArtifactDetails = GenericArtifactDetails.GenericArtifactDetails;
const GenericFleetDetails = __importStar(require("./generic-fleet-details"));
exports.GenericFleetDetails = GenericFleetDetails.GenericFleetDetails;
const GroupFleetDetails = __importStar(require("./group-fleet-details"));
exports.GroupFleetDetails = GroupFleetDetails.GroupFleetDetails;
const GroupsVariable = __importStar(require("./groups-variable"));
exports.GroupsVariable = GroupsVariable.GroupsVariable;
const IdentityDomainVariable = __importStar(require("./identity-domain-variable"));
exports.IdentityDomainVariable = IdentityDomainVariable.IdentityDomainVariable;
const ImageVariable = __importStar(require("./image-variable"));
exports.ImageVariable = ImageVariable.ImageVariable;
const InputFileObjectStorageBucketContentDetails = __importStar(require("./input-file-object-storage-bucket-content-details"));
exports.InputFileObjectStorageBucketContentDetails = InputFileObjectStorageBucketContentDetails.InputFileObjectStorageBucketContentDetails;
const InstanceShapeVariable = __importStar(require("./instance-shape-variable"));
exports.InstanceShapeVariable = InstanceShapeVariable.InstanceShapeVariable;
const InstanceShapeVariableWithFlex = __importStar(require("./instance-shape-variable-with-flex"));
exports.InstanceShapeVariableWithFlex = InstanceShapeVariableWithFlex.InstanceShapeVariableWithFlex;
const InstanceVariable = __importStar(require("./instance-variable"));
exports.InstanceVariable = InstanceVariable.InstanceVariable;
const IntegerVariable = __importStar(require("./integer-variable"));
exports.IntegerVariable = IntegerVariable.IntegerVariable;
const JsonOutput = __importStar(require("./json-output"));
exports.JsonOutput = JsonOutput.JsonOutput;
const KeyEncryptionCredentialDetails = __importStar(require("./key-encryption-credential-details"));
exports.KeyEncryptionCredentialDetails = KeyEncryptionCredentialDetails.KeyEncryptionCredentialDetails;
const KmsKeyVariable = __importStar(require("./kms-key-variable"));
exports.KmsKeyVariable = KmsKeyVariable.KmsKeyVariable;
const KmsSecretVariable = __importStar(require("./kms-secret-variable"));
exports.KmsSecretVariable = KmsSecretVariable.KmsSecretVariable;
const KmsVaultVariable = __importStar(require("./kms-vault-variable"));
exports.KmsVaultVariable = KmsVaultVariable.KmsVaultVariable;
const KubernetesVersionsVariable = __importStar(require("./kubernetes-versions-variable"));
exports.KubernetesVersionsVariable = KubernetesVersionsVariable.KubernetesVersionsVariable;
const LifecycleOperationConfigCategoryDetails = __importStar(require("./lifecycle-operation-config-category-details"));
exports.LifecycleOperationConfigCategoryDetails = LifecycleOperationConfigCategoryDetails.LifecycleOperationConfigCategoryDetails;
const LinkOutput = __importStar(require("./link-output"));
exports.LinkOutput = LinkOutput.LinkOutput;
const ListOutput = __importStar(require("./list-output"));
exports.ListOutput = ListOutput.ListOutput;
const LoadBalancerReservedIps = __importStar(require("./load-balancer-reserved-ips"));
exports.LoadBalancerReservedIps = LoadBalancerReservedIps.LoadBalancerReservedIps;
const LoadBalancerVariable = __importStar(require("./load-balancer-variable"));
exports.LoadBalancerVariable = LoadBalancerVariable.LoadBalancerVariable;
const LogAnalyticsEntityTypes = __importStar(require("./log-analytics-entity-types"));
exports.LogAnalyticsEntityTypes = LogAnalyticsEntityTypes.LogAnalyticsEntityTypes;
const LogAnalyticsLogEntities = __importStar(require("./log-analytics-log-entities"));
exports.LogAnalyticsLogEntities = LogAnalyticsLogEntities.LogAnalyticsLogEntities;
const LogAnalyticsLogGroup = __importStar(require("./log-analytics-log-group"));
exports.LogAnalyticsLogGroup = LogAnalyticsLogGroup.LogAnalyticsLogGroup;
const LogAnalyticsScheduledTasks = __importStar(require("./log-analytics-scheduled-tasks"));
exports.LogAnalyticsScheduledTasks = LogAnalyticsScheduledTasks.LogAnalyticsScheduledTasks;
const LogAnalyticsSources = __importStar(require("./log-analytics-sources"));
exports.LogAnalyticsSources = LogAnalyticsSources.LogAnalyticsSources;
const MaintenanceWindowSchedule = __importStar(require("./maintenance-window-schedule"));
exports.MaintenanceWindowSchedule = MaintenanceWindowSchedule.MaintenanceWindowSchedule;
const ManagementAgents = __importStar(require("./management-agents"));
exports.ManagementAgents = ManagementAgents.ManagementAgents;
const ManualResourceSelection = __importStar(require("./manual-resource-selection"));
exports.ManualResourceSelection = ManualResourceSelection.ManualResourceSelection;
const MapOutput = __importStar(require("./map-output"));
exports.MapOutput = MapOutput.MapOutput;
const MountTargetsVariable = __importStar(require("./mount-targets-variable"));
exports.MountTargetsVariable = MountTargetsVariable.MountTargetsVariable;
const MultiValueFilter = __importStar(require("./multi-value-filter"));
exports.MultiValueFilter = MultiValueFilter.MultiValueFilter;
const MultilineVariable = __importStar(require("./multiline-variable"));
exports.MultilineVariable = MultilineVariable.MultilineVariable;
const MysqlConfigurationId = __importStar(require("./mysql-configuration-id"));
exports.MysqlConfigurationId = MysqlConfigurationId.MysqlConfigurationId;
const MysqlShapeId = __importStar(require("./mysql-shape-id"));
exports.MysqlShapeId = MysqlShapeId.MysqlShapeId;
const NatGatewayVariable = __importStar(require("./nat-gateway-variable"));
exports.NatGatewayVariable = NatGatewayVariable.NatGatewayVariable;
const NetworkLoadBalancerId = __importStar(require("./network-load-balancer-id"));
exports.NetworkLoadBalancerId = NetworkLoadBalancerId.NetworkLoadBalancerId;
const NsgVariable = __importStar(require("./nsg-variable"));
exports.NsgVariable = NsgVariable.NsgVariable;
const NumberOutput = __importStar(require("./number-output"));
exports.NumberOutput = NumberOutput.NumberOutput;
const NumberVariable = __importStar(require("./number-variable"));
exports.NumberVariable = NumberVariable.NumberVariable;
const ObjectStorageBucketConfigFileDetails = __importStar(require("./object-storage-bucket-config-file-details"));
exports.ObjectStorageBucketConfigFileDetails = ObjectStorageBucketConfigFileDetails.ObjectStorageBucketConfigFileDetails;
const ObjectStorageBucketContentDetails = __importStar(require("./object-storage-bucket-content-details"));
exports.ObjectStorageBucketContentDetails = ObjectStorageBucketContentDetails.ObjectStorageBucketContentDetails;
const ObjectStorageBucketName = __importStar(require("./object-storage-bucket-name"));
exports.ObjectStorageBucketName = ObjectStorageBucketName.ObjectStorageBucketName;
const OcidOutput = __importStar(require("./ocid-output"));
exports.OcidOutput = OcidOutput.OcidOutput;
const OdsProjectVariable = __importStar(require("./ods-project-variable"));
exports.OdsProjectVariable = OdsProjectVariable.OdsProjectVariable;
const OutputVariableInputArgument = __importStar(require("./output-variable-input-argument"));
exports.OutputVariableInputArgument = OutputVariableInputArgument.OutputVariableInputArgument;
const PasswordVariable = __importStar(require("./password-variable"));
exports.PasswordVariable = PasswordVariable.PasswordVariable;
const PatchFileObjectStorageBucketContentDetails = __importStar(require("./patch-file-object-storage-bucket-content-details"));
exports.PatchFileObjectStorageBucketContentDetails = PatchFileObjectStorageBucketContentDetails.PatchFileObjectStorageBucketContentDetails;
const PatchLevelSelectionDetails = __importStar(require("./patch-level-selection-details"));
exports.PatchLevelSelectionDetails = PatchLevelSelectionDetails.PatchLevelSelectionDetails;
const PatchNameSelectionDetails = __importStar(require("./patch-name-selection-details"));
exports.PatchNameSelectionDetails = PatchNameSelectionDetails.PatchNameSelectionDetails;
const PatchReleaseDateSelectionDetails = __importStar(require("./patch-release-date-selection-details"));
exports.PatchReleaseDateSelectionDetails = PatchReleaseDateSelectionDetails.PatchReleaseDateSelectionDetails;
const PatchTypeConfigCategoryDetails = __importStar(require("./patch-type-config-category-details"));
exports.PatchTypeConfigCategoryDetails = PatchTypeConfigCategoryDetails.PatchTypeConfigCategoryDetails;
const PlainTextCredentialDetails = __importStar(require("./plain-text-credential-details"));
exports.PlainTextCredentialDetails = PlainTextCredentialDetails.PlainTextCredentialDetails;
const PlatformSpecificArtifactDetails = __importStar(require("./platform-specific-artifact-details"));
exports.PlatformSpecificArtifactDetails = PlatformSpecificArtifactDetails.PlatformSpecificArtifactDetails;
const PreviousTaskInstanceRunOnDetails = __importStar(require("./previous-task-instance-run-on-details"));
exports.PreviousTaskInstanceRunOnDetails = PreviousTaskInstanceRunOnDetails.PreviousTaskInstanceRunOnDetails;
const PrivateEndpointVariable = __importStar(require("./private-endpoint-variable"));
exports.PrivateEndpointVariable = PrivateEndpointVariable.PrivateEndpointVariable;
const ProductConfigCategoryDetails = __importStar(require("./product-config-category-details"));
exports.ProductConfigCategoryDetails = ProductConfigCategoryDetails.ProductConfigCategoryDetails;
const ProductFleetDetails = __importStar(require("./product-fleet-details"));
exports.ProductFleetDetails = ProductFleetDetails.ProductFleetDetails;
const ProductStackAsProductSubCategoryDetails = __importStar(require("./product-stack-as-product-sub-category-details"));
exports.ProductStackAsProductSubCategoryDetails = ProductStackAsProductSubCategoryDetails.ProductStackAsProductSubCategoryDetails;
const ProductStackConfigCategoryDetails = __importStar(require("./product-stack-config-category-details"));
exports.ProductStackConfigCategoryDetails = ProductStackConfigCategoryDetails.ProductStackConfigCategoryDetails;
const ProductStackGenericSubCategoryDetails = __importStar(require("./product-stack-generic-sub-category-details"));
exports.ProductStackGenericSubCategoryDetails = ProductStackGenericSubCategoryDetails.ProductStackGenericSubCategoryDetails;
const RegionVariable = __importStar(require("./region-variable"));
exports.RegionVariable = RegionVariable.RegionVariable;
const ResourceCredentialEntitySpecificDetails = __importStar(require("./resource-credential-entity-specific-details"));
exports.ResourceCredentialEntitySpecificDetails = ResourceCredentialEntitySpecificDetails.ResourceCredentialEntitySpecificDetails;
const RunbookImportAsRunbook = __importStar(require("./runbook-import-as-runbook"));
exports.RunbookImportAsRunbook = RunbookImportAsRunbook.RunbookImportAsRunbook;
const RunbookImportAsVersion = __importStar(require("./runbook-import-as-version"));
exports.RunbookImportAsVersion = RunbookImportAsVersion.RunbookImportAsVersion;
const ScheduleInstanceRunOnDetails = __importStar(require("./schedule-instance-run-on-details"));
exports.ScheduleInstanceRunOnDetails = ScheduleInstanceRunOnDetails.ScheduleInstanceRunOnDetails;
const ScheduleInstanceTargetRunOnDetails = __importStar(require("./schedule-instance-target-run-on-details"));
exports.ScheduleInstanceTargetRunOnDetails = ScheduleInstanceTargetRunOnDetails.ScheduleInstanceTargetRunOnDetails;
const ScriptBasedExecutionDetails = __importStar(require("./script-based-execution-details"));
exports.ScriptBasedExecutionDetails = ScriptBasedExecutionDetails.ScriptBasedExecutionDetails;
const SelfHostedInstanceConfigCategoryDetails = __importStar(require("./self-hosted-instance-config-category-details"));
exports.SelfHostedInstanceConfigCategoryDetails = SelfHostedInstanceConfigCategoryDetails.SelfHostedInstanceConfigCategoryDetails;
const SelfHostedInstanceRunOnDetails = __importStar(require("./self-hosted-instance-run-on-details"));
exports.SelfHostedInstanceRunOnDetails = SelfHostedInstanceRunOnDetails.SelfHostedInstanceRunOnDetails;
const ServiceGatewayVariable = __importStar(require("./service-gateway-variable"));
exports.ServiceGatewayVariable = ServiceGatewayVariable.ServiceGatewayVariable;
const SingleValueFilter = __importStar(require("./single-value-filter"));
exports.SingleValueFilter = SingleValueFilter.SingleValueFilter;
const SshPublicKeyVariable = __importStar(require("./ssh-public-key-variable"));
exports.SshPublicKeyVariable = SshPublicKeyVariable.SshPublicKeyVariable;
const StepBasedUserActionDetails = __importStar(require("./step-based-user-action-details"));
exports.StepBasedUserActionDetails = StepBasedUserActionDetails.StepBasedUserActionDetails;
const StringInputArgument = __importStar(require("./string-input-argument"));
exports.StringInputArgument = StringInputArgument.StringInputArgument;
const StringOutput = __importStar(require("./string-output"));
exports.StringOutput = StringOutput.StringOutput;
const StringTaskArgument = __importStar(require("./string-task-argument"));
exports.StringTaskArgument = StringTaskArgument.StringTaskArgument;
const StringVariable = __importStar(require("./string-variable"));
exports.StringVariable = StringVariable.StringVariable;
const SubnetVariable = __importStar(require("./subnet-variable"));
exports.SubnetVariable = SubnetVariable.SubnetVariable;
const TagVariable = __importStar(require("./tag-variable"));
exports.TagVariable = TagVariable.TagVariable;
const TargetCredentialEntitySpecificDetails = __importStar(require("./target-credential-entity-specific-details"));
exports.TargetCredentialEntitySpecificDetails = TargetCredentialEntitySpecificDetails.TargetCredentialEntitySpecificDetails;
const TerraformBasedExecutionDetails = __importStar(require("./terraform-based-execution-details"));
exports.TerraformBasedExecutionDetails = TerraformBasedExecutionDetails.TerraformBasedExecutionDetails;
const TimeBasedPauseDetails = __importStar(require("./time-based-pause-details"));
exports.TimeBasedPauseDetails = TimeBasedPauseDetails.TimeBasedPauseDetails;
const TransferRunbookObjectStorageBucketContentDetails = __importStar(require("./transfer-runbook-object-storage-bucket-content-details"));
exports.TransferRunbookObjectStorageBucketContentDetails = TransferRunbookObjectStorageBucketContentDetails.TransferRunbookObjectStorageBucketContentDetails;
const TransferRunbookParUrlContentDetails = __importStar(require("./transfer-runbook-par-url-content-details"));
exports.TransferRunbookParUrlContentDetails = TransferRunbookParUrlContentDetails.TransferRunbookParUrlContentDetails;
const UserActionBasedPauseDetails = __importStar(require("./user-action-based-pause-details"));
exports.UserActionBasedPauseDetails = UserActionBasedPauseDetails.UserActionBasedPauseDetails;
const VaultSecretCredentialDetails = __importStar(require("./vault-secret-credential-details"));
exports.VaultSecretCredentialDetails = VaultSecretCredentialDetails.VaultSecretCredentialDetails;
const VcnNetworkSecurityGroupId = __importStar(require("./vcn-network-security-group-id"));
exports.VcnNetworkSecurityGroupId = VcnNetworkSecurityGroupId.VcnNetworkSecurityGroupId;
const VcnSecListId = __importStar(require("./vcn-sec-list-id"));
exports.VcnSecListId = VcnSecListId.VcnSecListId;
const VcnVariable = __importStar(require("./vcn-variable"));
exports.VcnVariable = VcnVariable.VcnVariable;
const VolumeBackupPoliciesVariable = __importStar(require("./volume-backup-policies-variable"));
exports.VolumeBackupPoliciesVariable = VolumeBackupPoliciesVariable.VolumeBackupPoliciesVariable;
const VolumeId = __importStar(require("./volume-id"));
exports.VolumeId = VolumeId.VolumeId;
const WorkflowGroupComponent = __importStar(require("./workflow-group-component"));
exports.WorkflowGroupComponent = WorkflowGroupComponent.WorkflowGroupComponent;
const WorkflowTaskComponent = __importStar(require("./workflow-task-component"));
exports.WorkflowTaskComponent = WorkflowTaskComponent.WorkflowTaskComponent;
