// oci-fleetappsmanagement/lib/model/recommended-patch-summary.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RecommendedPatchSummary = void 0;
var RecommendedPatchSummary;
(function (RecommendedPatchSummary) {
    let PatchLevel;
    (function (PatchLevel) {
        PatchLevel["Latest"] = "LATEST";
        PatchLevel["LatestMinusOne"] = "LATEST_MINUS_ONE";
        PatchLevel["LatestMinusTwo"] = "LATEST_MINUS_TWO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PatchLevel["UnknownValue"] = "UNKNOWN_VALUE";
    })(PatchLevel = RecommendedPatchSummary.PatchLevel || (RecommendedPatchSummary.PatchLevel = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RecommendedPatchSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RecommendedPatchSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(RecommendedPatchSummary = exports.RecommendedPatchSummary || (exports.RecommendedPatchSummary = {}));
