// oci-fleetappsmanagement/lib/model/runbook-import-dependency-finder-required-value.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RunbookImportDependencyFinderRequiredValue = void 0;
var RunbookImportDependencyFinderRequiredValue;
(function (RunbookImportDependencyFinderRequiredValue) {
    let RenderingComponent;
    (function (RenderingComponent) {
        RenderingComponent["SingleChoice"] = "SINGLE_CHOICE";
        RenderingComponent["MultiChoice"] = "MULTI_CHOICE";
        RenderingComponent["InputText"] = "INPUT_TEXT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        RenderingComponent["UnknownValue"] = "UNKNOWN_VALUE";
    })(RenderingComponent = RunbookImportDependencyFinderRequiredValue.RenderingComponent || (RunbookImportDependencyFinderRequiredValue.RenderingComponent = {}));
    let ExpectedAssignment;
    (function (ExpectedAssignment) {
        ExpectedAssignment["TfPackage"] = "TF_PACKAGE";
        ExpectedAssignment["NonTfPackage"] = "NON_TF_PACKAGE";
        ExpectedAssignment["ConfigFile"] = "CONFIG_FILE";
        ExpectedAssignment["Null"] = "NULL";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ExpectedAssignment["UnknownValue"] = "UNKNOWN_VALUE";
    })(ExpectedAssignment = RunbookImportDependencyFinderRequiredValue.ExpectedAssignment || (RunbookImportDependencyFinderRequiredValue.ExpectedAssignment = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RunbookImportDependencyFinderRequiredValue.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    RunbookImportDependencyFinderRequiredValue.getDeserializedJsonObj = getDeserializedJsonObj;
})(RunbookImportDependencyFinderRequiredValue = exports.RunbookImportDependencyFinderRequiredValue || (exports.RunbookImportDependencyFinderRequiredValue = {}));
