// oci-fleetappsmanagement/lib/model/schema-document.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaDocument = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var SchemaDocument;
(function (SchemaDocument) {
    let SchemaVersion;
    (function (SchemaVersion) {
        SchemaVersion["V100"] = "V_1_0_0";
        SchemaVersion["V110"] = "V_1_1_0";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        SchemaVersion["UnknownValue"] = "UNKNOWN_VALUE";
    })(SchemaVersion = SchemaDocument.SchemaVersion || (SchemaDocument.SchemaVersion = {}));
    let Locale;
    (function (Locale) {
        Locale["En"] = "EN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Locale["UnknownValue"] = "UNKNOWN_VALUE";
    })(Locale = SchemaDocument.Locale || (SchemaDocument.Locale = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "source": obj.source ? model.StackSource.getJsonObj(obj.source) : undefined,
            "variables": obj.variables
                ? common.mapContainer(obj.variables, model.BaseVariable.getJsonObj)
                : undefined,
            "groupings": obj.groupings ? model.VariableGroups.getJsonObj(obj.groupings) : undefined,
            "variableGroups": obj.variableGroups
                ? obj.variableGroups.map(item => {
                    return model.VariableGroup.getJsonObj(item);
                })
                : undefined,
            "outputs": obj.outputs
                ? common.mapContainer(obj.outputs, model.BaseOutput.getJsonObj)
                : undefined,
            "outputGroups": obj.outputGroups
                ? obj.outputGroups.map(item => {
                    return model.OutputGroup.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    SchemaDocument.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "source": obj.source ? model.StackSource.getDeserializedJsonObj(obj.source) : undefined,
            "variables": obj.variables
                ? common.mapContainer(obj.variables, model.BaseVariable.getDeserializedJsonObj)
                : undefined,
            "groupings": obj.groupings
                ? model.VariableGroups.getDeserializedJsonObj(obj.groupings)
                : undefined,
            "variableGroups": obj.variableGroups
                ? obj.variableGroups.map(item => {
                    return model.VariableGroup.getDeserializedJsonObj(item);
                })
                : undefined,
            "outputs": obj.outputs
                ? common.mapContainer(obj.outputs, model.BaseOutput.getDeserializedJsonObj)
                : undefined,
            "outputGroups": obj.outputGroups
                ? obj.outputGroups.map(item => {
                    return model.OutputGroup.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    SchemaDocument.getDeserializedJsonObj = getDeserializedJsonObj;
})(SchemaDocument = exports.SchemaDocument || (exports.SchemaDocument = {}));
