// oci-fleetappsmanagement/lib/model/single-value-filter.js
"use strict";
/**
 * Fleet Application Management Service API
 * Fleet Application Management provides a centralized platform to help you automate resource management tasks, validate patch compliance, and enhance operational efficiency across an enterprise.

 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SingleValueFilter = void 0;
const model = __importStar(require("../model"));
var SingleValueFilter;
(function (SingleValueFilter) {
    let Expression;
    (function (Expression) {
        Expression["Equal"] = "EQUAL";
        Expression["NotEqual"] = "NOT_EQUAL";
        Expression["LessThan"] = "LESS_THAN";
        Expression["GreaterThan"] = "GREATER_THAN";
        Expression["LessThanOrEqualTo"] = "LESS_THAN_OR_EQUAL_TO";
        Expression["GreaterThanOrEqualTo"] = "GREATER_THAN_OR_EQUAL_TO";
    })(Expression = SingleValueFilter.Expression || (SingleValueFilter.Expression = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj ? obj : model.Filter.getJsonObj(obj))), {});
        return jsonObj;
    }
    SingleValueFilter.getJsonObj = getJsonObj;
    SingleValueFilter.filterType = "SINGLE_VALUE";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj ? obj : model.Filter.getDeserializedJsonObj(obj))), {});
        return jsonObj;
    }
    SingleValueFilter.getDeserializedJsonObj = getDeserializedJsonObj;
})(SingleValueFilter = exports.SingleValueFilter || (exports.SingleValueFilter = {}));
