// oci-dbmulticloud/index.js
"use strict";
/**
 * Oracle Database MultiCloud Data Plane Integration
 * <b>Microsoft Azure:</b> <br>
<b>Oracle Azure Connector Resource:</b>:&nbsp;&nbsp;The Oracle Azure Connector Resource is used to install the Azure Arc Server on an Exadata VM cluster in Oracle Exadata Database Service on Dedicated Infrastructure (ExaDB-D).
 The supported method to install the Azure Arc Server (Azure Identity) on the Exadata VM cluster:
<ul>
 <li>Using a Bearer Access Token</li>
</ul>

<b>Oracle Azure Blob Container Resource:</b>&nbsp;&nbsp;The Oracle Azure Blob Container Resource is used to capture the details of an Azure Blob Container.
This resource can then be reused across multiple Exadata VM clusters in Oracle Exadata Database Service on Dedicated Infrastructure (ExaDB-D) to mount the Azure container.

<b>Oracle Azure Blob Mount Resource:</b>&nbsp;&nbsp;The Oracle Azure Blob Mount Resource is used to mount an Azure Blob Container on an Exadata VM cluster in Oracle Exadata Database Service on Dedicated Infrastructure (ExaDB-D).
It relies on both the Oracle Azure Connector and the Oracle Azure Blob Container Resource to perform the mount operation.

<b>Discover Azure Vaults and Keys Resource:</b>&nbsp;&nbsp;The Discover Oracle Azure Vaults and Azure Keys Resource is used to discover Azure Vaults and the associated encryption keys available in your Azure project.

<b>Oracle Azure Vault:</b>&nbsp;&nbsp;The Oracle Azure Vault Resource is used to manage Azure Vaults within Oracle Cloud Infrastructure (OCI) for use with services such as Oracle Exadata Database Service on Dedicated Infrastructure.

<b>Oracle Azure Key:</b>&nbsp;&nbsp;Oracle Azure Key Resource is used to register and manage a Oracle Azure Key Key within Oracle Cloud Infrastructure (OCI) under an associated Azure Vault.

<br>

<b>Google Cloud:</b><br>
<b>Oracle Google Cloud Connector Resource:</b>&nbsp;&nbsp;The Oracle Google Cloud Connector Resource is used to install the Google Cloud Identity Connector on an Exadata VM cluster in Oracle Exadata Database Service on Dedicated Infrastructure (ExaDB-D).

<b>Discover Google Key Rings and Keys Resource:</b>&nbsp;&nbsp;The Discover Google Key Rings and Keys Resource is used to discover Google Cloud Key Rings and the associated encryption keys available in your Google Cloud project.

<b>Google Key Rings Resource:</b>&nbsp;&nbsp;The Google Key Rings Resource is used to register and manage Google Cloud Key Rings within Oracle Cloud Infrastructure (OCI) for use with services such as Oracle Exadata Database Service on Dedicated Infrastructure.

<b>Google Key Resource:</b>&nbsp;&nbsp;The Google Key Resource is used to register and manage a Google Cloud Key within Oracle Cloud Infrastructure (OCI) under an associated Google Key Ring.
<br>

<b>AWS</b>:<br>
<b>Oracle AWS Connector Resource:</b>&nbsp;&nbsp;The Oracle AWS Connector Resource is used to install the AWS Identity Connector on an Exadata VM cluster in Oracle Exadata Database Service on Dedicated Infrastructure (ExaDB-D).

<b>Google AWS Key Resource:</b>&nbsp;&nbsp;The Oracle AWS Key Resource is used to register and manage a AWS Key within Oracle Cloud Infrastructure (OCI).

 * OpenAPI spec version: 20240501
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestWaiter = exports.WorkRequestClient = exports.OracleDbAzureVaultAssociationWaiter = exports.OracleDbAzureVaultAssociationClient = exports.OracleDbAzureVaultWaiter = exports.OracleDbAzureVaultClient = exports.OracleDbAzureKeyWaiter = exports.OracleDbAzureKeyClient = exports.OracleDBAzureConnectorWaiter = exports.OracleDBAzureConnectorClient = exports.OracleDBAzureBlobMountWaiter = exports.OracleDBAzureBlobMountClient = exports.OracleDBAzureBlobContainerWaiter = exports.OracleDBAzureBlobContainerClient = exports.MultiCloudResourceDiscoveryWaiter = exports.MultiCloudResourceDiscoveryClient = exports.DbMulticloudGCPProviderWaiter = exports.DbMulticloudGCPProviderClient = exports.DbMulticloudAwsProviderWaiter = exports.DbMulticloudAwsProviderClient = exports.responses = exports.requests = exports.models = void 0;
const requests = __importStar(require("./lib/request"));
exports.requests = requests;
const models = __importStar(require("./lib/model"));
exports.models = models;
const responses = __importStar(require("./lib/response"));
exports.responses = responses;
const client = __importStar(require("./lib/client"));
const dbmulticloudawsprovider_waiter = __importStar(require("./lib/dbmulticloudawsprovider-waiter"));
const dbmulticloudgcpprovider_waiter = __importStar(require("./lib/dbmulticloudgcpprovider-waiter"));
const multicloudresourcediscovery_waiter = __importStar(require("./lib/multicloudresourcediscovery-waiter"));
const oracledbazureblobcontainer_waiter = __importStar(require("./lib/oracledbazureblobcontainer-waiter"));
const oracledbazureblobmount_waiter = __importStar(require("./lib/oracledbazureblobmount-waiter"));
const oracledbazureconnector_waiter = __importStar(require("./lib/oracledbazureconnector-waiter"));
const oracledbazurekey_waiter = __importStar(require("./lib/oracledbazurekey-waiter"));
const oracledbazurevault_waiter = __importStar(require("./lib/oracledbazurevault-waiter"));
const oracledbazurevaultassociation_waiter = __importStar(require("./lib/oracledbazurevaultassociation-waiter"));
const workrequest_waiter = __importStar(require("./lib/workrequest-waiter"));
exports.DbMulticloudAwsProviderClient = client.DbMulticloudAwsProviderClient;
exports.DbMulticloudAwsProviderWaiter = dbmulticloudawsprovider_waiter.DbMulticloudAwsProviderWaiter;
exports.DbMulticloudGCPProviderClient = client.DbMulticloudGCPProviderClient;
exports.DbMulticloudGCPProviderWaiter = dbmulticloudgcpprovider_waiter.DbMulticloudGCPProviderWaiter;
exports.MultiCloudResourceDiscoveryClient = client.MultiCloudResourceDiscoveryClient;
exports.MultiCloudResourceDiscoveryWaiter = multicloudresourcediscovery_waiter.MultiCloudResourceDiscoveryWaiter;
exports.OracleDBAzureBlobContainerClient = client.OracleDBAzureBlobContainerClient;
exports.OracleDBAzureBlobContainerWaiter = oracledbazureblobcontainer_waiter.OracleDBAzureBlobContainerWaiter;
exports.OracleDBAzureBlobMountClient = client.OracleDBAzureBlobMountClient;
exports.OracleDBAzureBlobMountWaiter = oracledbazureblobmount_waiter.OracleDBAzureBlobMountWaiter;
exports.OracleDBAzureConnectorClient = client.OracleDBAzureConnectorClient;
exports.OracleDBAzureConnectorWaiter = oracledbazureconnector_waiter.OracleDBAzureConnectorWaiter;
exports.OracleDbAzureKeyClient = client.OracleDbAzureKeyClient;
exports.OracleDbAzureKeyWaiter = oracledbazurekey_waiter.OracleDbAzureKeyWaiter;
exports.OracleDbAzureVaultClient = client.OracleDbAzureVaultClient;
exports.OracleDbAzureVaultWaiter = oracledbazurevault_waiter.OracleDbAzureVaultWaiter;
exports.OracleDbAzureVaultAssociationClient = client.OracleDbAzureVaultAssociationClient;
exports.OracleDbAzureVaultAssociationWaiter = oracledbazurevaultassociation_waiter.OracleDbAzureVaultAssociationWaiter;
exports.WorkRequestClient = client.WorkRequestClient;
exports.WorkRequestWaiter = workrequest_waiter.WorkRequestWaiter;
