// oci-jmsutils/lib/model/index.js
"use strict";
/**
 * Java Management Service Utilities API
 * The APIs for Analyze Applications and other utilities of Java Management Service.
 * OpenAPI spec version: 20250521
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PerformanceTuningWorkItemDetails = exports.JavaMigrationWorkItemDetails = exports.BasicWorkItemDetails = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.WorkItemType = exports.WorkItemSummary = exports.WorkItemStatus = exports.WorkItemDetails = exports.WorkItemCollection = exports.UpdateSubscriptionAcknowledgmentConfigurationDetails = exports.UpdateAnalyzeApplicationsConfigurationDetails = exports.SubscriptionAcknowledgmentConfiguration = exports.SortOrder = exports.RequestPerformanceTuningAnalysisDetails = exports.RequestJavaMigrationAnalysisDetails = exports.Principal = exports.PerformanceTuningAnalysisTarget = exports.PerformanceTuningAnalysisSummary = exports.PerformanceTuningAnalysisSortBy = exports.PerformanceTuningAnalysisCollection = exports.PerformanceTuningAnalysis = exports.OperationType = exports.OperationStatus = exports.JavaMigrationAnalysisTarget = exports.JavaMigrationAnalysisSummary = exports.JavaMigrationAnalysisSortBy = exports.JavaMigrationAnalysisCollection = exports.JavaMigrationAnalysis = exports.AnalyzeApplicationsConfiguration = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AnalyzeApplicationsConfiguration = __importStar(require("./analyze-applications-configuration"));
exports.AnalyzeApplicationsConfiguration = AnalyzeApplicationsConfiguration.AnalyzeApplicationsConfiguration;
const JavaMigrationAnalysis = __importStar(require("./java-migration-analysis"));
exports.JavaMigrationAnalysis = JavaMigrationAnalysis.JavaMigrationAnalysis;
const JavaMigrationAnalysisCollection = __importStar(require("./java-migration-analysis-collection"));
exports.JavaMigrationAnalysisCollection = JavaMigrationAnalysisCollection.JavaMigrationAnalysisCollection;
const JavaMigrationAnalysisSortBy = __importStar(require("./java-migration-analysis-sort-by"));
exports.JavaMigrationAnalysisSortBy = JavaMigrationAnalysisSortBy.JavaMigrationAnalysisSortBy;
const JavaMigrationAnalysisSummary = __importStar(require("./java-migration-analysis-summary"));
exports.JavaMigrationAnalysisSummary = JavaMigrationAnalysisSummary.JavaMigrationAnalysisSummary;
const JavaMigrationAnalysisTarget = __importStar(require("./java-migration-analysis-target"));
exports.JavaMigrationAnalysisTarget = JavaMigrationAnalysisTarget.JavaMigrationAnalysisTarget;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PerformanceTuningAnalysis = __importStar(require("./performance-tuning-analysis"));
exports.PerformanceTuningAnalysis = PerformanceTuningAnalysis.PerformanceTuningAnalysis;
const PerformanceTuningAnalysisCollection = __importStar(require("./performance-tuning-analysis-collection"));
exports.PerformanceTuningAnalysisCollection = PerformanceTuningAnalysisCollection.PerformanceTuningAnalysisCollection;
const PerformanceTuningAnalysisSortBy = __importStar(require("./performance-tuning-analysis-sort-by"));
exports.PerformanceTuningAnalysisSortBy = PerformanceTuningAnalysisSortBy.PerformanceTuningAnalysisSortBy;
const PerformanceTuningAnalysisSummary = __importStar(require("./performance-tuning-analysis-summary"));
exports.PerformanceTuningAnalysisSummary = PerformanceTuningAnalysisSummary.PerformanceTuningAnalysisSummary;
const PerformanceTuningAnalysisTarget = __importStar(require("./performance-tuning-analysis-target"));
exports.PerformanceTuningAnalysisTarget = PerformanceTuningAnalysisTarget.PerformanceTuningAnalysisTarget;
const Principal = __importStar(require("./principal"));
exports.Principal = Principal.Principal;
const RequestJavaMigrationAnalysisDetails = __importStar(require("./request-java-migration-analysis-details"));
exports.RequestJavaMigrationAnalysisDetails = RequestJavaMigrationAnalysisDetails.RequestJavaMigrationAnalysisDetails;
const RequestPerformanceTuningAnalysisDetails = __importStar(require("./request-performance-tuning-analysis-details"));
exports.RequestPerformanceTuningAnalysisDetails = RequestPerformanceTuningAnalysisDetails.RequestPerformanceTuningAnalysisDetails;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const SubscriptionAcknowledgmentConfiguration = __importStar(require("./subscription-acknowledgment-configuration"));
exports.SubscriptionAcknowledgmentConfiguration = SubscriptionAcknowledgmentConfiguration.SubscriptionAcknowledgmentConfiguration;
const UpdateAnalyzeApplicationsConfigurationDetails = __importStar(require("./update-analyze-applications-configuration-details"));
exports.UpdateAnalyzeApplicationsConfigurationDetails = UpdateAnalyzeApplicationsConfigurationDetails.UpdateAnalyzeApplicationsConfigurationDetails;
const UpdateSubscriptionAcknowledgmentConfigurationDetails = __importStar(require("./update-subscription-acknowledgment-configuration-details"));
exports.UpdateSubscriptionAcknowledgmentConfigurationDetails = UpdateSubscriptionAcknowledgmentConfigurationDetails.UpdateSubscriptionAcknowledgmentConfigurationDetails;
const WorkItemCollection = __importStar(require("./work-item-collection"));
exports.WorkItemCollection = WorkItemCollection.WorkItemCollection;
const WorkItemDetails = __importStar(require("./work-item-details"));
exports.WorkItemDetails = WorkItemDetails.WorkItemDetails;
const WorkItemStatus = __importStar(require("./work-item-status"));
exports.WorkItemStatus = WorkItemStatus.WorkItemStatus;
const WorkItemSummary = __importStar(require("./work-item-summary"));
exports.WorkItemSummary = WorkItemSummary.WorkItemSummary;
const WorkItemType = __importStar(require("./work-item-type"));
exports.WorkItemType = WorkItemType.WorkItemType;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const BasicWorkItemDetails = __importStar(require("./basic-work-item-details"));
exports.BasicWorkItemDetails = BasicWorkItemDetails.BasicWorkItemDetails;
const JavaMigrationWorkItemDetails = __importStar(require("./java-migration-work-item-details"));
exports.JavaMigrationWorkItemDetails = JavaMigrationWorkItemDetails.JavaMigrationWorkItemDetails;
const PerformanceTuningWorkItemDetails = __importStar(require("./performance-tuning-work-item-details"));
exports.PerformanceTuningWorkItemDetails = PerformanceTuningWorkItemDetails.PerformanceTuningWorkItemDetails;
