// oci-jmsutils/lib/model/performance-tuning-analysis-sort-by.js
"use strict";
/**
 * Java Management Service Utilities API
 * The APIs for Analyze Applications and other utilities of Java Management Service.
 * OpenAPI spec version: 20250521
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PerformanceTuningAnalysisSortBy = void 0;
/**
 * The field to sort Performance Tuning analysis.
 **/
var PerformanceTuningAnalysisSortBy;
(function (PerformanceTuningAnalysisSortBy) {
    PerformanceTuningAnalysisSortBy["TimeCreated"] = "TIME_CREATED";
    PerformanceTuningAnalysisSortBy["TimeStarted"] = "TIME_STARTED";
    PerformanceTuningAnalysisSortBy["TimeFinished"] = "TIME_FINISHED";
})(PerformanceTuningAnalysisSortBy = exports.PerformanceTuningAnalysisSortBy || (exports.PerformanceTuningAnalysisSortBy = {}));
(function (PerformanceTuningAnalysisSortBy) {
    function getJsonObj(obj) {
        return obj;
    }
    PerformanceTuningAnalysisSortBy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    PerformanceTuningAnalysisSortBy.getDeserializedJsonObj = getDeserializedJsonObj;
})(PerformanceTuningAnalysisSortBy = exports.PerformanceTuningAnalysisSortBy || (exports.PerformanceTuningAnalysisSortBy = {}));
