// oci-jmsutils/lib/model/work-item-status.js
"use strict";
/**
 * Java Management Service Utilities API
 * The APIs for Analyze Applications and other utilities of Java Management Service.
 * OpenAPI spec version: 20250521
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkItemStatus = void 0;
/**
 * Possible operation status.
 **/
var WorkItemStatus;
(function (WorkItemStatus) {
    WorkItemStatus["Accepted"] = "ACCEPTED";
    WorkItemStatus["InProgress"] = "IN_PROGRESS";
    WorkItemStatus["Canceling"] = "CANCELING";
    WorkItemStatus["Canceled"] = "CANCELED";
    WorkItemStatus["Succeeded"] = "SUCCEEDED";
    WorkItemStatus["NeedsAttention"] = "NEEDS_ATTENTION";
    WorkItemStatus["Retrying"] = "RETRYING";
    WorkItemStatus["Skipped"] = "SKIPPED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    WorkItemStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(WorkItemStatus = exports.WorkItemStatus || (exports.WorkItemStatus = {}));
(function (WorkItemStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    WorkItemStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    WorkItemStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(WorkItemStatus = exports.WorkItemStatus || (exports.WorkItemStatus = {}));
