// oci-apiplatform/lib/model/operation-type.js
"use strict";
/**
 * APIP Control Plane API
 * Control Plane designed to manage lifecycle of APIP Instances
 * OpenAPI spec version: 20240829
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * The asynchronous operation tracked by the work request.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateApipInstance"] = "CREATE_APIP_INSTANCE";
    OperationType["UpdateApipInstance"] = "UPDATE_APIP_INSTANCE";
    OperationType["DeleteApipInstance"] = "DELETE_APIP_INSTANCE";
    OperationType["MoveApipInstance"] = "MOVE_APIP_INSTANCE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
