// oci-iot/lib/model/index.js
"use strict";
/**
 * Internet of Things API
 * Use the Internet of Things (IoT) API to manage IoT domain groups, domains, and digital twin resources including models, adapters, instances, and relationships.
For more information, see [Internet of Things](https://docs.oracle.com/iaas/Content/internet-of-things/home.htm).

 * OpenAPI spec version: 20250531
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.Vcn = exports.UpdateIotDomainGroupDetails = exports.UpdateIotDomainDetails = exports.UpdateDigitalTwinRelationshipDetails = exports.UpdateDigitalTwinModelDetails = exports.UpdateDigitalTwinInstanceDetails = exports.UpdateDigitalTwinAdapterDetails = exports.OperationType = exports.OperationStatus = exports.LifecycleState = exports.IotDomainSummary = exports.IotDomainGroupSummary = exports.IotDomainGroupCollection = exports.IotDomainGroup = exports.IotDomainCollection = exports.IotDomain = exports.InvokeRawCommandDetails = exports.ErrorDetails = exports.DigitalTwinRelationshipSummary = exports.DigitalTwinRelationshipCollection = exports.DigitalTwinRelationship = exports.DigitalTwinModelSummary = exports.DigitalTwinModelCollection = exports.DigitalTwinModel = exports.DigitalTwinInstanceSummary = exports.DigitalTwinInstanceCollection = exports.DigitalTwinInstance = exports.DigitalTwinAdapterSummary = exports.DigitalTwinAdapterPayload = exports.DigitalTwinAdapterInboundRoute = exports.DigitalTwinAdapterInboundEnvelope = exports.DigitalTwinAdapterEnvelopeMapping = exports.DigitalTwinAdapterCollection = exports.DigitalTwinAdapter = exports.DataRetentionPeriodsInDays = exports.CreateIotDomainGroupDetails = exports.CreateIotDomainDetails = exports.CreateDigitalTwinRelationshipDetails = exports.CreateDigitalTwinModelDetails = exports.CreateDigitalTwinInstanceDetails = exports.CreateDigitalTwinAdapterDetails = exports.ConfigureIotDomainGroupDataAccessDetails = exports.ConfigureIotDomainDataAccessDetails = exports.ChangeIotDomainGroupCompartmentDetails = exports.ChangeIotDomainDataRetentionPeriodDetails = exports.ChangeIotDomainCompartmentDetails = exports.ActionType = void 0;
exports.OrdsDataAccessDetails = exports.InvokeRawTextCommandDetails = exports.InvokeRawJsonCommandDetails = exports.InvokeRawBinaryCommandDetails = exports.DirectDataAccessDetails = exports.DigitalTwinAdapterJsonPayload = exports.ApexDataAccessDetails = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ChangeIotDomainCompartmentDetails = __importStar(require("./change-iot-domain-compartment-details"));
exports.ChangeIotDomainCompartmentDetails = ChangeIotDomainCompartmentDetails.ChangeIotDomainCompartmentDetails;
const ChangeIotDomainDataRetentionPeriodDetails = __importStar(require("./change-iot-domain-data-retention-period-details"));
exports.ChangeIotDomainDataRetentionPeriodDetails = ChangeIotDomainDataRetentionPeriodDetails.ChangeIotDomainDataRetentionPeriodDetails;
const ChangeIotDomainGroupCompartmentDetails = __importStar(require("./change-iot-domain-group-compartment-details"));
exports.ChangeIotDomainGroupCompartmentDetails = ChangeIotDomainGroupCompartmentDetails.ChangeIotDomainGroupCompartmentDetails;
const ConfigureIotDomainDataAccessDetails = __importStar(require("./configure-iot-domain-data-access-details"));
exports.ConfigureIotDomainDataAccessDetails = ConfigureIotDomainDataAccessDetails.ConfigureIotDomainDataAccessDetails;
const ConfigureIotDomainGroupDataAccessDetails = __importStar(require("./configure-iot-domain-group-data-access-details"));
exports.ConfigureIotDomainGroupDataAccessDetails = ConfigureIotDomainGroupDataAccessDetails.ConfigureIotDomainGroupDataAccessDetails;
const CreateDigitalTwinAdapterDetails = __importStar(require("./create-digital-twin-adapter-details"));
exports.CreateDigitalTwinAdapterDetails = CreateDigitalTwinAdapterDetails.CreateDigitalTwinAdapterDetails;
const CreateDigitalTwinInstanceDetails = __importStar(require("./create-digital-twin-instance-details"));
exports.CreateDigitalTwinInstanceDetails = CreateDigitalTwinInstanceDetails.CreateDigitalTwinInstanceDetails;
const CreateDigitalTwinModelDetails = __importStar(require("./create-digital-twin-model-details"));
exports.CreateDigitalTwinModelDetails = CreateDigitalTwinModelDetails.CreateDigitalTwinModelDetails;
const CreateDigitalTwinRelationshipDetails = __importStar(require("./create-digital-twin-relationship-details"));
exports.CreateDigitalTwinRelationshipDetails = CreateDigitalTwinRelationshipDetails.CreateDigitalTwinRelationshipDetails;
const CreateIotDomainDetails = __importStar(require("./create-iot-domain-details"));
exports.CreateIotDomainDetails = CreateIotDomainDetails.CreateIotDomainDetails;
const CreateIotDomainGroupDetails = __importStar(require("./create-iot-domain-group-details"));
exports.CreateIotDomainGroupDetails = CreateIotDomainGroupDetails.CreateIotDomainGroupDetails;
const DataRetentionPeriodsInDays = __importStar(require("./data-retention-periods-in-days"));
exports.DataRetentionPeriodsInDays = DataRetentionPeriodsInDays.DataRetentionPeriodsInDays;
const DigitalTwinAdapter = __importStar(require("./digital-twin-adapter"));
exports.DigitalTwinAdapter = DigitalTwinAdapter.DigitalTwinAdapter;
const DigitalTwinAdapterCollection = __importStar(require("./digital-twin-adapter-collection"));
exports.DigitalTwinAdapterCollection = DigitalTwinAdapterCollection.DigitalTwinAdapterCollection;
const DigitalTwinAdapterEnvelopeMapping = __importStar(require("./digital-twin-adapter-envelope-mapping"));
exports.DigitalTwinAdapterEnvelopeMapping = DigitalTwinAdapterEnvelopeMapping.DigitalTwinAdapterEnvelopeMapping;
const DigitalTwinAdapterInboundEnvelope = __importStar(require("./digital-twin-adapter-inbound-envelope"));
exports.DigitalTwinAdapterInboundEnvelope = DigitalTwinAdapterInboundEnvelope.DigitalTwinAdapterInboundEnvelope;
const DigitalTwinAdapterInboundRoute = __importStar(require("./digital-twin-adapter-inbound-route"));
exports.DigitalTwinAdapterInboundRoute = DigitalTwinAdapterInboundRoute.DigitalTwinAdapterInboundRoute;
const DigitalTwinAdapterPayload = __importStar(require("./digital-twin-adapter-payload"));
exports.DigitalTwinAdapterPayload = DigitalTwinAdapterPayload.DigitalTwinAdapterPayload;
const DigitalTwinAdapterSummary = __importStar(require("./digital-twin-adapter-summary"));
exports.DigitalTwinAdapterSummary = DigitalTwinAdapterSummary.DigitalTwinAdapterSummary;
const DigitalTwinInstance = __importStar(require("./digital-twin-instance"));
exports.DigitalTwinInstance = DigitalTwinInstance.DigitalTwinInstance;
const DigitalTwinInstanceCollection = __importStar(require("./digital-twin-instance-collection"));
exports.DigitalTwinInstanceCollection = DigitalTwinInstanceCollection.DigitalTwinInstanceCollection;
const DigitalTwinInstanceSummary = __importStar(require("./digital-twin-instance-summary"));
exports.DigitalTwinInstanceSummary = DigitalTwinInstanceSummary.DigitalTwinInstanceSummary;
const DigitalTwinModel = __importStar(require("./digital-twin-model"));
exports.DigitalTwinModel = DigitalTwinModel.DigitalTwinModel;
const DigitalTwinModelCollection = __importStar(require("./digital-twin-model-collection"));
exports.DigitalTwinModelCollection = DigitalTwinModelCollection.DigitalTwinModelCollection;
const DigitalTwinModelSummary = __importStar(require("./digital-twin-model-summary"));
exports.DigitalTwinModelSummary = DigitalTwinModelSummary.DigitalTwinModelSummary;
const DigitalTwinRelationship = __importStar(require("./digital-twin-relationship"));
exports.DigitalTwinRelationship = DigitalTwinRelationship.DigitalTwinRelationship;
const DigitalTwinRelationshipCollection = __importStar(require("./digital-twin-relationship-collection"));
exports.DigitalTwinRelationshipCollection = DigitalTwinRelationshipCollection.DigitalTwinRelationshipCollection;
const DigitalTwinRelationshipSummary = __importStar(require("./digital-twin-relationship-summary"));
exports.DigitalTwinRelationshipSummary = DigitalTwinRelationshipSummary.DigitalTwinRelationshipSummary;
const ErrorDetails = __importStar(require("./error-details"));
exports.ErrorDetails = ErrorDetails.ErrorDetails;
const InvokeRawCommandDetails = __importStar(require("./invoke-raw-command-details"));
exports.InvokeRawCommandDetails = InvokeRawCommandDetails.InvokeRawCommandDetails;
const IotDomain = __importStar(require("./iot-domain"));
exports.IotDomain = IotDomain.IotDomain;
const IotDomainCollection = __importStar(require("./iot-domain-collection"));
exports.IotDomainCollection = IotDomainCollection.IotDomainCollection;
const IotDomainGroup = __importStar(require("./iot-domain-group"));
exports.IotDomainGroup = IotDomainGroup.IotDomainGroup;
const IotDomainGroupCollection = __importStar(require("./iot-domain-group-collection"));
exports.IotDomainGroupCollection = IotDomainGroupCollection.IotDomainGroupCollection;
const IotDomainGroupSummary = __importStar(require("./iot-domain-group-summary"));
exports.IotDomainGroupSummary = IotDomainGroupSummary.IotDomainGroupSummary;
const IotDomainSummary = __importStar(require("./iot-domain-summary"));
exports.IotDomainSummary = IotDomainSummary.IotDomainSummary;
const LifecycleState = __importStar(require("./lifecycle-state"));
exports.LifecycleState = LifecycleState.LifecycleState;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const UpdateDigitalTwinAdapterDetails = __importStar(require("./update-digital-twin-adapter-details"));
exports.UpdateDigitalTwinAdapterDetails = UpdateDigitalTwinAdapterDetails.UpdateDigitalTwinAdapterDetails;
const UpdateDigitalTwinInstanceDetails = __importStar(require("./update-digital-twin-instance-details"));
exports.UpdateDigitalTwinInstanceDetails = UpdateDigitalTwinInstanceDetails.UpdateDigitalTwinInstanceDetails;
const UpdateDigitalTwinModelDetails = __importStar(require("./update-digital-twin-model-details"));
exports.UpdateDigitalTwinModelDetails = UpdateDigitalTwinModelDetails.UpdateDigitalTwinModelDetails;
const UpdateDigitalTwinRelationshipDetails = __importStar(require("./update-digital-twin-relationship-details"));
exports.UpdateDigitalTwinRelationshipDetails = UpdateDigitalTwinRelationshipDetails.UpdateDigitalTwinRelationshipDetails;
const UpdateIotDomainDetails = __importStar(require("./update-iot-domain-details"));
exports.UpdateIotDomainDetails = UpdateIotDomainDetails.UpdateIotDomainDetails;
const UpdateIotDomainGroupDetails = __importStar(require("./update-iot-domain-group-details"));
exports.UpdateIotDomainGroupDetails = UpdateIotDomainGroupDetails.UpdateIotDomainGroupDetails;
const Vcn = __importStar(require("./vcn"));
exports.Vcn = Vcn.Vcn;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const ApexDataAccessDetails = __importStar(require("./apex-data-access-details"));
exports.ApexDataAccessDetails = ApexDataAccessDetails.ApexDataAccessDetails;
const DigitalTwinAdapterJsonPayload = __importStar(require("./digital-twin-adapter-json-payload"));
exports.DigitalTwinAdapterJsonPayload = DigitalTwinAdapterJsonPayload.DigitalTwinAdapterJsonPayload;
const DirectDataAccessDetails = __importStar(require("./direct-data-access-details"));
exports.DirectDataAccessDetails = DirectDataAccessDetails.DirectDataAccessDetails;
const InvokeRawBinaryCommandDetails = __importStar(require("./invoke-raw-binary-command-details"));
exports.InvokeRawBinaryCommandDetails = InvokeRawBinaryCommandDetails.InvokeRawBinaryCommandDetails;
const InvokeRawJsonCommandDetails = __importStar(require("./invoke-raw-json-command-details"));
exports.InvokeRawJsonCommandDetails = InvokeRawJsonCommandDetails.InvokeRawJsonCommandDetails;
const InvokeRawTextCommandDetails = __importStar(require("./invoke-raw-text-command-details"));
exports.InvokeRawTextCommandDetails = InvokeRawTextCommandDetails.InvokeRawTextCommandDetails;
const OrdsDataAccessDetails = __importStar(require("./ords-data-access-details"));
exports.OrdsDataAccessDetails = OrdsDataAccessDetails.OrdsDataAccessDetails;
