// oci-iot/lib/model/iot-domain.js
"use strict";
/**
 * Internet of Things API
 * Use the Internet of Things (IoT) API to manage IoT domain groups, domains, and digital twin resources including models, adapters, instances, and relationships.
For more information, see [Internet of Things](https://docs.oracle.com/iaas/Content/internet-of-things/home.htm).

 * OpenAPI spec version: 20250531
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.IotDomain = void 0;
const model = __importStar(require("../model"));
var IotDomain;
(function (IotDomain) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = IotDomain.LifecycleState || (IotDomain.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dataRetentionPeriodsInDays": obj.dataRetentionPeriodsInDays
                ? model.DataRetentionPeriodsInDays.getJsonObj(obj.dataRetentionPeriodsInDays)
                : undefined
        });
        return jsonObj;
    }
    IotDomain.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "dataRetentionPeriodsInDays": obj.dataRetentionPeriodsInDays
                ? model.DataRetentionPeriodsInDays.getDeserializedJsonObj(obj.dataRetentionPeriodsInDays)
                : undefined
        });
        return jsonObj;
    }
    IotDomain.getDeserializedJsonObj = getDeserializedJsonObj;
})(IotDomain = exports.IotDomain || (exports.IotDomain = {}));
