// oci-iot/lib/model/lifecycle-state.js
"use strict";
/**
 * Internet of Things API
 * Use the Internet of Things (IoT) API to manage IoT domain groups, domains, and digital twin resources including models, adapters, instances, and relationships.
For more information, see [Internet of Things](https://docs.oracle.com/iaas/Content/internet-of-things/home.htm).

 * OpenAPI spec version: 20250531
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LifecycleState = void 0;
/**
 * Possible lifecycle states.
 **/
var LifecycleState;
(function (LifecycleState) {
    LifecycleState["Active"] = "ACTIVE";
    LifecycleState["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(LifecycleState = exports.LifecycleState || (exports.LifecycleState = {}));
(function (LifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    LifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    LifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(LifecycleState = exports.LifecycleState || (exports.LifecycleState = {}));
