// oci-fleetappsmanagement/lib/request/list-recommended-patches-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListRecommendedPatchesRequest = void 0;
var ListRecommendedPatchesRequest;
(function (ListRecommendedPatchesRequest) {
    let PatchLevel;
    (function (PatchLevel) {
        PatchLevel["Latest"] = "LATEST";
        PatchLevel["LatestMinusOne"] = "LATEST_MINUS_ONE";
        PatchLevel["LatestMinusTwo"] = "LATEST_MINUS_TWO";
    })(PatchLevel = ListRecommendedPatchesRequest.PatchLevel || (ListRecommendedPatchesRequest.PatchLevel = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["PatchName"] = "patchName";
    })(SortBy = ListRecommendedPatchesRequest.SortBy || (ListRecommendedPatchesRequest.SortBy = {}));
})(ListRecommendedPatchesRequest = exports.ListRecommendedPatchesRequest || (exports.ListRecommendedPatchesRequest = {}));
