// oci-core/lib/model/instance-pool-pre-termination-action-handle-timeout-details.js
"use strict";
/**
 * Core Services API
 * Use the Core Services API to manage resources such as virtual cloud networks (VCNs),
compute instances, and block storage volumes. For more information, see the console
documentation for the [Networking](https://docs.oracle.com/iaas/Content/Network/Concepts/overview.htm),
[Compute](https://docs.oracle.com/iaas/Content/Compute/Concepts/computeoverview.htm), and
[Block Volume](https://docs.oracle.com/iaas/Content/Block/Concepts/overview.htm) services.
The required permissions are documented in the
[Details for the Core Services](https://docs.oracle.com/iaas/Content/Identity/Reference/corepolicyreference.htm) article.

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.InstancePoolPreTerminationActionHandleTimeoutDetails = void 0;
var InstancePoolPreTerminationActionHandleTimeoutDetails;
(function (InstancePoolPreTerminationActionHandleTimeoutDetails) {
    let PreserveBlockVolumeMode;
    (function (PreserveBlockVolumeMode) {
        PreserveBlockVolumeMode["PreserveAlways"] = "PRESERVE_ALWAYS";
        PreserveBlockVolumeMode["PreserveOnTimeout"] = "PRESERVE_ON_TIMEOUT";
        PreserveBlockVolumeMode["DeleteAlways"] = "DELETE_ALWAYS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PreserveBlockVolumeMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(PreserveBlockVolumeMode = InstancePoolPreTerminationActionHandleTimeoutDetails.PreserveBlockVolumeMode || (InstancePoolPreTerminationActionHandleTimeoutDetails.PreserveBlockVolumeMode = {}));
    let PreserveBootVolumeMode;
    (function (PreserveBootVolumeMode) {
        PreserveBootVolumeMode["PreserveAlways"] = "PRESERVE_ALWAYS";
        PreserveBootVolumeMode["PreserveOnTimeout"] = "PRESERVE_ON_TIMEOUT";
        PreserveBootVolumeMode["DeleteAlways"] = "DELETE_ALWAYS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        PreserveBootVolumeMode["UnknownValue"] = "UNKNOWN_VALUE";
    })(PreserveBootVolumeMode = InstancePoolPreTerminationActionHandleTimeoutDetails.PreserveBootVolumeMode || (InstancePoolPreTerminationActionHandleTimeoutDetails.PreserveBootVolumeMode = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    InstancePoolPreTerminationActionHandleTimeoutDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    InstancePoolPreTerminationActionHandleTimeoutDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(InstancePoolPreTerminationActionHandleTimeoutDetails = exports.InstancePoolPreTerminationActionHandleTimeoutDetails || (exports.InstancePoolPreTerminationActionHandleTimeoutDetails = {}));
