// oci-email/lib/model/email-outbound-ip-summary.js
"use strict";
/**
 * Email Delivery API
 * Use the Email Delivery API to do the necessary set up to send high-volume and application-generated emails through the OCI Email Delivery service.
For more information, see [Overview of the Email Delivery Service](https://docs.oracle.com/iaas/Content/Email/Concepts/overview.htm).

 **Note:** Write actions (POST, UPDATE, DELETE) may take several minutes to propagate and be reflected by the API.
 If a subsequent read request fails to reflect your changes, wait a few minutes and try again.

 * OpenAPI spec version: 20170907
 * Contact: email-dev_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailOutboundIpSummary = void 0;
var EmailOutboundIpSummary;
(function (EmailOutboundIpSummary) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Draining"] = "DRAINING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = EmailOutboundIpSummary.LifecycleState || (EmailOutboundIpSummary.LifecycleState = {}));
    let AssignmentState;
    (function (AssignmentState) {
        AssignmentState["Available"] = "AVAILABLE";
        AssignmentState["Assigned"] = "ASSIGNED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AssignmentState["UnknownValue"] = "UNKNOWN_VALUE";
    })(AssignmentState = EmailOutboundIpSummary.AssignmentState || (EmailOutboundIpSummary.AssignmentState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    EmailOutboundIpSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    EmailOutboundIpSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(EmailOutboundIpSummary = exports.EmailOutboundIpSummary || (exports.EmailOutboundIpSummary = {}));
