// oci-email/lib/model/index.js
"use strict";
/**
 * Email Delivery API
 * Use the Email Delivery API to do the necessary set up to send high-volume and application-generated emails through the OCI Email Delivery service.
For more information, see [Overview of the Email Delivery Service](https://docs.oracle.com/iaas/Content/Email/Concepts/overview.htm).

 **Note:** Write actions (POST, UPDATE, DELETE) may take several minutes to propagate and be reflected by the API.
 If a subsequent read request fails to reflect your changes, wait a few minutes and try again.

 * OpenAPI spec version: 20170907
 * Contact: email-dev_us_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateSenderDetails = exports.UpdateEmailReturnPathDetails = exports.UpdateEmailIpPoolDetails = exports.UpdateEmailDomainDetails = exports.UpdateDkimDetails = exports.SuppressionSummary = exports.SuppressionReason = exports.Suppression = exports.SortOrder = exports.SenderSummary = exports.Sender = exports.ResourceLock = exports.RemoveLockDetails = exports.RemoveEmailOutboundIpDetails = exports.OperationType = exports.OperationStatus = exports.EmailReturnPathSummary = exports.EmailReturnPathCollection = exports.EmailReturnPath = exports.EmailOutboundIpSummary = exports.EmailOutboundIpCollection = exports.EmailIpPoolSummary = exports.EmailIpPoolCollection = exports.EmailIpPool = exports.EmailDomainSummary = exports.EmailDomainCollection = exports.EmailDomain = exports.DomainVerificationStatusType = exports.DkimSummary = exports.DkimCollection = exports.Dkim = exports.CreateSuppressionDetails = exports.CreateSenderDetails = exports.CreateEmailReturnPathDetails = exports.CreateEmailIpPoolDetails = exports.CreateEmailDomainDetails = exports.CreateDkimDetails = exports.Configuration = exports.ChangeSenderCompartmentDetails = exports.ChangeEmailIpPoolCompartmentDetails = exports.ChangeEmailDomainCompartmentDetails = exports.AddLockDetails = exports.AddEmailOutboundIpDetails = exports.ActionType = void 0;
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AddEmailOutboundIpDetails = __importStar(require("./add-email-outbound-ip-details"));
exports.AddEmailOutboundIpDetails = AddEmailOutboundIpDetails.AddEmailOutboundIpDetails;
const AddLockDetails = __importStar(require("./add-lock-details"));
exports.AddLockDetails = AddLockDetails.AddLockDetails;
const ChangeEmailDomainCompartmentDetails = __importStar(require("./change-email-domain-compartment-details"));
exports.ChangeEmailDomainCompartmentDetails = ChangeEmailDomainCompartmentDetails.ChangeEmailDomainCompartmentDetails;
const ChangeEmailIpPoolCompartmentDetails = __importStar(require("./change-email-ip-pool-compartment-details"));
exports.ChangeEmailIpPoolCompartmentDetails = ChangeEmailIpPoolCompartmentDetails.ChangeEmailIpPoolCompartmentDetails;
const ChangeSenderCompartmentDetails = __importStar(require("./change-sender-compartment-details"));
exports.ChangeSenderCompartmentDetails = ChangeSenderCompartmentDetails.ChangeSenderCompartmentDetails;
const Configuration = __importStar(require("./configuration"));
exports.Configuration = Configuration.Configuration;
const CreateDkimDetails = __importStar(require("./create-dkim-details"));
exports.CreateDkimDetails = CreateDkimDetails.CreateDkimDetails;
const CreateEmailDomainDetails = __importStar(require("./create-email-domain-details"));
exports.CreateEmailDomainDetails = CreateEmailDomainDetails.CreateEmailDomainDetails;
const CreateEmailIpPoolDetails = __importStar(require("./create-email-ip-pool-details"));
exports.CreateEmailIpPoolDetails = CreateEmailIpPoolDetails.CreateEmailIpPoolDetails;
const CreateEmailReturnPathDetails = __importStar(require("./create-email-return-path-details"));
exports.CreateEmailReturnPathDetails = CreateEmailReturnPathDetails.CreateEmailReturnPathDetails;
const CreateSenderDetails = __importStar(require("./create-sender-details"));
exports.CreateSenderDetails = CreateSenderDetails.CreateSenderDetails;
const CreateSuppressionDetails = __importStar(require("./create-suppression-details"));
exports.CreateSuppressionDetails = CreateSuppressionDetails.CreateSuppressionDetails;
const Dkim = __importStar(require("./dkim"));
exports.Dkim = Dkim.Dkim;
const DkimCollection = __importStar(require("./dkim-collection"));
exports.DkimCollection = DkimCollection.DkimCollection;
const DkimSummary = __importStar(require("./dkim-summary"));
exports.DkimSummary = DkimSummary.DkimSummary;
const DomainVerificationStatusType = __importStar(require("./domain-verification-status-type"));
exports.DomainVerificationStatusType = DomainVerificationStatusType.DomainVerificationStatusType;
const EmailDomain = __importStar(require("./email-domain"));
exports.EmailDomain = EmailDomain.EmailDomain;
const EmailDomainCollection = __importStar(require("./email-domain-collection"));
exports.EmailDomainCollection = EmailDomainCollection.EmailDomainCollection;
const EmailDomainSummary = __importStar(require("./email-domain-summary"));
exports.EmailDomainSummary = EmailDomainSummary.EmailDomainSummary;
const EmailIpPool = __importStar(require("./email-ip-pool"));
exports.EmailIpPool = EmailIpPool.EmailIpPool;
const EmailIpPoolCollection = __importStar(require("./email-ip-pool-collection"));
exports.EmailIpPoolCollection = EmailIpPoolCollection.EmailIpPoolCollection;
const EmailIpPoolSummary = __importStar(require("./email-ip-pool-summary"));
exports.EmailIpPoolSummary = EmailIpPoolSummary.EmailIpPoolSummary;
const EmailOutboundIpCollection = __importStar(require("./email-outbound-ip-collection"));
exports.EmailOutboundIpCollection = EmailOutboundIpCollection.EmailOutboundIpCollection;
const EmailOutboundIpSummary = __importStar(require("./email-outbound-ip-summary"));
exports.EmailOutboundIpSummary = EmailOutboundIpSummary.EmailOutboundIpSummary;
const EmailReturnPath = __importStar(require("./email-return-path"));
exports.EmailReturnPath = EmailReturnPath.EmailReturnPath;
const EmailReturnPathCollection = __importStar(require("./email-return-path-collection"));
exports.EmailReturnPathCollection = EmailReturnPathCollection.EmailReturnPathCollection;
const EmailReturnPathSummary = __importStar(require("./email-return-path-summary"));
exports.EmailReturnPathSummary = EmailReturnPathSummary.EmailReturnPathSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const RemoveEmailOutboundIpDetails = __importStar(require("./remove-email-outbound-ip-details"));
exports.RemoveEmailOutboundIpDetails = RemoveEmailOutboundIpDetails.RemoveEmailOutboundIpDetails;
const RemoveLockDetails = __importStar(require("./remove-lock-details"));
exports.RemoveLockDetails = RemoveLockDetails.RemoveLockDetails;
const ResourceLock = __importStar(require("./resource-lock"));
exports.ResourceLock = ResourceLock.ResourceLock;
const Sender = __importStar(require("./sender"));
exports.Sender = Sender.Sender;
const SenderSummary = __importStar(require("./sender-summary"));
exports.SenderSummary = SenderSummary.SenderSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const Suppression = __importStar(require("./suppression"));
exports.Suppression = Suppression.Suppression;
const SuppressionReason = __importStar(require("./suppression-reason"));
exports.SuppressionReason = SuppressionReason.SuppressionReason;
const SuppressionSummary = __importStar(require("./suppression-summary"));
exports.SuppressionSummary = SuppressionSummary.SuppressionSummary;
const UpdateDkimDetails = __importStar(require("./update-dkim-details"));
exports.UpdateDkimDetails = UpdateDkimDetails.UpdateDkimDetails;
const UpdateEmailDomainDetails = __importStar(require("./update-email-domain-details"));
exports.UpdateEmailDomainDetails = UpdateEmailDomainDetails.UpdateEmailDomainDetails;
const UpdateEmailIpPoolDetails = __importStar(require("./update-email-ip-pool-details"));
exports.UpdateEmailIpPoolDetails = UpdateEmailIpPoolDetails.UpdateEmailIpPoolDetails;
const UpdateEmailReturnPathDetails = __importStar(require("./update-email-return-path-details"));
exports.UpdateEmailReturnPathDetails = UpdateEmailReturnPathDetails.UpdateEmailReturnPathDetails;
const UpdateSenderDetails = __importStar(require("./update-sender-details"));
exports.UpdateSenderDetails = UpdateSenderDetails.UpdateSenderDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
