// oci-marketplacepublisher/lib/request/list-customer-instance-report-records-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20241201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListCustomerInstanceReportRecordsRequest = void 0;
var ListCustomerInstanceReportRecordsRequest;
(function (ListCustomerInstanceReportRecordsRequest) {
    let DateRange;
    (function (DateRange) {
        DateRange["ThisWeek"] = "THIS_WEEK";
        DateRange["LastWeek"] = "LAST_WEEK";
        DateRange["ThisMonth"] = "THIS_MONTH";
        DateRange["LastMonth"] = "LAST_MONTH";
        DateRange["LastThreeMonths"] = "LAST_THREE_MONTHS";
    })(DateRange = ListCustomerInstanceReportRecordsRequest.DateRange || (ListCustomerInstanceReportRecordsRequest.DateRange = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListCustomerInstanceReportRecordsRequest.SortBy || (ListCustomerInstanceReportRecordsRequest.SortBy = {}));
})(ListCustomerInstanceReportRecordsRequest = exports.ListCustomerInstanceReportRecordsRequest || (exports.ListCustomerInstanceReportRecordsRequest = {}));
