// oci-mysql/lib/model/backup-validation-details.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackupValidationDetails = void 0;
const model = __importStar(require("../model"));
var BackupValidationDetails;
(function (BackupValidationDetails) {
    let ValidationStatus;
    (function (ValidationStatus) {
        ValidationStatus["NotValidated"] = "NOT_VALIDATED";
        ValidationStatus["Validated"] = "VALIDATED";
        ValidationStatus["NeedsAttention"] = "NEEDS_ATTENTION";
        ValidationStatus["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ValidationStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(ValidationStatus = BackupValidationDetails.ValidationStatus || (BackupValidationDetails.ValidationStatus = {}));
    let BackupPreparationStatus;
    (function (BackupPreparationStatus) {
        BackupPreparationStatus["Prepared"] = "PREPARED";
        BackupPreparationStatus["NotPrepared"] = "NOT_PREPARED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        BackupPreparationStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(BackupPreparationStatus = BackupValidationDetails.BackupPreparationStatus || (BackupValidationDetails.BackupPreparationStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "preparedBackupDetails": obj.preparedBackupDetails
                ? model.PreparedBackupDetails.getJsonObj(obj.preparedBackupDetails)
                : undefined
        });
        return jsonObj;
    }
    BackupValidationDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "preparedBackupDetails": obj.preparedBackupDetails
                ? model.PreparedBackupDetails.getDeserializedJsonObj(obj.preparedBackupDetails)
                : undefined
        });
        return jsonObj;
    }
    BackupValidationDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(BackupValidationDetails = exports.BackupValidationDetails || (exports.BackupValidationDetails = {}));
