// oci-mysql/lib/model/maintenance-type.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenanceType = void 0;
/**
 * How the maintenance event was triggered.
* <p>
AUTOMATIC: maintenance event triggered as part of scheduled maintenance.
* MANUAL:    maintenance event triggered manually.
* SHAPE:     maintenance event triggered by a shape update.
*
**/
var MaintenanceType;
(function (MaintenanceType) {
    MaintenanceType["Automatic"] = "AUTOMATIC";
    MaintenanceType["Manual"] = "MANUAL";
    MaintenanceType["Shape"] = "SHAPE";
})(MaintenanceType = exports.MaintenanceType || (exports.MaintenanceType = {}));
(function (MaintenanceType) {
    function getJsonObj(obj) {
        return obj;
    }
    MaintenanceType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MaintenanceType.getDeserializedJsonObj = getDeserializedJsonObj;
})(MaintenanceType = exports.MaintenanceType || (exports.MaintenanceType = {}));
