// oci-mysql/lib/model/maintenance-schedule-type.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MaintenanceScheduleType = void 0;
/**
 * The maintenance schedule type of the DB system.
* <p>
EARLY: Maintenance schedule follows a cycle where upgrades are performed when versions become deprecated.
* REGULAR: Maintenance schedule follows the normal cycle where upgrades are performed when versions become unavailable.
*
**/
var MaintenanceScheduleType;
(function (MaintenanceScheduleType) {
    MaintenanceScheduleType["Early"] = "EARLY";
    MaintenanceScheduleType["Regular"] = "REGULAR";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    MaintenanceScheduleType["UnknownValue"] = "UNKNOWN_VALUE";
})(MaintenanceScheduleType = exports.MaintenanceScheduleType || (exports.MaintenanceScheduleType = {}));
(function (MaintenanceScheduleType) {
    function getJsonObj(obj) {
        return obj;
    }
    MaintenanceScheduleType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    MaintenanceScheduleType.getDeserializedJsonObj = getDeserializedJsonObj;
})(MaintenanceScheduleType = exports.MaintenanceScheduleType || (exports.MaintenanceScheduleType = {}));
