// oci-mysql/lib/model/version-preference.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VersionPreference = void 0;
/**
 * The preferred version to target when performing an automatic MySQL upgrade.
* <p>
OLDEST: Choose the oldest available MySQL version based on the current version of the DB System.
* SECOND_NEWEST: Choose the MySQL version before the newest for auto-upgrade.
* NEWEST: Choose the latest and greatest MySQL version available for auto-upgrade.
*
**/
var VersionPreference;
(function (VersionPreference) {
    VersionPreference["Oldest"] = "OLDEST";
    VersionPreference["SecondNewest"] = "SECOND_NEWEST";
    VersionPreference["Newest"] = "NEWEST";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    VersionPreference["UnknownValue"] = "UNKNOWN_VALUE";
})(VersionPreference = exports.VersionPreference || (exports.VersionPreference = {}));
(function (VersionPreference) {
    function getJsonObj(obj) {
        return obj;
    }
    VersionPreference.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    VersionPreference.getDeserializedJsonObj = getDeserializedJsonObj;
})(VersionPreference = exports.VersionPreference || (exports.VersionPreference = {}));
