// oci-mysql/lib/model/version-track-preference.js
"use strict";
/**
 * MySQL Database Service API
 * The API for the MySQL Database Service
 * OpenAPI spec version: 20190415
 * Contact: mysql-cloud-dev_ww_grp@oracle.com
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VersionTrackPreference = void 0;
/**
 * The preferred version track to target when performing an automatic MySQL upgrade.
 * LONG_TERM_SUPPORT: No MySQL database behavior changes.
 * INNOVATION: Provides access to the latest features and all bug fixes.
 * FOLLOW: Follows the track of the DB System's MySQL version.
 *
 **/
var VersionTrackPreference;
(function (VersionTrackPreference) {
    VersionTrackPreference["LongTermSupport"] = "LONG_TERM_SUPPORT";
    VersionTrackPreference["Innovation"] = "INNOVATION";
    VersionTrackPreference["Follow"] = "FOLLOW";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    VersionTrackPreference["UnknownValue"] = "UNKNOWN_VALUE";
})(VersionTrackPreference = exports.VersionTrackPreference || (exports.VersionTrackPreference = {}));
(function (VersionTrackPreference) {
    function getJsonObj(obj) {
        return obj;
    }
    VersionTrackPreference.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    VersionTrackPreference.getDeserializedJsonObj = getDeserializedJsonObj;
})(VersionTrackPreference = exports.VersionTrackPreference || (exports.VersionTrackPreference = {}));
