// oci-managedkafka/lib/model/index.js
"use strict";
/**
 * Oracle Streaming with Apache Kafka (OSAK) API
 * Use Oracle Streaming with Apache Kafka Control Plane API to create/update/delete managed Kafka clusters.
 * OpenAPI spec version: 20240901
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateKafkaClusterDetails = exports.UpdateKafkaClusterConfigDetails = exports.SubnetSet = exports.SortOrder = exports.OperationType = exports.OperationStatus = exports.KafkaClusterSummary = exports.KafkaClusterConfigVersionSummary = exports.KafkaClusterConfigVersionCollection = exports.KafkaClusterConfigVersion = exports.KafkaClusterConfigSummary = exports.KafkaClusterConfigCollection = exports.KafkaClusterConfig = exports.KafkaClusterCollection = exports.KafkaCluster = exports.EnableSuperuserDetails = exports.CreateKafkaClusterDetails = exports.CreateKafkaClusterConfigDetails = exports.ChangeKafkaClusterConfigCompartmentDetails = exports.ChangeKafkaClusterCompartmentDetails = exports.BrokerShape = exports.BootstrapUrl = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const BootstrapUrl = __importStar(require("./bootstrap-url"));
exports.BootstrapUrl = BootstrapUrl.BootstrapUrl;
const BrokerShape = __importStar(require("./broker-shape"));
exports.BrokerShape = BrokerShape.BrokerShape;
const ChangeKafkaClusterCompartmentDetails = __importStar(require("./change-kafka-cluster-compartment-details"));
exports.ChangeKafkaClusterCompartmentDetails = ChangeKafkaClusterCompartmentDetails.ChangeKafkaClusterCompartmentDetails;
const ChangeKafkaClusterConfigCompartmentDetails = __importStar(require("./change-kafka-cluster-config-compartment-details"));
exports.ChangeKafkaClusterConfigCompartmentDetails = ChangeKafkaClusterConfigCompartmentDetails.ChangeKafkaClusterConfigCompartmentDetails;
const CreateKafkaClusterConfigDetails = __importStar(require("./create-kafka-cluster-config-details"));
exports.CreateKafkaClusterConfigDetails = CreateKafkaClusterConfigDetails.CreateKafkaClusterConfigDetails;
const CreateKafkaClusterDetails = __importStar(require("./create-kafka-cluster-details"));
exports.CreateKafkaClusterDetails = CreateKafkaClusterDetails.CreateKafkaClusterDetails;
const EnableSuperuserDetails = __importStar(require("./enable-superuser-details"));
exports.EnableSuperuserDetails = EnableSuperuserDetails.EnableSuperuserDetails;
const KafkaCluster = __importStar(require("./kafka-cluster"));
exports.KafkaCluster = KafkaCluster.KafkaCluster;
const KafkaClusterCollection = __importStar(require("./kafka-cluster-collection"));
exports.KafkaClusterCollection = KafkaClusterCollection.KafkaClusterCollection;
const KafkaClusterConfig = __importStar(require("./kafka-cluster-config"));
exports.KafkaClusterConfig = KafkaClusterConfig.KafkaClusterConfig;
const KafkaClusterConfigCollection = __importStar(require("./kafka-cluster-config-collection"));
exports.KafkaClusterConfigCollection = KafkaClusterConfigCollection.KafkaClusterConfigCollection;
const KafkaClusterConfigSummary = __importStar(require("./kafka-cluster-config-summary"));
exports.KafkaClusterConfigSummary = KafkaClusterConfigSummary.KafkaClusterConfigSummary;
const KafkaClusterConfigVersion = __importStar(require("./kafka-cluster-config-version"));
exports.KafkaClusterConfigVersion = KafkaClusterConfigVersion.KafkaClusterConfigVersion;
const KafkaClusterConfigVersionCollection = __importStar(require("./kafka-cluster-config-version-collection"));
exports.KafkaClusterConfigVersionCollection = KafkaClusterConfigVersionCollection.KafkaClusterConfigVersionCollection;
const KafkaClusterConfigVersionSummary = __importStar(require("./kafka-cluster-config-version-summary"));
exports.KafkaClusterConfigVersionSummary = KafkaClusterConfigVersionSummary.KafkaClusterConfigVersionSummary;
const KafkaClusterSummary = __importStar(require("./kafka-cluster-summary"));
exports.KafkaClusterSummary = KafkaClusterSummary.KafkaClusterSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const SubnetSet = __importStar(require("./subnet-set"));
exports.SubnetSet = SubnetSet.SubnetSet;
const UpdateKafkaClusterConfigDetails = __importStar(require("./update-kafka-cluster-config-details"));
exports.UpdateKafkaClusterConfigDetails = UpdateKafkaClusterConfigDetails.UpdateKafkaClusterConfigDetails;
const UpdateKafkaClusterDetails = __importStar(require("./update-kafka-cluster-details"));
exports.UpdateKafkaClusterDetails = UpdateKafkaClusterDetails.UpdateKafkaClusterDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
