// oci-marketplacepublisher/lib/model/geo-location.js
"use strict";
/**
 * MarketplacePublisherService API
 * Use the Marketplace Publisher API to manage the publishing of applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20241201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GeoLocation = void 0;
var GeoLocation;
(function (GeoLocation) {
    let GeographicRegion;
    (function (GeographicRegion) {
        GeographicRegion["Africa"] = "AFRICA";
        GeographicRegion["AsiaPacific"] = "ASIA_PACIFIC";
        GeographicRegion["Europe"] = "EUROPE";
        GeographicRegion["Japan"] = "JAPAN";
        GeographicRegion["LatinAmerica"] = "LATIN_AMERICA";
        GeographicRegion["MiddleEast"] = "MIDDLE_EAST";
        GeographicRegion["NorthAmerica"] = "NORTH_AMERICA";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        GeographicRegion["UnknownValue"] = "UNKNOWN_VALUE";
    })(GeographicRegion = GeoLocation.GeographicRegion || (GeoLocation.GeographicRegion = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    GeoLocation.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    GeoLocation.getDeserializedJsonObj = getDeserializedJsonObj;
})(GeoLocation = exports.GeoLocation || (exports.GeoLocation = {}));
