// oci-marketplacepublisher/lib/model/index.js
"use strict";
/**
 * MarketplacePublisherService API
 * Use the Marketplace Publisher API to manage the publishing of applications in Oracle Cloud Infrastructure Marketplace.
 * OpenAPI spec version: 20241201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingRevision = exports.ListingProductAdditionalFilter = exports.ListingProduct = exports.ListingPart = exports.ListingMeter = exports.ListingCollection = exports.Listing = exports.LeadSummary = exports.LeadCollection = exports.Lead = exports.LanguageItem = exports.ImageShapeCompatibility = exports.ImageOcpuConstraints = exports.ImageMemoryConstraints = exports.HelmChartImageDetails = exports.GeoLocation = exports.FilterProperty = exports.DownloadInfo = exports.DisbursementReportRecordCollection = exports.DisbursementReportRecord = exports.CustomerInstanceReportRecordCollection = exports.CustomerInstanceReportRecord = exports.CreateVideoDetails = exports.CreateTermDetails = exports.CreateStackDetails = exports.CreateMachineImageDetails = exports.CreateListingRevisionPackageDetails = exports.CreateListingRevisionNoteDetails = exports.CreateListingRevisionDetails = exports.CreateListingRevisionAttachmentDetails = exports.CreateListingDetails = exports.CreateHelmChartImageDetails = exports.CreateContainerImageDetails = exports.CreateArtifactDetails = exports.ContainerImageDetails = exports.ChangeTermCompartmentDetails = exports.ChangeListingCompartmentDetails = exports.ChangeArtifactCompartmentDetails = exports.CategorySummary = exports.CategoryCollection = exports.Category = exports.AvailableServiceSummary = exports.AvailableServiceCollection = exports.ArtifactTypeEnum = exports.ArtifactSummary = exports.ArtifactCollection = exports.Artifact = exports.AdditionalInfo = exports.AdditionalFilter = exports.ActionType = void 0;
exports.TermCollection = exports.Term = exports.SupportedShapeSummary = exports.SupportedShapeCollection = exports.SupportedCurrencySummary = exports.SupportedCurrencyCollection = exports.SupportDocSummary = exports.SupportDocCollection = exports.SupportDoc = exports.SupportContact = exports.SubmitListingRevisionForReviewDetails = exports.StackArtifactDetails = exports.SortOrder = exports.ShapeOcpuOptions = exports.ShapeMemoryOptions = exports.PublisherSummary = exports.PublisherCollection = exports.Publisher = exports.PublishListingRevisionAsPrivateDetails = exports.PublicationStatus = exports.ProductSummary = exports.ProductCollection = exports.Product = exports.PrivateOfferAccountDetails = exports.PricingRate = exports.PricingPlan = exports.PackageType = exports.OpnMembership = exports.OperationType = exports.OperationStatus = exports.NamedLink = exports.MarketSummary = exports.MarketCollection = exports.Market = exports.MachineImagePackageDetails = exports.MachineImageDetails = exports.ListingType = exports.ListingSummary = exports.ListingRevisionSummary = exports.ListingRevisionPackageSummary = exports.ListingRevisionPackageCollection = exports.ListingRevisionPackage = exports.ListingRevisionNoteSummary = exports.ListingRevisionNoteCollection = exports.ListingRevisionNote = exports.ListingRevisionIconAttachment = exports.ListingRevisionCollection = exports.ListingRevisionAttachmentSummary = exports.ListingRevisionAttachmentCollection = exports.ListingRevisionAttachment = void 0;
exports.MachineImageArtifact = exports.LeadGenListingRevision = exports.KubernetesImageArtifact = exports.HelmChartPackage = exports.CustomerSuccessAttachment = exports.CreateVideoAttachmentDetails = exports.CreateSupportedServiceAttachment = exports.CreateStackArtifactDetails = exports.CreateServiceListingRevisionDetails = exports.CreateScreenShotAttachmentDetails = exports.CreateReviewSupportDocumentAttachment = exports.CreateRelatedDocumentAttachmentDetails = exports.CreateOciListingRevisionDetails = exports.CreateMachineImageArtifactDetails = exports.CreateLeadGenListingRevisionDetails = exports.CreateKubernetesImageArtifactDetails = exports.CreateCustomerSuccessAttachment = exports.CreateContainerImageArtifactDetails = exports.ContainerPackage = exports.ContainerImageArtifact = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.VersionDetails = exports.ValidationStatus = exports.UploadData = exports.UpdateTermVersionDetails = exports.UpdateTermDetails = exports.UpdateStackDetails = exports.UpdateMachineImageDetails = exports.UpdateListingRevisionPackageDetails = exports.UpdateListingRevisionNoteDetails = exports.UpdateListingRevisionDetails = exports.UpdateListingRevisionAttachmentDetails = exports.UpdateListingDetails = exports.UpdateHelmChartImageDetails = exports.UpdateContainerImageDetails = exports.UpdateArtifactDetails = exports.TrainedProfessionals = exports.TermVersionSummary = exports.TermVersionCollection = exports.TermVersionAttachment = exports.TermVersion = exports.TermSummary = void 0;
exports.VideoAttachment = exports.UpdateVideoAttachmentDetails = exports.UpdateSupportedServiceAttachment = exports.UpdateStackArtifactDetails = exports.UpdateServiceListingRevisionDetails = exports.UpdateScreenShotAttachmentDetails = exports.UpdateReviewSupportDocumentAttachment = exports.UpdateRelatedDocumentAttachmentDetails = exports.UpdateOciListingRevisionDetails = exports.UpdateMachineImageArtifactDetails = exports.UpdateLeadGenListingRevisionDetails = exports.UpdateKubernetesImageArtifactDetails = exports.UpdateCustomerSuccessAttachment = exports.UpdateContainerImageArtifactDetails = exports.SupportedServiceAttachment = exports.StackPackage = exports.StackArtifact = exports.ServiceListingRevision = exports.ScreenShotAttachment = exports.SaaSPricingPlan = exports.ReviewSupportDocumentAttachment = exports.RelatedDocumentAttachment = exports.OciListingRevision = exports.MeteredPricingPlan = exports.MachineImagePackage = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AdditionalFilter = __importStar(require("./additional-filter"));
exports.AdditionalFilter = AdditionalFilter.AdditionalFilter;
const AdditionalInfo = __importStar(require("./additional-info"));
exports.AdditionalInfo = AdditionalInfo.AdditionalInfo;
const Artifact = __importStar(require("./artifact"));
exports.Artifact = Artifact.Artifact;
const ArtifactCollection = __importStar(require("./artifact-collection"));
exports.ArtifactCollection = ArtifactCollection.ArtifactCollection;
const ArtifactSummary = __importStar(require("./artifact-summary"));
exports.ArtifactSummary = ArtifactSummary.ArtifactSummary;
const ArtifactTypeEnum = __importStar(require("./artifact-type-enum"));
exports.ArtifactTypeEnum = ArtifactTypeEnum.ArtifactTypeEnum;
const AvailableServiceCollection = __importStar(require("./available-service-collection"));
exports.AvailableServiceCollection = AvailableServiceCollection.AvailableServiceCollection;
const AvailableServiceSummary = __importStar(require("./available-service-summary"));
exports.AvailableServiceSummary = AvailableServiceSummary.AvailableServiceSummary;
const Category = __importStar(require("./category"));
exports.Category = Category.Category;
const CategoryCollection = __importStar(require("./category-collection"));
exports.CategoryCollection = CategoryCollection.CategoryCollection;
const CategorySummary = __importStar(require("./category-summary"));
exports.CategorySummary = CategorySummary.CategorySummary;
const ChangeArtifactCompartmentDetails = __importStar(require("./change-artifact-compartment-details"));
exports.ChangeArtifactCompartmentDetails = ChangeArtifactCompartmentDetails.ChangeArtifactCompartmentDetails;
const ChangeListingCompartmentDetails = __importStar(require("./change-listing-compartment-details"));
exports.ChangeListingCompartmentDetails = ChangeListingCompartmentDetails.ChangeListingCompartmentDetails;
const ChangeTermCompartmentDetails = __importStar(require("./change-term-compartment-details"));
exports.ChangeTermCompartmentDetails = ChangeTermCompartmentDetails.ChangeTermCompartmentDetails;
const ContainerImageDetails = __importStar(require("./container-image-details"));
exports.ContainerImageDetails = ContainerImageDetails.ContainerImageDetails;
const CreateArtifactDetails = __importStar(require("./create-artifact-details"));
exports.CreateArtifactDetails = CreateArtifactDetails.CreateArtifactDetails;
const CreateContainerImageDetails = __importStar(require("./create-container-image-details"));
exports.CreateContainerImageDetails = CreateContainerImageDetails.CreateContainerImageDetails;
const CreateHelmChartImageDetails = __importStar(require("./create-helm-chart-image-details"));
exports.CreateHelmChartImageDetails = CreateHelmChartImageDetails.CreateHelmChartImageDetails;
const CreateListingDetails = __importStar(require("./create-listing-details"));
exports.CreateListingDetails = CreateListingDetails.CreateListingDetails;
const CreateListingRevisionAttachmentDetails = __importStar(require("./create-listing-revision-attachment-details"));
exports.CreateListingRevisionAttachmentDetails = CreateListingRevisionAttachmentDetails.CreateListingRevisionAttachmentDetails;
const CreateListingRevisionDetails = __importStar(require("./create-listing-revision-details"));
exports.CreateListingRevisionDetails = CreateListingRevisionDetails.CreateListingRevisionDetails;
const CreateListingRevisionNoteDetails = __importStar(require("./create-listing-revision-note-details"));
exports.CreateListingRevisionNoteDetails = CreateListingRevisionNoteDetails.CreateListingRevisionNoteDetails;
const CreateListingRevisionPackageDetails = __importStar(require("./create-listing-revision-package-details"));
exports.CreateListingRevisionPackageDetails = CreateListingRevisionPackageDetails.CreateListingRevisionPackageDetails;
const CreateMachineImageDetails = __importStar(require("./create-machine-image-details"));
exports.CreateMachineImageDetails = CreateMachineImageDetails.CreateMachineImageDetails;
const CreateStackDetails = __importStar(require("./create-stack-details"));
exports.CreateStackDetails = CreateStackDetails.CreateStackDetails;
const CreateTermDetails = __importStar(require("./create-term-details"));
exports.CreateTermDetails = CreateTermDetails.CreateTermDetails;
const CreateVideoDetails = __importStar(require("./create-video-details"));
exports.CreateVideoDetails = CreateVideoDetails.CreateVideoDetails;
const CustomerInstanceReportRecord = __importStar(require("./customer-instance-report-record"));
exports.CustomerInstanceReportRecord = CustomerInstanceReportRecord.CustomerInstanceReportRecord;
const CustomerInstanceReportRecordCollection = __importStar(require("./customer-instance-report-record-collection"));
exports.CustomerInstanceReportRecordCollection = CustomerInstanceReportRecordCollection.CustomerInstanceReportRecordCollection;
const DisbursementReportRecord = __importStar(require("./disbursement-report-record"));
exports.DisbursementReportRecord = DisbursementReportRecord.DisbursementReportRecord;
const DisbursementReportRecordCollection = __importStar(require("./disbursement-report-record-collection"));
exports.DisbursementReportRecordCollection = DisbursementReportRecordCollection.DisbursementReportRecordCollection;
const DownloadInfo = __importStar(require("./download-info"));
exports.DownloadInfo = DownloadInfo.DownloadInfo;
const FilterProperty = __importStar(require("./filter-property"));
exports.FilterProperty = FilterProperty.FilterProperty;
const GeoLocation = __importStar(require("./geo-location"));
exports.GeoLocation = GeoLocation.GeoLocation;
const HelmChartImageDetails = __importStar(require("./helm-chart-image-details"));
exports.HelmChartImageDetails = HelmChartImageDetails.HelmChartImageDetails;
const ImageMemoryConstraints = __importStar(require("./image-memory-constraints"));
exports.ImageMemoryConstraints = ImageMemoryConstraints.ImageMemoryConstraints;
const ImageOcpuConstraints = __importStar(require("./image-ocpu-constraints"));
exports.ImageOcpuConstraints = ImageOcpuConstraints.ImageOcpuConstraints;
const ImageShapeCompatibility = __importStar(require("./image-shape-compatibility"));
exports.ImageShapeCompatibility = ImageShapeCompatibility.ImageShapeCompatibility;
const LanguageItem = __importStar(require("./language-item"));
exports.LanguageItem = LanguageItem.LanguageItem;
const Lead = __importStar(require("./lead"));
exports.Lead = Lead.Lead;
const LeadCollection = __importStar(require("./lead-collection"));
exports.LeadCollection = LeadCollection.LeadCollection;
const LeadSummary = __importStar(require("./lead-summary"));
exports.LeadSummary = LeadSummary.LeadSummary;
const Listing = __importStar(require("./listing"));
exports.Listing = Listing.Listing;
const ListingCollection = __importStar(require("./listing-collection"));
exports.ListingCollection = ListingCollection.ListingCollection;
const ListingMeter = __importStar(require("./listing-meter"));
exports.ListingMeter = ListingMeter.ListingMeter;
const ListingPart = __importStar(require("./listing-part"));
exports.ListingPart = ListingPart.ListingPart;
const ListingProduct = __importStar(require("./listing-product"));
exports.ListingProduct = ListingProduct.ListingProduct;
const ListingProductAdditionalFilter = __importStar(require("./listing-product-additional-filter"));
exports.ListingProductAdditionalFilter = ListingProductAdditionalFilter.ListingProductAdditionalFilter;
const ListingRevision = __importStar(require("./listing-revision"));
exports.ListingRevision = ListingRevision.ListingRevision;
const ListingRevisionAttachment = __importStar(require("./listing-revision-attachment"));
exports.ListingRevisionAttachment = ListingRevisionAttachment.ListingRevisionAttachment;
const ListingRevisionAttachmentCollection = __importStar(require("./listing-revision-attachment-collection"));
exports.ListingRevisionAttachmentCollection = ListingRevisionAttachmentCollection.ListingRevisionAttachmentCollection;
const ListingRevisionAttachmentSummary = __importStar(require("./listing-revision-attachment-summary"));
exports.ListingRevisionAttachmentSummary = ListingRevisionAttachmentSummary.ListingRevisionAttachmentSummary;
const ListingRevisionCollection = __importStar(require("./listing-revision-collection"));
exports.ListingRevisionCollection = ListingRevisionCollection.ListingRevisionCollection;
const ListingRevisionIconAttachment = __importStar(require("./listing-revision-icon-attachment"));
exports.ListingRevisionIconAttachment = ListingRevisionIconAttachment.ListingRevisionIconAttachment;
const ListingRevisionNote = __importStar(require("./listing-revision-note"));
exports.ListingRevisionNote = ListingRevisionNote.ListingRevisionNote;
const ListingRevisionNoteCollection = __importStar(require("./listing-revision-note-collection"));
exports.ListingRevisionNoteCollection = ListingRevisionNoteCollection.ListingRevisionNoteCollection;
const ListingRevisionNoteSummary = __importStar(require("./listing-revision-note-summary"));
exports.ListingRevisionNoteSummary = ListingRevisionNoteSummary.ListingRevisionNoteSummary;
const ListingRevisionPackage = __importStar(require("./listing-revision-package"));
exports.ListingRevisionPackage = ListingRevisionPackage.ListingRevisionPackage;
const ListingRevisionPackageCollection = __importStar(require("./listing-revision-package-collection"));
exports.ListingRevisionPackageCollection = ListingRevisionPackageCollection.ListingRevisionPackageCollection;
const ListingRevisionPackageSummary = __importStar(require("./listing-revision-package-summary"));
exports.ListingRevisionPackageSummary = ListingRevisionPackageSummary.ListingRevisionPackageSummary;
const ListingRevisionSummary = __importStar(require("./listing-revision-summary"));
exports.ListingRevisionSummary = ListingRevisionSummary.ListingRevisionSummary;
const ListingSummary = __importStar(require("./listing-summary"));
exports.ListingSummary = ListingSummary.ListingSummary;
const ListingType = __importStar(require("./listing-type"));
exports.ListingType = ListingType.ListingType;
const MachineImageDetails = __importStar(require("./machine-image-details"));
exports.MachineImageDetails = MachineImageDetails.MachineImageDetails;
const MachineImagePackageDetails = __importStar(require("./machine-image-package-details"));
exports.MachineImagePackageDetails = MachineImagePackageDetails.MachineImagePackageDetails;
const Market = __importStar(require("./market"));
exports.Market = Market.Market;
const MarketCollection = __importStar(require("./market-collection"));
exports.MarketCollection = MarketCollection.MarketCollection;
const MarketSummary = __importStar(require("./market-summary"));
exports.MarketSummary = MarketSummary.MarketSummary;
const NamedLink = __importStar(require("./named-link"));
exports.NamedLink = NamedLink.NamedLink;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const OpnMembership = __importStar(require("./opn-membership"));
exports.OpnMembership = OpnMembership.OpnMembership;
const PackageType = __importStar(require("./package-type"));
exports.PackageType = PackageType.PackageType;
const PricingPlan = __importStar(require("./pricing-plan"));
exports.PricingPlan = PricingPlan.PricingPlan;
const PricingRate = __importStar(require("./pricing-rate"));
exports.PricingRate = PricingRate.PricingRate;
const PrivateOfferAccountDetails = __importStar(require("./private-offer-account-details"));
exports.PrivateOfferAccountDetails = PrivateOfferAccountDetails.PrivateOfferAccountDetails;
const Product = __importStar(require("./product"));
exports.Product = Product.Product;
const ProductCollection = __importStar(require("./product-collection"));
exports.ProductCollection = ProductCollection.ProductCollection;
const ProductSummary = __importStar(require("./product-summary"));
exports.ProductSummary = ProductSummary.ProductSummary;
const PublicationStatus = __importStar(require("./publication-status"));
exports.PublicationStatus = PublicationStatus.PublicationStatus;
const PublishListingRevisionAsPrivateDetails = __importStar(require("./publish-listing-revision-as-private-details"));
exports.PublishListingRevisionAsPrivateDetails = PublishListingRevisionAsPrivateDetails.PublishListingRevisionAsPrivateDetails;
const Publisher = __importStar(require("./publisher"));
exports.Publisher = Publisher.Publisher;
const PublisherCollection = __importStar(require("./publisher-collection"));
exports.PublisherCollection = PublisherCollection.PublisherCollection;
const PublisherSummary = __importStar(require("./publisher-summary"));
exports.PublisherSummary = PublisherSummary.PublisherSummary;
const ShapeMemoryOptions = __importStar(require("./shape-memory-options"));
exports.ShapeMemoryOptions = ShapeMemoryOptions.ShapeMemoryOptions;
const ShapeOcpuOptions = __importStar(require("./shape-ocpu-options"));
exports.ShapeOcpuOptions = ShapeOcpuOptions.ShapeOcpuOptions;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const StackArtifactDetails = __importStar(require("./stack-artifact-details"));
exports.StackArtifactDetails = StackArtifactDetails.StackArtifactDetails;
const SubmitListingRevisionForReviewDetails = __importStar(require("./submit-listing-revision-for-review-details"));
exports.SubmitListingRevisionForReviewDetails = SubmitListingRevisionForReviewDetails.SubmitListingRevisionForReviewDetails;
const SupportContact = __importStar(require("./support-contact"));
exports.SupportContact = SupportContact.SupportContact;
const SupportDoc = __importStar(require("./support-doc"));
exports.SupportDoc = SupportDoc.SupportDoc;
const SupportDocCollection = __importStar(require("./support-doc-collection"));
exports.SupportDocCollection = SupportDocCollection.SupportDocCollection;
const SupportDocSummary = __importStar(require("./support-doc-summary"));
exports.SupportDocSummary = SupportDocSummary.SupportDocSummary;
const SupportedCurrencyCollection = __importStar(require("./supported-currency-collection"));
exports.SupportedCurrencyCollection = SupportedCurrencyCollection.SupportedCurrencyCollection;
const SupportedCurrencySummary = __importStar(require("./supported-currency-summary"));
exports.SupportedCurrencySummary = SupportedCurrencySummary.SupportedCurrencySummary;
const SupportedShapeCollection = __importStar(require("./supported-shape-collection"));
exports.SupportedShapeCollection = SupportedShapeCollection.SupportedShapeCollection;
const SupportedShapeSummary = __importStar(require("./supported-shape-summary"));
exports.SupportedShapeSummary = SupportedShapeSummary.SupportedShapeSummary;
const Term = __importStar(require("./term"));
exports.Term = Term.Term;
const TermCollection = __importStar(require("./term-collection"));
exports.TermCollection = TermCollection.TermCollection;
const TermSummary = __importStar(require("./term-summary"));
exports.TermSummary = TermSummary.TermSummary;
const TermVersion = __importStar(require("./term-version"));
exports.TermVersion = TermVersion.TermVersion;
const TermVersionAttachment = __importStar(require("./term-version-attachment"));
exports.TermVersionAttachment = TermVersionAttachment.TermVersionAttachment;
const TermVersionCollection = __importStar(require("./term-version-collection"));
exports.TermVersionCollection = TermVersionCollection.TermVersionCollection;
const TermVersionSummary = __importStar(require("./term-version-summary"));
exports.TermVersionSummary = TermVersionSummary.TermVersionSummary;
const TrainedProfessionals = __importStar(require("./trained-professionals"));
exports.TrainedProfessionals = TrainedProfessionals.TrainedProfessionals;
const UpdateArtifactDetails = __importStar(require("./update-artifact-details"));
exports.UpdateArtifactDetails = UpdateArtifactDetails.UpdateArtifactDetails;
const UpdateContainerImageDetails = __importStar(require("./update-container-image-details"));
exports.UpdateContainerImageDetails = UpdateContainerImageDetails.UpdateContainerImageDetails;
const UpdateHelmChartImageDetails = __importStar(require("./update-helm-chart-image-details"));
exports.UpdateHelmChartImageDetails = UpdateHelmChartImageDetails.UpdateHelmChartImageDetails;
const UpdateListingDetails = __importStar(require("./update-listing-details"));
exports.UpdateListingDetails = UpdateListingDetails.UpdateListingDetails;
const UpdateListingRevisionAttachmentDetails = __importStar(require("./update-listing-revision-attachment-details"));
exports.UpdateListingRevisionAttachmentDetails = UpdateListingRevisionAttachmentDetails.UpdateListingRevisionAttachmentDetails;
const UpdateListingRevisionDetails = __importStar(require("./update-listing-revision-details"));
exports.UpdateListingRevisionDetails = UpdateListingRevisionDetails.UpdateListingRevisionDetails;
const UpdateListingRevisionNoteDetails = __importStar(require("./update-listing-revision-note-details"));
exports.UpdateListingRevisionNoteDetails = UpdateListingRevisionNoteDetails.UpdateListingRevisionNoteDetails;
const UpdateListingRevisionPackageDetails = __importStar(require("./update-listing-revision-package-details"));
exports.UpdateListingRevisionPackageDetails = UpdateListingRevisionPackageDetails.UpdateListingRevisionPackageDetails;
const UpdateMachineImageDetails = __importStar(require("./update-machine-image-details"));
exports.UpdateMachineImageDetails = UpdateMachineImageDetails.UpdateMachineImageDetails;
const UpdateStackDetails = __importStar(require("./update-stack-details"));
exports.UpdateStackDetails = UpdateStackDetails.UpdateStackDetails;
const UpdateTermDetails = __importStar(require("./update-term-details"));
exports.UpdateTermDetails = UpdateTermDetails.UpdateTermDetails;
const UpdateTermVersionDetails = __importStar(require("./update-term-version-details"));
exports.UpdateTermVersionDetails = UpdateTermVersionDetails.UpdateTermVersionDetails;
const UploadData = __importStar(require("./upload-data"));
exports.UploadData = UploadData.UploadData;
const ValidationStatus = __importStar(require("./validation-status"));
exports.ValidationStatus = ValidationStatus.ValidationStatus;
const VersionDetails = __importStar(require("./version-details"));
exports.VersionDetails = VersionDetails.VersionDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const ContainerImageArtifact = __importStar(require("./container-image-artifact"));
exports.ContainerImageArtifact = ContainerImageArtifact.ContainerImageArtifact;
const ContainerPackage = __importStar(require("./container-package"));
exports.ContainerPackage = ContainerPackage.ContainerPackage;
const CreateContainerImageArtifactDetails = __importStar(require("./create-container-image-artifact-details"));
exports.CreateContainerImageArtifactDetails = CreateContainerImageArtifactDetails.CreateContainerImageArtifactDetails;
const CreateCustomerSuccessAttachment = __importStar(require("./create-customer-success-attachment"));
exports.CreateCustomerSuccessAttachment = CreateCustomerSuccessAttachment.CreateCustomerSuccessAttachment;
const CreateKubernetesImageArtifactDetails = __importStar(require("./create-kubernetes-image-artifact-details"));
exports.CreateKubernetesImageArtifactDetails = CreateKubernetesImageArtifactDetails.CreateKubernetesImageArtifactDetails;
const CreateLeadGenListingRevisionDetails = __importStar(require("./create-lead-gen-listing-revision-details"));
exports.CreateLeadGenListingRevisionDetails = CreateLeadGenListingRevisionDetails.CreateLeadGenListingRevisionDetails;
const CreateMachineImageArtifactDetails = __importStar(require("./create-machine-image-artifact-details"));
exports.CreateMachineImageArtifactDetails = CreateMachineImageArtifactDetails.CreateMachineImageArtifactDetails;
const CreateOciListingRevisionDetails = __importStar(require("./create-oci-listing-revision-details"));
exports.CreateOciListingRevisionDetails = CreateOciListingRevisionDetails.CreateOciListingRevisionDetails;
const CreateRelatedDocumentAttachmentDetails = __importStar(require("./create-related-document-attachment-details"));
exports.CreateRelatedDocumentAttachmentDetails = CreateRelatedDocumentAttachmentDetails.CreateRelatedDocumentAttachmentDetails;
const CreateReviewSupportDocumentAttachment = __importStar(require("./create-review-support-document-attachment"));
exports.CreateReviewSupportDocumentAttachment = CreateReviewSupportDocumentAttachment.CreateReviewSupportDocumentAttachment;
const CreateScreenShotAttachmentDetails = __importStar(require("./create-screen-shot-attachment-details"));
exports.CreateScreenShotAttachmentDetails = CreateScreenShotAttachmentDetails.CreateScreenShotAttachmentDetails;
const CreateServiceListingRevisionDetails = __importStar(require("./create-service-listing-revision-details"));
exports.CreateServiceListingRevisionDetails = CreateServiceListingRevisionDetails.CreateServiceListingRevisionDetails;
const CreateStackArtifactDetails = __importStar(require("./create-stack-artifact-details"));
exports.CreateStackArtifactDetails = CreateStackArtifactDetails.CreateStackArtifactDetails;
const CreateSupportedServiceAttachment = __importStar(require("./create-supported-service-attachment"));
exports.CreateSupportedServiceAttachment = CreateSupportedServiceAttachment.CreateSupportedServiceAttachment;
const CreateVideoAttachmentDetails = __importStar(require("./create-video-attachment-details"));
exports.CreateVideoAttachmentDetails = CreateVideoAttachmentDetails.CreateVideoAttachmentDetails;
const CustomerSuccessAttachment = __importStar(require("./customer-success-attachment"));
exports.CustomerSuccessAttachment = CustomerSuccessAttachment.CustomerSuccessAttachment;
const HelmChartPackage = __importStar(require("./helm-chart-package"));
exports.HelmChartPackage = HelmChartPackage.HelmChartPackage;
const KubernetesImageArtifact = __importStar(require("./kubernetes-image-artifact"));
exports.KubernetesImageArtifact = KubernetesImageArtifact.KubernetesImageArtifact;
const LeadGenListingRevision = __importStar(require("./lead-gen-listing-revision"));
exports.LeadGenListingRevision = LeadGenListingRevision.LeadGenListingRevision;
const MachineImageArtifact = __importStar(require("./machine-image-artifact"));
exports.MachineImageArtifact = MachineImageArtifact.MachineImageArtifact;
const MachineImagePackage = __importStar(require("./machine-image-package"));
exports.MachineImagePackage = MachineImagePackage.MachineImagePackage;
const MeteredPricingPlan = __importStar(require("./metered-pricing-plan"));
exports.MeteredPricingPlan = MeteredPricingPlan.MeteredPricingPlan;
const OciListingRevision = __importStar(require("./oci-listing-revision"));
exports.OciListingRevision = OciListingRevision.OciListingRevision;
const RelatedDocumentAttachment = __importStar(require("./related-document-attachment"));
exports.RelatedDocumentAttachment = RelatedDocumentAttachment.RelatedDocumentAttachment;
const ReviewSupportDocumentAttachment = __importStar(require("./review-support-document-attachment"));
exports.ReviewSupportDocumentAttachment = ReviewSupportDocumentAttachment.ReviewSupportDocumentAttachment;
const SaaSPricingPlan = __importStar(require("./saa-spricing-plan"));
exports.SaaSPricingPlan = SaaSPricingPlan.SaaSPricingPlan;
const ScreenShotAttachment = __importStar(require("./screen-shot-attachment"));
exports.ScreenShotAttachment = ScreenShotAttachment.ScreenShotAttachment;
const ServiceListingRevision = __importStar(require("./service-listing-revision"));
exports.ServiceListingRevision = ServiceListingRevision.ServiceListingRevision;
const StackArtifact = __importStar(require("./stack-artifact"));
exports.StackArtifact = StackArtifact.StackArtifact;
const StackPackage = __importStar(require("./stack-package"));
exports.StackPackage = StackPackage.StackPackage;
const SupportedServiceAttachment = __importStar(require("./supported-service-attachment"));
exports.SupportedServiceAttachment = SupportedServiceAttachment.SupportedServiceAttachment;
const UpdateContainerImageArtifactDetails = __importStar(require("./update-container-image-artifact-details"));
exports.UpdateContainerImageArtifactDetails = UpdateContainerImageArtifactDetails.UpdateContainerImageArtifactDetails;
const UpdateCustomerSuccessAttachment = __importStar(require("./update-customer-success-attachment"));
exports.UpdateCustomerSuccessAttachment = UpdateCustomerSuccessAttachment.UpdateCustomerSuccessAttachment;
const UpdateKubernetesImageArtifactDetails = __importStar(require("./update-kubernetes-image-artifact-details"));
exports.UpdateKubernetesImageArtifactDetails = UpdateKubernetesImageArtifactDetails.UpdateKubernetesImageArtifactDetails;
const UpdateLeadGenListingRevisionDetails = __importStar(require("./update-lead-gen-listing-revision-details"));
exports.UpdateLeadGenListingRevisionDetails = UpdateLeadGenListingRevisionDetails.UpdateLeadGenListingRevisionDetails;
const UpdateMachineImageArtifactDetails = __importStar(require("./update-machine-image-artifact-details"));
exports.UpdateMachineImageArtifactDetails = UpdateMachineImageArtifactDetails.UpdateMachineImageArtifactDetails;
const UpdateOciListingRevisionDetails = __importStar(require("./update-oci-listing-revision-details"));
exports.UpdateOciListingRevisionDetails = UpdateOciListingRevisionDetails.UpdateOciListingRevisionDetails;
const UpdateRelatedDocumentAttachmentDetails = __importStar(require("./update-related-document-attachment-details"));
exports.UpdateRelatedDocumentAttachmentDetails = UpdateRelatedDocumentAttachmentDetails.UpdateRelatedDocumentAttachmentDetails;
const UpdateReviewSupportDocumentAttachment = __importStar(require("./update-review-support-document-attachment"));
exports.UpdateReviewSupportDocumentAttachment = UpdateReviewSupportDocumentAttachment.UpdateReviewSupportDocumentAttachment;
const UpdateScreenShotAttachmentDetails = __importStar(require("./update-screen-shot-attachment-details"));
exports.UpdateScreenShotAttachmentDetails = UpdateScreenShotAttachmentDetails.UpdateScreenShotAttachmentDetails;
const UpdateServiceListingRevisionDetails = __importStar(require("./update-service-listing-revision-details"));
exports.UpdateServiceListingRevisionDetails = UpdateServiceListingRevisionDetails.UpdateServiceListingRevisionDetails;
const UpdateStackArtifactDetails = __importStar(require("./update-stack-artifact-details"));
exports.UpdateStackArtifactDetails = UpdateStackArtifactDetails.UpdateStackArtifactDetails;
const UpdateSupportedServiceAttachment = __importStar(require("./update-supported-service-attachment"));
exports.UpdateSupportedServiceAttachment = UpdateSupportedServiceAttachment.UpdateSupportedServiceAttachment;
const UpdateVideoAttachmentDetails = __importStar(require("./update-video-attachment-details"));
exports.UpdateVideoAttachmentDetails = UpdateVideoAttachmentDetails.UpdateVideoAttachmentDetails;
const VideoAttachment = __importStar(require("./video-attachment"));
exports.VideoAttachment = VideoAttachment.VideoAttachment;
