// oci-aidataplatform/lib/model/operation-type.js
"use strict";
/**
 * AiDataPlatform Control Plane API
 * Use the AiDataPlatform Control Plane API to manage Data Lakes.
 * OpenAPI spec version: 20240831
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * The asynchronous operation tracked by the work request.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateDataLake"] = "CREATE_DATA_LAKE";
    OperationType["UpdateDataLake"] = "UPDATE_DATA_LAKE";
    OperationType["DeleteDataLake"] = "DELETE_DATA_LAKE";
    OperationType["MoveDataLake"] = "MOVE_DATA_LAKE";
    OperationType["CreateWorkspace"] = "CREATE_WORKSPACE";
    OperationType["UpdateWorkspace"] = "UPDATE_WORKSPACE";
    OperationType["DeleteWorkspace"] = "DELETE_WORKSPACE";
    OperationType["EvaluateIamPolicies"] = "EVALUATE_IAM_POLICIES";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
