// oci-datasafe/lib/model/associated-resource-summary.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AssociatedResourceSummary = void 0;
var AssociatedResourceSummary;
(function (AssociatedResourceSummary) {
    let AssociatedResourceType;
    (function (AssociatedResourceType) {
        AssociatedResourceType["AuditPolicy"] = "AUDIT_POLICY";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AssociatedResourceType["UnknownValue"] = "UNKNOWN_VALUE";
    })(AssociatedResourceType = AssociatedResourceSummary.AssociatedResourceType || (AssociatedResourceSummary.AssociatedResourceType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AssociatedResourceSummary.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AssociatedResourceSummary.getDeserializedJsonObj = getDeserializedJsonObj;
})(AssociatedResourceSummary = exports.AssociatedResourceSummary || (exports.AssociatedResourceSummary = {}));
