// oci-datasafe/lib/model/attribute-set.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.AttributeSet = void 0;
var AttributeSet;
(function (AttributeSet) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Updating"] = "UPDATING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = AttributeSet.LifecycleState || (AttributeSet.LifecycleState = {}));
    let AttributeSetType;
    (function (AttributeSetType) {
        AttributeSetType["IpAddress"] = "IP_ADDRESS";
        AttributeSetType["ClientProgram"] = "CLIENT_PROGRAM";
        AttributeSetType["OsUser"] = "OS_USER";
        AttributeSetType["DatabaseUser"] = "DATABASE_USER";
        AttributeSetType["DatabaseObject"] = "DATABASE_OBJECT";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AttributeSetType["UnknownValue"] = "UNKNOWN_VALUE";
    })(AttributeSetType = AttributeSet.AttributeSetType || (AttributeSet.AttributeSetType = {}));
    let InUse;
    (function (InUse) {
        InUse["Yes"] = "YES";
        InUse["No"] = "NO";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        InUse["UnknownValue"] = "UNKNOWN_VALUE";
    })(InUse = AttributeSet.InUse || (AttributeSet.InUse = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AttributeSet.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    AttributeSet.getDeserializedJsonObj = getDeserializedJsonObj;
})(AttributeSet = exports.AttributeSet || (exports.AttributeSet = {}));
