// oci-datasafe/lib/model/confidence-level-enum.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConfidenceLevelEnum = void 0;
/**
 * The confidence level of a sensitive column or a discovery job result
 **/
var ConfidenceLevelEnum;
(function (ConfidenceLevelEnum) {
    ConfidenceLevelEnum["None"] = "NONE";
    ConfidenceLevelEnum["High"] = "HIGH";
    ConfidenceLevelEnum["Medium"] = "MEDIUM";
    ConfidenceLevelEnum["Low"] = "LOW";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ConfidenceLevelEnum["UnknownValue"] = "UNKNOWN_VALUE";
})(ConfidenceLevelEnum = exports.ConfidenceLevelEnum || (exports.ConfidenceLevelEnum = {}));
(function (ConfidenceLevelEnum) {
    function getJsonObj(obj) {
        return obj;
    }
    ConfidenceLevelEnum.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ConfidenceLevelEnum.getDeserializedJsonObj = getDeserializedJsonObj;
})(ConfidenceLevelEnum = exports.ConfidenceLevelEnum || (exports.ConfidenceLevelEnum = {}));
