// oci-datasafe/lib/model/firewall-config-details.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FirewallConfigDetails = void 0;
var FirewallConfigDetails;
(function (FirewallConfigDetails) {
    let Status;
    (function (Status) {
        Status["Enabled"] = "ENABLED";
        Status["Disabled"] = "DISABLED";
    })(Status = FirewallConfigDetails.Status || (FirewallConfigDetails.Status = {}));
    let ViolationLogAutoPurge;
    (function (ViolationLogAutoPurge) {
        ViolationLogAutoPurge["Enabled"] = "ENABLED";
        ViolationLogAutoPurge["Disabled"] = "DISABLED";
    })(ViolationLogAutoPurge = FirewallConfigDetails.ViolationLogAutoPurge || (FirewallConfigDetails.ViolationLogAutoPurge = {}));
    let ExcludeJob;
    (function (ExcludeJob) {
        ExcludeJob["Excluded"] = "EXCLUDED";
        ExcludeJob["Included"] = "INCLUDED";
    })(ExcludeJob = FirewallConfigDetails.ExcludeJob || (FirewallConfigDetails.ExcludeJob = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FirewallConfigDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FirewallConfigDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(FirewallConfigDetails = exports.FirewallConfigDetails || (exports.FirewallConfigDetails = {}));
