// oci-datasafe/lib/model/firewall-config.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FirewallConfig = void 0;
var FirewallConfig;
(function (FirewallConfig) {
    let Status;
    (function (Status) {
        Status["Enabled"] = "ENABLED";
        Status["Disabled"] = "DISABLED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = FirewallConfig.Status || (FirewallConfig.Status = {}));
    let ViolationLogAutoPurge;
    (function (ViolationLogAutoPurge) {
        ViolationLogAutoPurge["Enabled"] = "ENABLED";
        ViolationLogAutoPurge["Disabled"] = "DISABLED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ViolationLogAutoPurge["UnknownValue"] = "UNKNOWN_VALUE";
    })(ViolationLogAutoPurge = FirewallConfig.ViolationLogAutoPurge || (FirewallConfig.ViolationLogAutoPurge = {}));
    let ExcludeJob;
    (function (ExcludeJob) {
        ExcludeJob["Excluded"] = "EXCLUDED";
        ExcludeJob["Included"] = "INCLUDED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ExcludeJob["UnknownValue"] = "UNKNOWN_VALUE";
    })(ExcludeJob = FirewallConfig.ExcludeJob || (FirewallConfig.ExcludeJob = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FirewallConfig.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    FirewallConfig.getDeserializedJsonObj = getDeserializedJsonObj;
})(FirewallConfig = exports.FirewallConfig || (exports.FirewallConfig = {}));
