// oci-datasafe/lib/model/policy-condition.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PolicyCondition = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var PolicyCondition;
(function (PolicyCondition) {
    let EntitySelection;
    (function (EntitySelection) {
        EntitySelection["Include"] = "INCLUDE";
        EntitySelection["Exclude"] = "EXCLUDE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        EntitySelection["UnknownValue"] = "UNKNOWN_VALUE";
    })(EntitySelection = PolicyCondition.EntitySelection || (PolicyCondition.EntitySelection = {}));
    let OperationStatus;
    (function (OperationStatus) {
        OperationStatus["Success"] = "SUCCESS";
        OperationStatus["Failure"] = "FAILURE";
        OperationStatus["Both"] = "BOTH";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        OperationStatus["UnknownValue"] = "UNKNOWN_VALUE";
    })(OperationStatus = PolicyCondition.OperationStatus || (PolicyCondition.OperationStatus = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "entityType" in obj && obj.entityType) {
            switch (obj.entityType) {
                case "ATTRIBUTE_SET":
                    return model.AttributeSetCondition.getJsonObj(jsonObj, true);
                case "USER":
                    return model.UserCondition.getJsonObj(jsonObj, true);
                case "ROLE":
                    return model.RoleCondition.getJsonObj(jsonObj, true);
                case "ALL_USERS":
                    return model.AllUserCondition.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.entityType}`);
            }
        }
        return jsonObj;
    }
    PolicyCondition.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        if (obj && "entityType" in obj && obj.entityType) {
            switch (obj.entityType) {
                case "ATTRIBUTE_SET":
                    return model.AttributeSetCondition.getDeserializedJsonObj(jsonObj, true);
                case "USER":
                    return model.UserCondition.getDeserializedJsonObj(jsonObj, true);
                case "ROLE":
                    return model.RoleCondition.getDeserializedJsonObj(jsonObj, true);
                case "ALL_USERS":
                    return model.AllUserCondition.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.entityType}`);
            }
        }
        return jsonObj;
    }
    PolicyCondition.getDeserializedJsonObj = getDeserializedJsonObj;
})(PolicyCondition = exports.PolicyCondition || (exports.PolicyCondition = {}));
