// oci-datasafe/lib/model/security-policy-config-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityPolicyConfigLifecycleState = void 0;
/**
 * The current state of the security policy configuration.
 **/
var SecurityPolicyConfigLifecycleState;
(function (SecurityPolicyConfigLifecycleState) {
    SecurityPolicyConfigLifecycleState["Creating"] = "CREATING";
    SecurityPolicyConfigLifecycleState["Updating"] = "UPDATING";
    SecurityPolicyConfigLifecycleState["Active"] = "ACTIVE";
    SecurityPolicyConfigLifecycleState["Failed"] = "FAILED";
    SecurityPolicyConfigLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    SecurityPolicyConfigLifecycleState["Deleting"] = "DELETING";
    SecurityPolicyConfigLifecycleState["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SecurityPolicyConfigLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(SecurityPolicyConfigLifecycleState = exports.SecurityPolicyConfigLifecycleState || (exports.SecurityPolicyConfigLifecycleState = {}));
(function (SecurityPolicyConfigLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    SecurityPolicyConfigLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SecurityPolicyConfigLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(SecurityPolicyConfigLifecycleState = exports.SecurityPolicyConfigLifecycleState || (exports.SecurityPolicyConfigLifecycleState = {}));
