// oci-datasafe/lib/model/security-assessment-target-type.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityAssessmentTargetType = void 0;
/**
 * Indicates whether the security assessment is for a target database or a target database group.
 **/
var SecurityAssessmentTargetType;
(function (SecurityAssessmentTargetType) {
    SecurityAssessmentTargetType["TargetDatabase"] = "TARGET_DATABASE";
    SecurityAssessmentTargetType["TargetDatabaseGroup"] = "TARGET_DATABASE_GROUP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SecurityAssessmentTargetType["UnknownValue"] = "UNKNOWN_VALUE";
})(SecurityAssessmentTargetType = exports.SecurityAssessmentTargetType || (exports.SecurityAssessmentTargetType = {}));
(function (SecurityAssessmentTargetType) {
    function getJsonObj(obj) {
        return obj;
    }
    SecurityAssessmentTargetType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SecurityAssessmentTargetType.getDeserializedJsonObj = getDeserializedJsonObj;
})(SecurityAssessmentTargetType = exports.SecurityAssessmentTargetType || (exports.SecurityAssessmentTargetType = {}));
