// oci-datasafe/lib/model/unified-audit-policy-definition-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnifiedAuditPolicyDefinitionLifecycleState = void 0;
/**
 * The current state of Unified Audit Policy
 **/
var UnifiedAuditPolicyDefinitionLifecycleState;
(function (UnifiedAuditPolicyDefinitionLifecycleState) {
    UnifiedAuditPolicyDefinitionLifecycleState["Creating"] = "CREATING";
    UnifiedAuditPolicyDefinitionLifecycleState["Updating"] = "UPDATING";
    UnifiedAuditPolicyDefinitionLifecycleState["Active"] = "ACTIVE";
    UnifiedAuditPolicyDefinitionLifecycleState["Inactive"] = "INACTIVE";
    UnifiedAuditPolicyDefinitionLifecycleState["Failed"] = "FAILED";
    UnifiedAuditPolicyDefinitionLifecycleState["Deleting"] = "DELETING";
    UnifiedAuditPolicyDefinitionLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    UnifiedAuditPolicyDefinitionLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(UnifiedAuditPolicyDefinitionLifecycleState = exports.UnifiedAuditPolicyDefinitionLifecycleState || (exports.UnifiedAuditPolicyDefinitionLifecycleState = {}));
(function (UnifiedAuditPolicyDefinitionLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    UnifiedAuditPolicyDefinitionLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    UnifiedAuditPolicyDefinitionLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(UnifiedAuditPolicyDefinitionLifecycleState = exports.UnifiedAuditPolicyDefinitionLifecycleState || (exports.UnifiedAuditPolicyDefinitionLifecycleState = {}));
