// oci-datasafe/lib/model/unified-audit-policy-definition.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnifiedAuditPolicyDefinition = void 0;
var UnifiedAuditPolicyDefinition;
(function (UnifiedAuditPolicyDefinition) {
    let AuditPolicyCategory;
    (function (AuditPolicyCategory) {
        AuditPolicyCategory["BasicActivity"] = "BASIC_ACTIVITY";
        AuditPolicyCategory["AdminUserActivity"] = "ADMIN_USER_ACTIVITY";
        AuditPolicyCategory["UserActivity"] = "USER_ACTIVITY";
        AuditPolicyCategory["OraclePredefined"] = "ORACLE_PREDEFINED";
        AuditPolicyCategory["ComplianceStandard"] = "COMPLIANCE_STANDARD";
        AuditPolicyCategory["SqlFirewallAuditing"] = "SQL_FIREWALL_AUDITING";
        AuditPolicyCategory["Custom"] = "CUSTOM";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        AuditPolicyCategory["UnknownValue"] = "UNKNOWN_VALUE";
    })(AuditPolicyCategory = UnifiedAuditPolicyDefinition.AuditPolicyCategory || (UnifiedAuditPolicyDefinition.AuditPolicyCategory = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UnifiedAuditPolicyDefinition.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    UnifiedAuditPolicyDefinition.getDeserializedJsonObj = getDeserializedJsonObj;
})(UnifiedAuditPolicyDefinition = exports.UnifiedAuditPolicyDefinition || (exports.UnifiedAuditPolicyDefinition = {}));
