// oci-datasafe/lib/model/unified-audit-policy-lifecycle-state.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnifiedAuditPolicyLifecycleState = void 0;
/**
 * The current state of unified audit policy
 **/
var UnifiedAuditPolicyLifecycleState;
(function (UnifiedAuditPolicyLifecycleState) {
    UnifiedAuditPolicyLifecycleState["Creating"] = "CREATING";
    UnifiedAuditPolicyLifecycleState["Updating"] = "UPDATING";
    UnifiedAuditPolicyLifecycleState["Active"] = "ACTIVE";
    UnifiedAuditPolicyLifecycleState["Inactive"] = "INACTIVE";
    UnifiedAuditPolicyLifecycleState["Failed"] = "FAILED";
    UnifiedAuditPolicyLifecycleState["Deleting"] = "DELETING";
    UnifiedAuditPolicyLifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
    UnifiedAuditPolicyLifecycleState["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    UnifiedAuditPolicyLifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
})(UnifiedAuditPolicyLifecycleState = exports.UnifiedAuditPolicyLifecycleState || (exports.UnifiedAuditPolicyLifecycleState = {}));
(function (UnifiedAuditPolicyLifecycleState) {
    function getJsonObj(obj) {
        return obj;
    }
    UnifiedAuditPolicyLifecycleState.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    UnifiedAuditPolicyLifecycleState.getDeserializedJsonObj = getDeserializedJsonObj;
})(UnifiedAuditPolicyLifecycleState = exports.UnifiedAuditPolicyLifecycleState || (exports.UnifiedAuditPolicyLifecycleState = {}));
