// oci-datasafe/lib/model/unified-audit-policy.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnifiedAuditPolicy = void 0;
const model = __importStar(require("../model"));
var UnifiedAuditPolicy;
(function (UnifiedAuditPolicy) {
    let Status;
    (function (Status) {
        Status["Enabled"] = "ENABLED";
        Status["Disabled"] = "DISABLED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = UnifiedAuditPolicy.Status || (UnifiedAuditPolicy.Status = {}));
    let EnabledEntities;
    (function (EnabledEntities) {
        EnabledEntities["AllUsers"] = "ALL_USERS";
        EnabledEntities["IncludeUsers"] = "INCLUDE_USERS";
        EnabledEntities["IncludeRoles"] = "INCLUDE_ROLES";
        EnabledEntities["ExcludeUsers"] = "EXCLUDE_USERS";
        EnabledEntities["IncludeUsersRoles"] = "INCLUDE_USERS_ROLES";
        EnabledEntities["Disabled"] = "DISABLED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        EnabledEntities["UnknownValue"] = "UNKNOWN_VALUE";
    })(EnabledEntities = UnifiedAuditPolicy.EnabledEntities || (UnifiedAuditPolicy.EnabledEntities = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "conditions": obj.conditions
                ? obj.conditions.map(item => {
                    return model.PolicyCondition.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    UnifiedAuditPolicy.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "conditions": obj.conditions
                ? obj.conditions.map(item => {
                    return model.PolicyCondition.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    UnifiedAuditPolicy.getDeserializedJsonObj = getDeserializedJsonObj;
})(UnifiedAuditPolicy = exports.UnifiedAuditPolicy || (exports.UnifiedAuditPolicy = {}));
