// oci-datasafe/lib/model/user-assessment-target-type.js
"use strict";
/**
 * Data Safe API
 * APIs for using Oracle Data Safe.
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserAssessmentTargetType = void 0;
/**
 * Indicates whether the user assessment is for a target database or a target database group.
 **/
var UserAssessmentTargetType;
(function (UserAssessmentTargetType) {
    UserAssessmentTargetType["TargetDatabase"] = "TARGET_DATABASE";
    UserAssessmentTargetType["TargetDatabaseGroup"] = "TARGET_DATABASE_GROUP";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    UserAssessmentTargetType["UnknownValue"] = "UNKNOWN_VALUE";
})(UserAssessmentTargetType = exports.UserAssessmentTargetType || (exports.UserAssessmentTargetType = {}));
(function (UserAssessmentTargetType) {
    function getJsonObj(obj) {
        return obj;
    }
    UserAssessmentTargetType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    UserAssessmentTargetType.getDeserializedJsonObj = getDeserializedJsonObj;
})(UserAssessmentTargetType = exports.UserAssessmentTargetType || (exports.UserAssessmentTargetType = {}));
