// oci-resourceanalytics/lib/model/index.js
"use strict";
/**
 * Resource Analytics API
 * Use the Resource Analytics API to manage Resource Analytics Instances.
 * OpenAPI spec version: 20241031
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VaultSecretPasswordDetails = exports.PlainTextPasswordDetails = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateTenancyAttachmentDetails = exports.UpdateResourceAnalyticsInstanceDetails = exports.TenancyAttachmentSummary = exports.TenancyAttachmentCollection = exports.TenancyAttachment = exports.SortOrder = exports.ResourceAnalyticsInstanceSummary = exports.ResourceAnalyticsInstanceOacNetworkDetails = exports.ResourceAnalyticsInstanceOacAttachmentDetails = exports.ResourceAnalyticsInstanceEnableOacDetails = exports.ResourceAnalyticsInstanceCollection = exports.ResourceAnalyticsInstance = exports.OperationType = exports.OperationStatus = exports.MonitoredRegionSummary = exports.MonitoredRegionCollection = exports.MonitoredRegion = exports.CreateTenancyAttachmentDetails = exports.CreateResourceAnalyticsInstanceDetails = exports.CreateMonitoredRegionDetails = exports.ChangeResourceAnalyticsInstanceCompartmentDetails = exports.AdwAdminPasswordDetails = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AdwAdminPasswordDetails = __importStar(require("./adw-admin-password-details"));
exports.AdwAdminPasswordDetails = AdwAdminPasswordDetails.AdwAdminPasswordDetails;
const ChangeResourceAnalyticsInstanceCompartmentDetails = __importStar(require("./change-resource-analytics-instance-compartment-details"));
exports.ChangeResourceAnalyticsInstanceCompartmentDetails = ChangeResourceAnalyticsInstanceCompartmentDetails.ChangeResourceAnalyticsInstanceCompartmentDetails;
const CreateMonitoredRegionDetails = __importStar(require("./create-monitored-region-details"));
exports.CreateMonitoredRegionDetails = CreateMonitoredRegionDetails.CreateMonitoredRegionDetails;
const CreateResourceAnalyticsInstanceDetails = __importStar(require("./create-resource-analytics-instance-details"));
exports.CreateResourceAnalyticsInstanceDetails = CreateResourceAnalyticsInstanceDetails.CreateResourceAnalyticsInstanceDetails;
const CreateTenancyAttachmentDetails = __importStar(require("./create-tenancy-attachment-details"));
exports.CreateTenancyAttachmentDetails = CreateTenancyAttachmentDetails.CreateTenancyAttachmentDetails;
const MonitoredRegion = __importStar(require("./monitored-region"));
exports.MonitoredRegion = MonitoredRegion.MonitoredRegion;
const MonitoredRegionCollection = __importStar(require("./monitored-region-collection"));
exports.MonitoredRegionCollection = MonitoredRegionCollection.MonitoredRegionCollection;
const MonitoredRegionSummary = __importStar(require("./monitored-region-summary"));
exports.MonitoredRegionSummary = MonitoredRegionSummary.MonitoredRegionSummary;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const ResourceAnalyticsInstance = __importStar(require("./resource-analytics-instance"));
exports.ResourceAnalyticsInstance = ResourceAnalyticsInstance.ResourceAnalyticsInstance;
const ResourceAnalyticsInstanceCollection = __importStar(require("./resource-analytics-instance-collection"));
exports.ResourceAnalyticsInstanceCollection = ResourceAnalyticsInstanceCollection.ResourceAnalyticsInstanceCollection;
const ResourceAnalyticsInstanceEnableOacDetails = __importStar(require("./resource-analytics-instance-enable-oac-details"));
exports.ResourceAnalyticsInstanceEnableOacDetails = ResourceAnalyticsInstanceEnableOacDetails.ResourceAnalyticsInstanceEnableOacDetails;
const ResourceAnalyticsInstanceOacAttachmentDetails = __importStar(require("./resource-analytics-instance-oac-attachment-details"));
exports.ResourceAnalyticsInstanceOacAttachmentDetails = ResourceAnalyticsInstanceOacAttachmentDetails.ResourceAnalyticsInstanceOacAttachmentDetails;
const ResourceAnalyticsInstanceOacNetworkDetails = __importStar(require("./resource-analytics-instance-oac-network-details"));
exports.ResourceAnalyticsInstanceOacNetworkDetails = ResourceAnalyticsInstanceOacNetworkDetails.ResourceAnalyticsInstanceOacNetworkDetails;
const ResourceAnalyticsInstanceSummary = __importStar(require("./resource-analytics-instance-summary"));
exports.ResourceAnalyticsInstanceSummary = ResourceAnalyticsInstanceSummary.ResourceAnalyticsInstanceSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const TenancyAttachment = __importStar(require("./tenancy-attachment"));
exports.TenancyAttachment = TenancyAttachment.TenancyAttachment;
const TenancyAttachmentCollection = __importStar(require("./tenancy-attachment-collection"));
exports.TenancyAttachmentCollection = TenancyAttachmentCollection.TenancyAttachmentCollection;
const TenancyAttachmentSummary = __importStar(require("./tenancy-attachment-summary"));
exports.TenancyAttachmentSummary = TenancyAttachmentSummary.TenancyAttachmentSummary;
const UpdateResourceAnalyticsInstanceDetails = __importStar(require("./update-resource-analytics-instance-details"));
exports.UpdateResourceAnalyticsInstanceDetails = UpdateResourceAnalyticsInstanceDetails.UpdateResourceAnalyticsInstanceDetails;
const UpdateTenancyAttachmentDetails = __importStar(require("./update-tenancy-attachment-details"));
exports.UpdateTenancyAttachmentDetails = UpdateTenancyAttachmentDetails.UpdateTenancyAttachmentDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const PlainTextPasswordDetails = __importStar(require("./plain-text-password-details"));
exports.PlainTextPasswordDetails = PlainTextPasswordDetails.PlainTextPasswordDetails;
const VaultSecretPasswordDetails = __importStar(require("./vault-secret-password-details"));
exports.VaultSecretPasswordDetails = VaultSecretPasswordDetails.VaultSecretPasswordDetails;
