// oci-resourceanalytics/lib/model/operation-type.js
"use strict";
/**
 * Resource Analytics API
 * Use the Resource Analytics API to manage Resource Analytics Instances.
 * OpenAPI spec version: 20241031
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * The asynchronous operation tracked by the work request.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateResourceAnalyticsInstance"] = "CREATE_RESOURCE_ANALYTICS_INSTANCE";
    OperationType["UpdateResourceAnalyticsInstance"] = "UPDATE_RESOURCE_ANALYTICS_INSTANCE";
    OperationType["DeleteResourceAnalyticsInstance"] = "DELETE_RESOURCE_ANALYTICS_INSTANCE";
    OperationType["MoveResourceAnalyticsInstance"] = "MOVE_RESOURCE_ANALYTICS_INSTANCE";
    OperationType["CreateTenancyAttachment"] = "CREATE_TENANCY_ATTACHMENT";
    OperationType["UpdateTenancyAttachment"] = "UPDATE_TENANCY_ATTACHMENT";
    OperationType["DeleteTenancyAttachment"] = "DELETE_TENANCY_ATTACHMENT";
    OperationType["CreateMonitoredRegion"] = "CREATE_MONITORED_REGION";
    OperationType["DeleteMonitoredRegion"] = "DELETE_MONITORED_REGION";
    OperationType["EnableOac"] = "ENABLE_OAC";
    OperationType["DisableOac"] = "DISABLE_OAC";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
