// oci-resourceanalytics/lib/model/resource-analytics-instance.js
"use strict";
/**
 * Resource Analytics API
 * Use the Resource Analytics API to manage Resource Analytics Instances.
 * OpenAPI spec version: 20241031
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceAnalyticsInstance = void 0;
var ResourceAnalyticsInstance;
(function (ResourceAnalyticsInstance) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ResourceAnalyticsInstance.LifecycleState || (ResourceAnalyticsInstance.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ResourceAnalyticsInstance.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ResourceAnalyticsInstance.getDeserializedJsonObj = getDeserializedJsonObj;
})(ResourceAnalyticsInstance = exports.ResourceAnalyticsInstance || (exports.ResourceAnalyticsInstance = {}));
