// oci-psa/lib/model/index.js
"use strict";
/**
 * PrivateServiceAccess Control Plane API
 * Use the PrivateServiceAccess Control Plane API to manage privateServiceAccess.
 * OpenAPI spec version: 20240301
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdatePrivateServiceAccessDetails = exports.SortOrder = exports.PsaServiceSummary = exports.PsaServiceCollection = exports.PrivateServiceAccessSummary = exports.PrivateServiceAccessCollection = exports.PrivateServiceAccess = exports.OperationType = exports.OperationStatus = exports.CreatePrivateServiceAccessDetails = exports.ChangePrivateServiceAccessCompartmentDetails = exports.ActionType = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const ChangePrivateServiceAccessCompartmentDetails = __importStar(require("./change-private-service-access-compartment-details"));
exports.ChangePrivateServiceAccessCompartmentDetails = ChangePrivateServiceAccessCompartmentDetails.ChangePrivateServiceAccessCompartmentDetails;
const CreatePrivateServiceAccessDetails = __importStar(require("./create-private-service-access-details"));
exports.CreatePrivateServiceAccessDetails = CreatePrivateServiceAccessDetails.CreatePrivateServiceAccessDetails;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const PrivateServiceAccess = __importStar(require("./private-service-access"));
exports.PrivateServiceAccess = PrivateServiceAccess.PrivateServiceAccess;
const PrivateServiceAccessCollection = __importStar(require("./private-service-access-collection"));
exports.PrivateServiceAccessCollection = PrivateServiceAccessCollection.PrivateServiceAccessCollection;
const PrivateServiceAccessSummary = __importStar(require("./private-service-access-summary"));
exports.PrivateServiceAccessSummary = PrivateServiceAccessSummary.PrivateServiceAccessSummary;
const PsaServiceCollection = __importStar(require("./psa-service-collection"));
exports.PsaServiceCollection = PsaServiceCollection.PsaServiceCollection;
const PsaServiceSummary = __importStar(require("./psa-service-summary"));
exports.PsaServiceSummary = PsaServiceSummary.PsaServiceSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const UpdatePrivateServiceAccessDetails = __importStar(require("./update-private-service-access-details"));
exports.UpdatePrivateServiceAccessDetails = UpdatePrivateServiceAccessDetails.UpdatePrivateServiceAccessDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
