// oci-dif/lib/model/index.js
"use strict";
/**
 * DataIntelligences Control Plane API
 * Use the DataIntelligences Control Plane API to manage dataIntelligences.
 * OpenAPI spec version: 20250830
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkRequestErrorCollection = exports.WorkRequestError = exports.WorkRequest = exports.UpdateStackDetails = exports.UpdateDataflowConnections = exports.StorageTier = exports.StackTemplate = exports.StackSummary = exports.StackCollection = exports.Stack = exports.SortOrder = exports.ShapeConfig = exports.ServiceDetailResponse = exports.Service = exports.OperationType = exports.OperationStatus = exports.ObjectVersioning = exports.ObjectStorageUpdateDetail = exports.ObjectStorageDetail = exports.GgcsUserDetail = exports.GgcsUpdateDetail = exports.GgcsTargetDetail = exports.GgcsSourceDetail = exports.GgcsDetail = exports.GgcsConnectionDetails = exports.GgcsArtifactsDetail = exports.GenAiUpdateDetail = exports.GenAiDetail = exports.EndpointDetails = exports.EndpointAdditional = exports.DifDependencyDetails = exports.DeployArtifactsDetails = exports.DbWorkload = exports.DbCredentialsDetail = exports.DataflowUpdateDetail = exports.DataflowDetail = exports.DataflowConnections = exports.DataflowConnectionDetails = exports.DataflowArtifactsDetail = exports.CreateStackDetails = exports.ClusterType = exports.ChangeStackCompartmentDetails = exports.AutoTiering = exports.AssignedConnectionDetails = exports.AdditionalDetails = exports.AddServiceDetails = exports.AdbUpdateDetail = exports.AdbDetail = exports.AdbArtifactsDetail = exports.ActionType = void 0;
exports.WorkflowAction = exports.WorkRequestSummaryCollection = exports.WorkRequestSummary = exports.WorkRequestResourceMetadataKey = exports.WorkRequestResource = exports.WorkRequestLogEntryCollection = exports.WorkRequestLogEntry = void 0;
const ActionType = __importStar(require("./action-type"));
exports.ActionType = ActionType.ActionType;
const AdbArtifactsDetail = __importStar(require("./adb-artifacts-detail"));
exports.AdbArtifactsDetail = AdbArtifactsDetail.AdbArtifactsDetail;
const AdbDetail = __importStar(require("./adb-detail"));
exports.AdbDetail = AdbDetail.AdbDetail;
const AdbUpdateDetail = __importStar(require("./adb-update-detail"));
exports.AdbUpdateDetail = AdbUpdateDetail.AdbUpdateDetail;
const AddServiceDetails = __importStar(require("./add-service-details"));
exports.AddServiceDetails = AddServiceDetails.AddServiceDetails;
const AdditionalDetails = __importStar(require("./additional-details"));
exports.AdditionalDetails = AdditionalDetails.AdditionalDetails;
const AssignedConnectionDetails = __importStar(require("./assigned-connection-details"));
exports.AssignedConnectionDetails = AssignedConnectionDetails.AssignedConnectionDetails;
const AutoTiering = __importStar(require("./auto-tiering"));
exports.AutoTiering = AutoTiering.AutoTiering;
const ChangeStackCompartmentDetails = __importStar(require("./change-stack-compartment-details"));
exports.ChangeStackCompartmentDetails = ChangeStackCompartmentDetails.ChangeStackCompartmentDetails;
const ClusterType = __importStar(require("./cluster-type"));
exports.ClusterType = ClusterType.ClusterType;
const CreateStackDetails = __importStar(require("./create-stack-details"));
exports.CreateStackDetails = CreateStackDetails.CreateStackDetails;
const DataflowArtifactsDetail = __importStar(require("./dataflow-artifacts-detail"));
exports.DataflowArtifactsDetail = DataflowArtifactsDetail.DataflowArtifactsDetail;
const DataflowConnectionDetails = __importStar(require("./dataflow-connection-details"));
exports.DataflowConnectionDetails = DataflowConnectionDetails.DataflowConnectionDetails;
const DataflowConnections = __importStar(require("./dataflow-connections"));
exports.DataflowConnections = DataflowConnections.DataflowConnections;
const DataflowDetail = __importStar(require("./dataflow-detail"));
exports.DataflowDetail = DataflowDetail.DataflowDetail;
const DataflowUpdateDetail = __importStar(require("./dataflow-update-detail"));
exports.DataflowUpdateDetail = DataflowUpdateDetail.DataflowUpdateDetail;
const DbCredentialsDetail = __importStar(require("./db-credentials-detail"));
exports.DbCredentialsDetail = DbCredentialsDetail.DbCredentialsDetail;
const DbWorkload = __importStar(require("./db-workload"));
exports.DbWorkload = DbWorkload.DbWorkload;
const DeployArtifactsDetails = __importStar(require("./deploy-artifacts-details"));
exports.DeployArtifactsDetails = DeployArtifactsDetails.DeployArtifactsDetails;
const DifDependencyDetails = __importStar(require("./dif-dependency-details"));
exports.DifDependencyDetails = DifDependencyDetails.DifDependencyDetails;
const EndpointAdditional = __importStar(require("./endpoint-additional"));
exports.EndpointAdditional = EndpointAdditional.EndpointAdditional;
const EndpointDetails = __importStar(require("./endpoint-details"));
exports.EndpointDetails = EndpointDetails.EndpointDetails;
const GenAiDetail = __importStar(require("./gen-ai-detail"));
exports.GenAiDetail = GenAiDetail.GenAiDetail;
const GenAiUpdateDetail = __importStar(require("./gen-ai-update-detail"));
exports.GenAiUpdateDetail = GenAiUpdateDetail.GenAiUpdateDetail;
const GgcsArtifactsDetail = __importStar(require("./ggcs-artifacts-detail"));
exports.GgcsArtifactsDetail = GgcsArtifactsDetail.GgcsArtifactsDetail;
const GgcsConnectionDetails = __importStar(require("./ggcs-connection-details"));
exports.GgcsConnectionDetails = GgcsConnectionDetails.GgcsConnectionDetails;
const GgcsDetail = __importStar(require("./ggcs-detail"));
exports.GgcsDetail = GgcsDetail.GgcsDetail;
const GgcsSourceDetail = __importStar(require("./ggcs-source-detail"));
exports.GgcsSourceDetail = GgcsSourceDetail.GgcsSourceDetail;
const GgcsTargetDetail = __importStar(require("./ggcs-target-detail"));
exports.GgcsTargetDetail = GgcsTargetDetail.GgcsTargetDetail;
const GgcsUpdateDetail = __importStar(require("./ggcs-update-detail"));
exports.GgcsUpdateDetail = GgcsUpdateDetail.GgcsUpdateDetail;
const GgcsUserDetail = __importStar(require("./ggcs-user-detail"));
exports.GgcsUserDetail = GgcsUserDetail.GgcsUserDetail;
const ObjectStorageDetail = __importStar(require("./object-storage-detail"));
exports.ObjectStorageDetail = ObjectStorageDetail.ObjectStorageDetail;
const ObjectStorageUpdateDetail = __importStar(require("./object-storage-update-detail"));
exports.ObjectStorageUpdateDetail = ObjectStorageUpdateDetail.ObjectStorageUpdateDetail;
const ObjectVersioning = __importStar(require("./object-versioning"));
exports.ObjectVersioning = ObjectVersioning.ObjectVersioning;
const OperationStatus = __importStar(require("./operation-status"));
exports.OperationStatus = OperationStatus.OperationStatus;
const OperationType = __importStar(require("./operation-type"));
exports.OperationType = OperationType.OperationType;
const Service = __importStar(require("./service"));
exports.Service = Service.Service;
const ServiceDetailResponse = __importStar(require("./service-detail-response"));
exports.ServiceDetailResponse = ServiceDetailResponse.ServiceDetailResponse;
const ShapeConfig = __importStar(require("./shape-config"));
exports.ShapeConfig = ShapeConfig.ShapeConfig;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const Stack = __importStar(require("./stack"));
exports.Stack = Stack.Stack;
const StackCollection = __importStar(require("./stack-collection"));
exports.StackCollection = StackCollection.StackCollection;
const StackSummary = __importStar(require("./stack-summary"));
exports.StackSummary = StackSummary.StackSummary;
const StackTemplate = __importStar(require("./stack-template"));
exports.StackTemplate = StackTemplate.StackTemplate;
const StorageTier = __importStar(require("./storage-tier"));
exports.StorageTier = StorageTier.StorageTier;
const UpdateDataflowConnections = __importStar(require("./update-dataflow-connections"));
exports.UpdateDataflowConnections = UpdateDataflowConnections.UpdateDataflowConnections;
const UpdateStackDetails = __importStar(require("./update-stack-details"));
exports.UpdateStackDetails = UpdateStackDetails.UpdateStackDetails;
const WorkRequest = __importStar(require("./work-request"));
exports.WorkRequest = WorkRequest.WorkRequest;
const WorkRequestError = __importStar(require("./work-request-error"));
exports.WorkRequestError = WorkRequestError.WorkRequestError;
const WorkRequestErrorCollection = __importStar(require("./work-request-error-collection"));
exports.WorkRequestErrorCollection = WorkRequestErrorCollection.WorkRequestErrorCollection;
const WorkRequestLogEntry = __importStar(require("./work-request-log-entry"));
exports.WorkRequestLogEntry = WorkRequestLogEntry.WorkRequestLogEntry;
const WorkRequestLogEntryCollection = __importStar(require("./work-request-log-entry-collection"));
exports.WorkRequestLogEntryCollection = WorkRequestLogEntryCollection.WorkRequestLogEntryCollection;
const WorkRequestResource = __importStar(require("./work-request-resource"));
exports.WorkRequestResource = WorkRequestResource.WorkRequestResource;
const WorkRequestResourceMetadataKey = __importStar(require("./work-request-resource-metadata-key"));
exports.WorkRequestResourceMetadataKey = WorkRequestResourceMetadataKey.WorkRequestResourceMetadataKey;
const WorkRequestSummary = __importStar(require("./work-request-summary"));
exports.WorkRequestSummary = WorkRequestSummary.WorkRequestSummary;
const WorkRequestSummaryCollection = __importStar(require("./work-request-summary-collection"));
exports.WorkRequestSummaryCollection = WorkRequestSummaryCollection.WorkRequestSummaryCollection;
const WorkflowAction = __importStar(require("./workflow-action"));
exports.WorkflowAction = WorkflowAction.WorkflowAction;
