// oci-dif/lib/model/object-versioning.js
"use strict";
/**
 * DataIntelligences Control Plane API
 * Use the DataIntelligences Control Plane API to manage dataIntelligences.
 * OpenAPI spec version: 20250830
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectVersioning = void 0;
/**
 * Mentions whether the object versioning to be enabled or not,Allowed values are \"ENABLED\" / \"DISABLED\"/\"SUSPENDED\"
 **/
var ObjectVersioning;
(function (ObjectVersioning) {
    ObjectVersioning["Enabled"] = "ENABLED";
    ObjectVersioning["Suspended"] = "SUSPENDED";
    ObjectVersioning["Disabled"] = "DISABLED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ObjectVersioning["UnknownValue"] = "UNKNOWN_VALUE";
})(ObjectVersioning = exports.ObjectVersioning || (exports.ObjectVersioning = {}));
(function (ObjectVersioning) {
    function getJsonObj(obj) {
        return obj;
    }
    ObjectVersioning.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ObjectVersioning.getDeserializedJsonObj = getDeserializedJsonObj;
})(ObjectVersioning = exports.ObjectVersioning || (exports.ObjectVersioning = {}));
