// oci-dif/lib/model/operation-type.js
"use strict";
/**
 * DataIntelligences Control Plane API
 * Use the DataIntelligences Control Plane API to manage dataIntelligences.
 * OpenAPI spec version: 20250830
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OperationType = void 0;
/**
 * The asynchronous operation tracked by the work request.
 **/
var OperationType;
(function (OperationType) {
    OperationType["CreateStack"] = "CREATE_STACK";
    OperationType["UpdateStack"] = "UPDATE_STACK";
    OperationType["DeleteStack"] = "DELETE_STACK";
    OperationType["MoveStack"] = "MOVE_STACK";
    OperationType["DeployArtifacts"] = "DEPLOY_ARTIFACTS";
    OperationType["AddServiceInStack"] = "ADD_SERVICE_IN_STACK";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    OperationType["UnknownValue"] = "UNKNOWN_VALUE";
})(OperationType = exports.OperationType || (exports.OperationType = {}));
(function (OperationType) {
    function getJsonObj(obj) {
        return obj;
    }
    OperationType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    OperationType.getDeserializedJsonObj = getDeserializedJsonObj;
})(OperationType = exports.OperationType || (exports.OperationType = {}));
