// oci-dif/lib/model/service.js
"use strict";
/**
 * DataIntelligences Control Plane API
 * Use the DataIntelligences Control Plane API to manage dataIntelligences.
 * OpenAPI spec version: 20250830
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Service = void 0;
/**
 * Service for the stack
 **/
var Service;
(function (Service) {
    Service["Adb"] = "ADB";
    Service["Ggcs"] = "GGCS";
    Service["Objectstorage"] = "OBJECTSTORAGE";
    Service["Genai"] = "GENAI";
    Service["Dataflow"] = "DATAFLOW";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    Service["UnknownValue"] = "UNKNOWN_VALUE";
})(Service = exports.Service || (exports.Service = {}));
(function (Service) {
    function getJsonObj(obj) {
        return obj;
    }
    Service.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    Service.getDeserializedJsonObj = getDeserializedJsonObj;
})(Service = exports.Service || (exports.Service = {}));
