// oci-dif/lib/model/stack.js
"use strict";
/**
 * DataIntelligences Control Plane API
 * Use the DataIntelligences Control Plane API to manage dataIntelligences.
 * OpenAPI spec version: 20250830
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Stack = void 0;
const model = __importStar(require("../model"));
var Stack;
(function (Stack) {
    let StackTemplates;
    (function (StackTemplates) {
        StackTemplates["Datalake"] = "DATALAKE";
        StackTemplates["Datapipeline"] = "DATAPIPELINE";
        StackTemplates["Aiservices"] = "AISERVICES";
        StackTemplates["Datatransformation"] = "DATATRANSFORMATION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        StackTemplates["UnknownValue"] = "UNKNOWN_VALUE";
    })(StackTemplates = Stack.StackTemplates || (Stack.StackTemplates = {}));
    let Services;
    (function (Services) {
        Services["Adb"] = "ADB";
        Services["Ggcs"] = "GGCS";
        Services["Objectstorage"] = "OBJECTSTORAGE";
        Services["Genai"] = "GENAI";
        Services["Dataflow"] = "DATAFLOW";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Services["UnknownValue"] = "UNKNOWN_VALUE";
    })(Services = Stack.Services || (Stack.Services = {}));
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        LifecycleState["NeedsAttention"] = "NEEDS_ATTENTION";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = Stack.LifecycleState || (Stack.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "adb": obj.adb
                ? obj.adb.map(item => {
                    return model.AdbDetail.getJsonObj(item);
                })
                : undefined,
            "ggcs": obj.ggcs
                ? obj.ggcs.map(item => {
                    return model.GgcsDetail.getJsonObj(item);
                })
                : undefined,
            "dataflow": obj.dataflow
                ? obj.dataflow.map(item => {
                    return model.DataflowDetail.getJsonObj(item);
                })
                : undefined,
            "objectstorage": obj.objectstorage
                ? obj.objectstorage.map(item => {
                    return model.ObjectStorageDetail.getJsonObj(item);
                })
                : undefined,
            "genai": obj.genai
                ? obj.genai.map(item => {
                    return model.GenAiDetail.getJsonObj(item);
                })
                : undefined,
            "serviceDetails": obj.serviceDetails
                ? obj.serviceDetails.map(item => {
                    return model.ServiceDetailResponse.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    Stack.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "adb": obj.adb
                ? obj.adb.map(item => {
                    return model.AdbDetail.getDeserializedJsonObj(item);
                })
                : undefined,
            "ggcs": obj.ggcs
                ? obj.ggcs.map(item => {
                    return model.GgcsDetail.getDeserializedJsonObj(item);
                })
                : undefined,
            "dataflow": obj.dataflow
                ? obj.dataflow.map(item => {
                    return model.DataflowDetail.getDeserializedJsonObj(item);
                })
                : undefined,
            "objectstorage": obj.objectstorage
                ? obj.objectstorage.map(item => {
                    return model.ObjectStorageDetail.getDeserializedJsonObj(item);
                })
                : undefined,
            "genai": obj.genai
                ? obj.genai.map(item => {
                    return model.GenAiDetail.getDeserializedJsonObj(item);
                })
                : undefined,
            "serviceDetails": obj.serviceDetails
                ? obj.serviceDetails.map(item => {
                    return model.ServiceDetailResponse.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    Stack.getDeserializedJsonObj = getDeserializedJsonObj;
})(Stack = exports.Stack || (exports.Stack = {}));
