// oci-dif/lib/model/stack-template.js
"use strict";
/**
 * DataIntelligences Control Plane API
 * Use the DataIntelligences Control Plane API to manage dataIntelligences.
 * OpenAPI spec version: 20250830
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StackTemplate = void 0;
/**
 * Template for the stack
 **/
var StackTemplate;
(function (StackTemplate) {
    StackTemplate["Datalake"] = "DATALAKE";
    StackTemplate["Datapipeline"] = "DATAPIPELINE";
    StackTemplate["Aiservices"] = "AISERVICES";
    StackTemplate["Datatransformation"] = "DATATRANSFORMATION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    StackTemplate["UnknownValue"] = "UNKNOWN_VALUE";
})(StackTemplate = exports.StackTemplate || (exports.StackTemplate = {}));
(function (StackTemplate) {
    function getJsonObj(obj) {
        return obj;
    }
    StackTemplate.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    StackTemplate.getDeserializedJsonObj = getDeserializedJsonObj;
})(StackTemplate = exports.StackTemplate || (exports.StackTemplate = {}));
