// oci-dif/lib/model/update-stack-details.js
"use strict";
/**
 * DataIntelligences Control Plane API
 * Use the DataIntelligences Control Plane API to manage dataIntelligences.
 * OpenAPI spec version: 20250830
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateStackDetails = void 0;
const model = __importStar(require("../model"));
var UpdateStackDetails;
(function (UpdateStackDetails) {
    let StackTemplates;
    (function (StackTemplates) {
        StackTemplates["Datalake"] = "DATALAKE";
        StackTemplates["Datapipeline"] = "DATAPIPELINE";
        StackTemplates["Aiservices"] = "AISERVICES";
        StackTemplates["Datatransformation"] = "DATATRANSFORMATION";
    })(StackTemplates = UpdateStackDetails.StackTemplates || (UpdateStackDetails.StackTemplates = {}));
    let Services;
    (function (Services) {
        Services["Adb"] = "ADB";
        Services["Ggcs"] = "GGCS";
        Services["Objectstorage"] = "OBJECTSTORAGE";
        Services["Genai"] = "GENAI";
        Services["Dataflow"] = "DATAFLOW";
    })(Services = UpdateStackDetails.Services || (UpdateStackDetails.Services = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "adb": obj.adb
                ? obj.adb.map(item => {
                    return model.AdbUpdateDetail.getJsonObj(item);
                })
                : undefined,
            "ggcs": obj.ggcs
                ? obj.ggcs.map(item => {
                    return model.GgcsUpdateDetail.getJsonObj(item);
                })
                : undefined,
            "dataflow": obj.dataflow
                ? obj.dataflow.map(item => {
                    return model.DataflowUpdateDetail.getJsonObj(item);
                })
                : undefined,
            "objectstorage": obj.objectstorage
                ? obj.objectstorage.map(item => {
                    return model.ObjectStorageUpdateDetail.getJsonObj(item);
                })
                : undefined,
            "genai": obj.genai
                ? obj.genai.map(item => {
                    return model.GenAiUpdateDetail.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    UpdateStackDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "adb": obj.adb
                ? obj.adb.map(item => {
                    return model.AdbUpdateDetail.getDeserializedJsonObj(item);
                })
                : undefined,
            "ggcs": obj.ggcs
                ? obj.ggcs.map(item => {
                    return model.GgcsUpdateDetail.getDeserializedJsonObj(item);
                })
                : undefined,
            "dataflow": obj.dataflow
                ? obj.dataflow.map(item => {
                    return model.DataflowUpdateDetail.getDeserializedJsonObj(item);
                })
                : undefined,
            "objectstorage": obj.objectstorage
                ? obj.objectstorage.map(item => {
                    return model.ObjectStorageUpdateDetail.getDeserializedJsonObj(item);
                })
                : undefined,
            "genai": obj.genai
                ? obj.genai.map(item => {
                    return model.GenAiUpdateDetail.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    UpdateStackDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(UpdateStackDetails = exports.UpdateStackDetails || (exports.UpdateStackDetails = {}));
