// oci-databasemanagement/lib/model/cloud-cluster-scan-listener-configuration.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudClusterScanListenerConfiguration = void 0;
var CloudClusterScanListenerConfiguration;
(function (CloudClusterScanListenerConfiguration) {
    let ScanProtocol;
    (function (ScanProtocol) {
        ScanProtocol["Tcp"] = "TCP";
        ScanProtocol["Tcps"] = "TCPS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ScanProtocol["UnknownValue"] = "UNKNOWN_VALUE";
    })(ScanProtocol = CloudClusterScanListenerConfiguration.ScanProtocol || (CloudClusterScanListenerConfiguration.ScanProtocol = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CloudClusterScanListenerConfiguration.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CloudClusterScanListenerConfiguration.getDeserializedJsonObj = getDeserializedJsonObj;
})(CloudClusterScanListenerConfiguration = exports.CloudClusterScanListenerConfiguration || (exports.CloudClusterScanListenerConfiguration = {}));
