// oci-databasemanagement/lib/model/cloud-cluster.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudCluster = void 0;
const model = __importStar(require("../model"));
var CloudCluster;
(function (CloudCluster) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["NotConnected"] = "NOT_CONNECTED";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Inactive"] = "INACTIVE";
        LifecycleState["Updating"] = "UPDATING";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = CloudCluster.LifecycleState || (CloudCluster.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "networkConfigurations": obj.networkConfigurations
                ? obj.networkConfigurations.map(item => {
                    return model.CloudClusterNetworkConfiguration.getJsonObj(item);
                })
                : undefined,
            "vipConfigurations": obj.vipConfigurations
                ? obj.vipConfigurations.map(item => {
                    return model.CloudClusterVipConfiguration.getJsonObj(item);
                })
                : undefined,
            "scanConfigurations": obj.scanConfigurations
                ? obj.scanConfigurations.map(item => {
                    return model.CloudClusterScanListenerConfiguration.getJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    CloudCluster.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "networkConfigurations": obj.networkConfigurations
                ? obj.networkConfigurations.map(item => {
                    return model.CloudClusterNetworkConfiguration.getDeserializedJsonObj(item);
                })
                : undefined,
            "vipConfigurations": obj.vipConfigurations
                ? obj.vipConfigurations.map(item => {
                    return model.CloudClusterVipConfiguration.getDeserializedJsonObj(item);
                })
                : undefined,
            "scanConfigurations": obj.scanConfigurations
                ? obj.scanConfigurations.map(item => {
                    return model.CloudClusterScanListenerConfiguration.getDeserializedJsonObj(item);
                })
                : undefined
        });
        return jsonObj;
    }
    CloudCluster.getDeserializedJsonObj = getDeserializedJsonObj;
})(CloudCluster = exports.CloudCluster || (exports.CloudCluster = {}));
