// oci-databasemanagement/lib/model/cloud-db-system-component-type.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudDbSystemComponentType = void 0;
/**
 * The cloud DB system component type.
 **/
var CloudDbSystemComponentType;
(function (CloudDbSystemComponentType) {
    CloudDbSystemComponentType["Asm"] = "ASM";
    CloudDbSystemComponentType["AsmInstance"] = "ASM_INSTANCE";
    CloudDbSystemComponentType["Cluster"] = "CLUSTER";
    CloudDbSystemComponentType["ClusterInstance"] = "CLUSTER_INSTANCE";
    CloudDbSystemComponentType["Database"] = "DATABASE";
    CloudDbSystemComponentType["DatabaseInstance"] = "DATABASE_INSTANCE";
    CloudDbSystemComponentType["DatabaseHome"] = "DATABASE_HOME";
    CloudDbSystemComponentType["DatabaseNode"] = "DATABASE_NODE";
    CloudDbSystemComponentType["Dbsystem"] = "DBSYSTEM";
    CloudDbSystemComponentType["Listener"] = "LISTENER";
    CloudDbSystemComponentType["PluggableDatabase"] = "PLUGGABLE_DATABASE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    CloudDbSystemComponentType["UnknownValue"] = "UNKNOWN_VALUE";
})(CloudDbSystemComponentType = exports.CloudDbSystemComponentType || (exports.CloudDbSystemComponentType = {}));
(function (CloudDbSystemComponentType) {
    function getJsonObj(obj) {
        return obj;
    }
    CloudDbSystemComponentType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    CloudDbSystemComponentType.getDeserializedJsonObj = getDeserializedJsonObj;
})(CloudDbSystemComponentType = exports.CloudDbSystemComponentType || (exports.CloudDbSystemComponentType = {}));
