// oci-databasemanagement/lib/model/discovered-cloud-db-system-component.js
"use strict";
/**
 * Database Management API
 * Use the Database Management API to monitor and manage resources such as
Oracle Databases, MySQL Databases, and External Database Systems.
For more information, see [Database Management](https://docs.oracle.com/iaas/database-management/home.htm).

 * OpenAPI spec version: 20201101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DiscoveredCloudDbSystemComponent = void 0;
const model = __importStar(require("../model"));
const common = require("oci-common");
var DiscoveredCloudDbSystemComponent;
(function (DiscoveredCloudDbSystemComponent) {
    let Status;
    (function (Status) {
        Status["New"] = "NEW";
        Status["Existing"] = "EXISTING";
        Status["ExistingBasic"] = "EXISTING_BASIC";
        Status["ExistingPe"] = "EXISTING_PE";
        Status["ExistingBasicPe"] = "EXISTING_BASIC_PE";
        Status["MarkedForUpgrade"] = "MARKED_FOR_UPGRADE";
        Status["MarkedForMigration"] = "MARKED_FOR_MIGRATION";
        Status["MarkedForUpgradeMigration"] = "MARKED_FOR_UPGRADE_MIGRATION";
        Status["MarkedForDeletion"] = "MARKED_FOR_DELETION";
        Status["Unknown"] = "UNKNOWN";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Status["UnknownValue"] = "UNKNOWN_VALUE";
    })(Status = DiscoveredCloudDbSystemComponent.Status || (DiscoveredCloudDbSystemComponent.Status = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "associatedComponents": obj.associatedComponents
                ? obj.associatedComponents.map(item => {
                    return model.AssociatedCloudComponent.getJsonObj(item);
                })
                : undefined
        });
        if (obj && "componentType" in obj && obj.componentType) {
            switch (obj.componentType) {
                case "CLUSTER_INSTANCE":
                    return model.DiscoveredCloudClusterInstance.getJsonObj(jsonObj, true);
                case "PLUGGABLE_DATABASE":
                    return model.DiscoveredCloudPluggableDatabase.getJsonObj(jsonObj, true);
                case "CLUSTER":
                    return model.DiscoveredCloudCluster.getJsonObj(jsonObj, true);
                case "ASM":
                    return model.DiscoveredCloudAsm.getJsonObj(jsonObj, true);
                case "LISTENER":
                    return model.DiscoveredCloudListener.getJsonObj(jsonObj, true);
                case "DATABASE_INSTANCE":
                    return model.DiscoveredCloudDbInstance.getJsonObj(jsonObj, true);
                case "DATABASE":
                    return model.DiscoveredCloudDatabase.getJsonObj(jsonObj, true);
                case "DATABASE_HOME":
                    return model.DiscoveredCloudDbHome.getJsonObj(jsonObj, true);
                case "DATABASE_NODE":
                    return model.DiscoveredCloudDbNode.getJsonObj(jsonObj, true);
                case "ASM_INSTANCE":
                    return model.DiscoveredCloudAsmInstance.getJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.componentType}`);
            }
        }
        return jsonObj;
    }
    DiscoveredCloudDbSystemComponent.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {
            "associatedComponents": obj.associatedComponents
                ? obj.associatedComponents.map(item => {
                    return model.AssociatedCloudComponent.getDeserializedJsonObj(item);
                })
                : undefined
        });
        if (obj && "componentType" in obj && obj.componentType) {
            switch (obj.componentType) {
                case "CLUSTER_INSTANCE":
                    return model.DiscoveredCloudClusterInstance.getDeserializedJsonObj(jsonObj, true);
                case "PLUGGABLE_DATABASE":
                    return model.DiscoveredCloudPluggableDatabase.getDeserializedJsonObj(jsonObj, true);
                case "CLUSTER":
                    return model.DiscoveredCloudCluster.getDeserializedJsonObj(jsonObj, true);
                case "ASM":
                    return model.DiscoveredCloudAsm.getDeserializedJsonObj(jsonObj, true);
                case "LISTENER":
                    return model.DiscoveredCloudListener.getDeserializedJsonObj(jsonObj, true);
                case "DATABASE_INSTANCE":
                    return model.DiscoveredCloudDbInstance.getDeserializedJsonObj(jsonObj, true);
                case "DATABASE":
                    return model.DiscoveredCloudDatabase.getDeserializedJsonObj(jsonObj, true);
                case "DATABASE_HOME":
                    return model.DiscoveredCloudDbHome.getDeserializedJsonObj(jsonObj, true);
                case "DATABASE_NODE":
                    return model.DiscoveredCloudDbNode.getDeserializedJsonObj(jsonObj, true);
                case "ASM_INSTANCE":
                    return model.DiscoveredCloudAsmInstance.getDeserializedJsonObj(jsonObj, true);
                default:
                    if (common.LOG.logger)
                        common.LOG.logger.info(`Unknown value for: ${obj.componentType}`);
            }
        }
        return jsonObj;
    }
    DiscoveredCloudDbSystemComponent.getDeserializedJsonObj = getDeserializedJsonObj;
})(DiscoveredCloudDbSystemComponent = exports.DiscoveredCloudDbSystemComponent || (exports.DiscoveredCloudDbSystemComponent = {}));
