// oci-multicloud/lib/model/index.js
"use strict";
/**
 * Oracle Multicloud API
 * Use the Oracle Multicloud API to retrieve resource anchors and network anchors, and the metadata mappings related a Cloud Service Provider. For more information, see <link to docs>.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GcpCloudServiceProviderMetadataItem = exports.AzureCloudServiceProviderMetadataItem = exports.AwsCloudServiceProviderMetadataItem = exports.SubscriptionType = exports.SortOrder = exports.ResourceAnchorSummary = exports.ResourceAnchorCollection = exports.ResourceAnchor = exports.Properties = exports.OmHubCp = exports.OciVcn = exports.OciNetworkSubnet = exports.OciNetworkMetadata = exports.OciDns = exports.NetworkAnchorSummary = exports.NetworkAnchorConnectionStatus = exports.NetworkAnchorCollection = exports.NetworkAnchor = exports.MulticloudSubscriptionSummary = exports.MulticloudSubscriptionCollection = exports.MulticloudResourceSummary = exports.MulticloudResourceCollection = exports.MultiCloudMetadataSummary = exports.MultiCloudMetadataCollection = exports.MultiCloudMetadata = exports.ExternalLocationsMetadatumSummary = exports.ExternalLocationsMetadatumCollection = exports.ExternalLocationSummary = exports.ExternalLocationSummariesMetadatumSummaryCollection = exports.ExternalLocationSummariesMetadatumSummary = exports.ExternalLocationMappingMetadatumSummaryCollection = exports.ExternalLocationMappingMetadatumSummary = exports.ExternalLocationDetail = exports.ExternalLocation = exports.CspZoneKeyReferenceId = exports.CloudServiceProviderNetworkMetadataItem = exports.CloudServiceProviderMetadataItem = exports.ActionKindHeader = void 0;
const ActionKindHeader = __importStar(require("./action-kind-header"));
exports.ActionKindHeader = ActionKindHeader.ActionKindHeader;
const CloudServiceProviderMetadataItem = __importStar(require("./cloud-service-provider-metadata-item"));
exports.CloudServiceProviderMetadataItem = CloudServiceProviderMetadataItem.CloudServiceProviderMetadataItem;
const CloudServiceProviderNetworkMetadataItem = __importStar(require("./cloud-service-provider-network-metadata-item"));
exports.CloudServiceProviderNetworkMetadataItem = CloudServiceProviderNetworkMetadataItem.CloudServiceProviderNetworkMetadataItem;
const CspZoneKeyReferenceId = __importStar(require("./csp-zone-key-reference-id"));
exports.CspZoneKeyReferenceId = CspZoneKeyReferenceId.CspZoneKeyReferenceId;
const ExternalLocation = __importStar(require("./external-location"));
exports.ExternalLocation = ExternalLocation.ExternalLocation;
const ExternalLocationDetail = __importStar(require("./external-location-detail"));
exports.ExternalLocationDetail = ExternalLocationDetail.ExternalLocationDetail;
const ExternalLocationMappingMetadatumSummary = __importStar(require("./external-location-mapping-metadatum-summary"));
exports.ExternalLocationMappingMetadatumSummary = ExternalLocationMappingMetadatumSummary.ExternalLocationMappingMetadatumSummary;
const ExternalLocationMappingMetadatumSummaryCollection = __importStar(require("./external-location-mapping-metadatum-summary-collection"));
exports.ExternalLocationMappingMetadatumSummaryCollection = ExternalLocationMappingMetadatumSummaryCollection.ExternalLocationMappingMetadatumSummaryCollection;
const ExternalLocationSummariesMetadatumSummary = __importStar(require("./external-location-summaries-metadatum-summary"));
exports.ExternalLocationSummariesMetadatumSummary = ExternalLocationSummariesMetadatumSummary.ExternalLocationSummariesMetadatumSummary;
const ExternalLocationSummariesMetadatumSummaryCollection = __importStar(require("./external-location-summaries-metadatum-summary-collection"));
exports.ExternalLocationSummariesMetadatumSummaryCollection = ExternalLocationSummariesMetadatumSummaryCollection.ExternalLocationSummariesMetadatumSummaryCollection;
const ExternalLocationSummary = __importStar(require("./external-location-summary"));
exports.ExternalLocationSummary = ExternalLocationSummary.ExternalLocationSummary;
const ExternalLocationsMetadatumCollection = __importStar(require("./external-locations-metadatum-collection"));
exports.ExternalLocationsMetadatumCollection = ExternalLocationsMetadatumCollection.ExternalLocationsMetadatumCollection;
const ExternalLocationsMetadatumSummary = __importStar(require("./external-locations-metadatum-summary"));
exports.ExternalLocationsMetadatumSummary = ExternalLocationsMetadatumSummary.ExternalLocationsMetadatumSummary;
const MultiCloudMetadata = __importStar(require("./multi-cloud-metadata"));
exports.MultiCloudMetadata = MultiCloudMetadata.MultiCloudMetadata;
const MultiCloudMetadataCollection = __importStar(require("./multi-cloud-metadata-collection"));
exports.MultiCloudMetadataCollection = MultiCloudMetadataCollection.MultiCloudMetadataCollection;
const MultiCloudMetadataSummary = __importStar(require("./multi-cloud-metadata-summary"));
exports.MultiCloudMetadataSummary = MultiCloudMetadataSummary.MultiCloudMetadataSummary;
const MulticloudResourceCollection = __importStar(require("./multicloud-resource-collection"));
exports.MulticloudResourceCollection = MulticloudResourceCollection.MulticloudResourceCollection;
const MulticloudResourceSummary = __importStar(require("./multicloud-resource-summary"));
exports.MulticloudResourceSummary = MulticloudResourceSummary.MulticloudResourceSummary;
const MulticloudSubscriptionCollection = __importStar(require("./multicloud-subscription-collection"));
exports.MulticloudSubscriptionCollection = MulticloudSubscriptionCollection.MulticloudSubscriptionCollection;
const MulticloudSubscriptionSummary = __importStar(require("./multicloud-subscription-summary"));
exports.MulticloudSubscriptionSummary = MulticloudSubscriptionSummary.MulticloudSubscriptionSummary;
const NetworkAnchor = __importStar(require("./network-anchor"));
exports.NetworkAnchor = NetworkAnchor.NetworkAnchor;
const NetworkAnchorCollection = __importStar(require("./network-anchor-collection"));
exports.NetworkAnchorCollection = NetworkAnchorCollection.NetworkAnchorCollection;
const NetworkAnchorConnectionStatus = __importStar(require("./network-anchor-connection-status"));
exports.NetworkAnchorConnectionStatus = NetworkAnchorConnectionStatus.NetworkAnchorConnectionStatus;
const NetworkAnchorSummary = __importStar(require("./network-anchor-summary"));
exports.NetworkAnchorSummary = NetworkAnchorSummary.NetworkAnchorSummary;
const OciDns = __importStar(require("./oci-dns"));
exports.OciDns = OciDns.OciDns;
const OciNetworkMetadata = __importStar(require("./oci-network-metadata"));
exports.OciNetworkMetadata = OciNetworkMetadata.OciNetworkMetadata;
const OciNetworkSubnet = __importStar(require("./oci-network-subnet"));
exports.OciNetworkSubnet = OciNetworkSubnet.OciNetworkSubnet;
const OciVcn = __importStar(require("./oci-vcn"));
exports.OciVcn = OciVcn.OciVcn;
const OmHubCp = __importStar(require("./om-hub-cp"));
exports.OmHubCp = OmHubCp.OmHubCp;
const Properties = __importStar(require("./properties"));
exports.Properties = Properties.Properties;
const ResourceAnchor = __importStar(require("./resource-anchor"));
exports.ResourceAnchor = ResourceAnchor.ResourceAnchor;
const ResourceAnchorCollection = __importStar(require("./resource-anchor-collection"));
exports.ResourceAnchorCollection = ResourceAnchorCollection.ResourceAnchorCollection;
const ResourceAnchorSummary = __importStar(require("./resource-anchor-summary"));
exports.ResourceAnchorSummary = ResourceAnchorSummary.ResourceAnchorSummary;
const SortOrder = __importStar(require("./sort-order"));
exports.SortOrder = SortOrder.SortOrder;
const SubscriptionType = __importStar(require("./subscription-type"));
exports.SubscriptionType = SubscriptionType.SubscriptionType;
const AwsCloudServiceProviderMetadataItem = __importStar(require("./aws-cloud-service-provider-metadata-item"));
exports.AwsCloudServiceProviderMetadataItem = AwsCloudServiceProviderMetadataItem.AwsCloudServiceProviderMetadataItem;
const AzureCloudServiceProviderMetadataItem = __importStar(require("./azure-cloud-service-provider-metadata-item"));
exports.AzureCloudServiceProviderMetadataItem = AzureCloudServiceProviderMetadataItem.AzureCloudServiceProviderMetadataItem;
const GcpCloudServiceProviderMetadataItem = __importStar(require("./gcp-cloud-service-provider-metadata-item"));
exports.GcpCloudServiceProviderMetadataItem = GcpCloudServiceProviderMetadataItem.GcpCloudServiceProviderMetadataItem;
