// oci-multicloud/lib/model/network-anchor-connection-status.js
"use strict";
/**
 * Oracle Multicloud API
 * Use the Oracle Multicloud API to retrieve resource anchors and network anchors, and the metadata mappings related a Cloud Service Provider. For more information, see <link to docs>.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NetworkAnchorConnectionStatus = void 0;
/**
 * This can be merge to lifecycleState
 * CONNECTED - Partner and CSI information is assigned and MulticloudLink provisioned.
 * DISCONNECTED - Only partner cloud information is assigned.
 * CONNECTING - Oracle Cloud Infrastructure information is assigned and the control plane is provisioning resources.
 * ACTIVE - Network anchor is connected and resources (VNICs) exist within a subnet.
 * ERROR - DRG attach fails during connection.
 * FAILED - Network anchor creation failed
 * NEEDS_ATTENTION - Network anchor is in temporary bad state
 * UPDATING - Network anchor is getting updated.
 * DELETING - Network anchor is getting deleted
 * DELETED - A connected network anchor is deleted.
 *
 **/
var NetworkAnchorConnectionStatus;
(function (NetworkAnchorConnectionStatus) {
    NetworkAnchorConnectionStatus["Connected"] = "CONNECTED";
    NetworkAnchorConnectionStatus["Disconnected"] = "DISCONNECTED";
    NetworkAnchorConnectionStatus["Connecting"] = "CONNECTING";
    NetworkAnchorConnectionStatus["Active"] = "ACTIVE";
    NetworkAnchorConnectionStatus["Error"] = "ERROR";
    NetworkAnchorConnectionStatus["Updating"] = "UPDATING";
    NetworkAnchorConnectionStatus["NeedsAttention"] = "NEEDS_ATTENTION";
    NetworkAnchorConnectionStatus["Failed"] = "FAILED";
    NetworkAnchorConnectionStatus["Deleting"] = "DELETING";
    NetworkAnchorConnectionStatus["Deleted"] = "DELETED";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    NetworkAnchorConnectionStatus["UnknownValue"] = "UNKNOWN_VALUE";
})(NetworkAnchorConnectionStatus = exports.NetworkAnchorConnectionStatus || (exports.NetworkAnchorConnectionStatus = {}));
(function (NetworkAnchorConnectionStatus) {
    function getJsonObj(obj) {
        return obj;
    }
    NetworkAnchorConnectionStatus.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    NetworkAnchorConnectionStatus.getDeserializedJsonObj = getDeserializedJsonObj;
})(NetworkAnchorConnectionStatus = exports.NetworkAnchorConnectionStatus || (exports.NetworkAnchorConnectionStatus = {}));
