// oci-multicloud/lib/model/subscription-type.js
"use strict";
/**
 * Oracle Multicloud API
 * Use the Oracle Multicloud API to retrieve resource anchors and network anchors, and the metadata mappings related a Cloud Service Provider. For more information, see <link to docs>.
 * OpenAPI spec version: 20180828
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubscriptionType = void 0;
/**
 * Oracle Cloud Infrastructure Subscription Type.
 **/
var SubscriptionType;
(function (SubscriptionType) {
    SubscriptionType["Oracledbatazure"] = "ORACLEDBATAZURE";
    SubscriptionType["Oracledbatgoogle"] = "ORACLEDBATGOOGLE";
    SubscriptionType["Oracledbataws"] = "ORACLEDBATAWS";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    SubscriptionType["UnknownValue"] = "UNKNOWN_VALUE";
})(SubscriptionType = exports.SubscriptionType || (exports.SubscriptionType = {}));
(function (SubscriptionType) {
    function getJsonObj(obj) {
        return obj;
    }
    SubscriptionType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    SubscriptionType.getDeserializedJsonObj = getDeserializedJsonObj;
})(SubscriptionType = exports.SubscriptionType || (exports.SubscriptionType = {}));
