// oci-fleetsoftwareupdate/lib/model/exadata-release-versions.js
"use strict";
/**
 * Exadata Fleet Update service API
 * Use the Exadata Fleet Update service to patch large collections of components directly,
as a single entity, orchestrating the maintenance actions to update all chosen components in the stack in a single cycle.

 * OpenAPI spec version: 20220528
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExadataReleaseVersions = void 0;
/**
 * Possible Exadata Releases.
 * Note that an Exadata Release depends on underlying major Oracle Linux OS version.
 * For example, Exadata Release 21.2 depends on Oracle Linux 7.
 * For more details, refer to [Oracle document 2075007.1](https://support.oracle.com/knowledge/Oracle%20Database%20Products/2075007_1.html)
 *
 **/
var ExadataReleaseVersions;
(function (ExadataReleaseVersions) {
    ExadataReleaseVersions["ExaOl5111"] = "EXA_OL_5_11_1";
    ExadataReleaseVersions["ExaOl5112"] = "EXA_OL_5_11_2";
    ExadataReleaseVersions["ExaOl5121"] = "EXA_OL_5_12_1";
    ExadataReleaseVersions["ExaOl6122"] = "EXA_OL_6_12_2";
    ExadataReleaseVersions["ExaOl6181"] = "EXA_OL_6_18_1";
    ExadataReleaseVersions["ExaOl7191"] = "EXA_OL_7_19_1";
    ExadataReleaseVersions["ExaOl7192"] = "EXA_OL_7_19_2";
    ExadataReleaseVersions["ExaOl7193"] = "EXA_OL_7_19_3";
    ExadataReleaseVersions["ExaOl7201"] = "EXA_OL_7_20_1";
    ExadataReleaseVersions["ExaOl7212"] = "EXA_OL_7_21_2";
    ExadataReleaseVersions["ExaOl7221"] = "EXA_OL_7_22_1";
    ExadataReleaseVersions["ExaOl8231"] = "EXA_OL_8_23_1";
    ExadataReleaseVersions["ExaOl8241"] = "EXA_OL_8_24_1";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ExadataReleaseVersions["UnknownValue"] = "UNKNOWN_VALUE";
})(ExadataReleaseVersions = exports.ExadataReleaseVersions || (exports.ExadataReleaseVersions = {}));
(function (ExadataReleaseVersions) {
    function getJsonObj(obj) {
        return obj;
    }
    ExadataReleaseVersions.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ExadataReleaseVersions.getDeserializedJsonObj = getDeserializedJsonObj;
})(ExadataReleaseVersions = exports.ExadataReleaseVersions || (exports.ExadataReleaseVersions = {}));
