// oci-fleetsoftwareupdate/lib/model/gi-goal-software-component-details.js
"use strict";
/**
 * Exadata Fleet Update service API
 * Use the Exadata Fleet Update service to patch large collections of components directly,
as a single entity, orchestrating the maintenance actions to update all chosen components in the stack in a single cycle.

 * OpenAPI spec version: 20220528
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GiGoalSoftwareComponentDetails = void 0;
const model = __importStar(require("../model"));
var GiGoalSoftwareComponentDetails;
(function (GiGoalSoftwareComponentDetails) {
    let HomePolicy;
    (function (HomePolicy) {
        HomePolicy["CreateNew"] = "CREATE_NEW";
        HomePolicy["UseExisting"] = "USE_EXISTING";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        HomePolicy["UnknownValue"] = "UNKNOWN_VALUE";
    })(HomePolicy = GiGoalSoftwareComponentDetails.HomePolicy || (GiGoalSoftwareComponentDetails.HomePolicy = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.GoalSoftwareComponentDetails.getJsonObj(obj))), {
            "goalVersionDetails": obj.goalVersionDetails
                ? model.GiGoalVersionDetails.getJsonObj(obj.goalVersionDetails)
                : undefined
        });
        return jsonObj;
    }
    GiGoalSoftwareComponentDetails.getJsonObj = getJsonObj;
    GiGoalSoftwareComponentDetails.componentType = "GI";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.GoalSoftwareComponentDetails.getDeserializedJsonObj(obj))), {
            "goalVersionDetails": obj.goalVersionDetails
                ? model.GiGoalVersionDetails.getDeserializedJsonObj(obj.goalVersionDetails)
                : undefined
        });
        return jsonObj;
    }
    GiGoalSoftwareComponentDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(GiGoalSoftwareComponentDetails = exports.GiGoalSoftwareComponentDetails || (exports.GiGoalSoftwareComponentDetails = {}));
