// oci-fleetsoftwareupdate/lib/model/guest-os-exadata-release-version-filter.js
"use strict";
/**
 * Exadata Fleet Update service API
 * Use the Exadata Fleet Update service to patch large collections of components directly,
as a single entity, orchestrating the maintenance actions to update all chosen components in the stack in a single cycle.

 * OpenAPI spec version: 20220528
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.GuestOsExadataReleaseVersionFilter = void 0;
const model = __importStar(require("../model"));
var GuestOsExadataReleaseVersionFilter;
(function (GuestOsExadataReleaseVersionFilter) {
    let ExadataReleases;
    (function (ExadataReleases) {
        ExadataReleases["ExaOl5111"] = "EXA_OL_5_11_1";
        ExadataReleases["ExaOl5112"] = "EXA_OL_5_11_2";
        ExadataReleases["ExaOl5121"] = "EXA_OL_5_12_1";
        ExadataReleases["ExaOl6122"] = "EXA_OL_6_12_2";
        ExadataReleases["ExaOl6181"] = "EXA_OL_6_18_1";
        ExadataReleases["ExaOl7191"] = "EXA_OL_7_19_1";
        ExadataReleases["ExaOl7192"] = "EXA_OL_7_19_2";
        ExadataReleases["ExaOl7193"] = "EXA_OL_7_19_3";
        ExadataReleases["ExaOl7201"] = "EXA_OL_7_20_1";
        ExadataReleases["ExaOl7212"] = "EXA_OL_7_21_2";
        ExadataReleases["ExaOl7221"] = "EXA_OL_7_22_1";
        ExadataReleases["ExaOl8231"] = "EXA_OL_8_23_1";
        ExadataReleases["ExaOl8241"] = "EXA_OL_8_24_1";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ExadataReleases["UnknownValue"] = "UNKNOWN_VALUE";
    })(ExadataReleases = GuestOsExadataReleaseVersionFilter.ExadataReleases || (GuestOsExadataReleaseVersionFilter.ExadataReleases = {}));
    function getJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.GuestOsFleetDiscoveryFilter.getJsonObj(obj))), {});
        return jsonObj;
    }
    GuestOsExadataReleaseVersionFilter.getJsonObj = getJsonObj;
    GuestOsExadataReleaseVersionFilter.type = "EXADATA_RELEASE_VERSION";
    function getDeserializedJsonObj(obj, isParentJsonObj) {
        const jsonObj = Object.assign(Object.assign({}, (isParentJsonObj
            ? obj
            : model.GuestOsFleetDiscoveryFilter.getDeserializedJsonObj(obj))), {});
        return jsonObj;
    }
    GuestOsExadataReleaseVersionFilter.getDeserializedJsonObj = getDeserializedJsonObj;
})(GuestOsExadataReleaseVersionFilter = exports.GuestOsExadataReleaseVersionFilter || (exports.GuestOsExadataReleaseVersionFilter = {}));
