// oci-datasafe/lib/request/list-sensitive-columns-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSensitiveColumnsRequest = void 0;
var ListSensitiveColumnsRequest;
(function (ListSensitiveColumnsRequest) {
    let ColumnDataCountFilter;
    (function (ColumnDataCountFilter) {
        ColumnDataCountFilter["ShowAllColumns"] = "SHOW_ALL_COLUMNS";
        ColumnDataCountFilter["ShowColumnsWithData"] = "SHOW_COLUMNS_WITH_DATA";
        ColumnDataCountFilter["ShowColumnsWithoutData"] = "SHOW_COLUMNS_WITHOUT_DATA";
    })(ColumnDataCountFilter = ListSensitiveColumnsRequest.ColumnDataCountFilter || (ListSensitiveColumnsRequest.ColumnDataCountFilter = {}));
    let ObjectType;
    (function (ObjectType) {
        ObjectType["All"] = "ALL";
        ObjectType["Table"] = "TABLE";
        ObjectType["EditioningView"] = "EDITIONING_VIEW";
    })(ObjectType = ListSensitiveColumnsRequest.ObjectType || (ListSensitiveColumnsRequest.ObjectType = {}));
    let Status;
    (function (Status) {
        Status["Valid"] = "VALID";
        Status["Invalid"] = "INVALID";
    })(Status = ListSensitiveColumnsRequest.Status || (ListSensitiveColumnsRequest.Status = {}));
    let RelationType;
    (function (RelationType) {
        RelationType["None"] = "NONE";
        RelationType["AppDefined"] = "APP_DEFINED";
        RelationType["DbDefined"] = "DB_DEFINED";
    })(RelationType = ListSensitiveColumnsRequest.RelationType || (ListSensitiveColumnsRequest.RelationType = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListSensitiveColumnsRequest.SortOrder || (ListSensitiveColumnsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["SchemaName"] = "schemaName";
        SortBy["ObjectName"] = "objectName";
        SortBy["ColumnName"] = "columnName";
        SortBy["DataType"] = "dataType";
        SortBy["ConfidenceLevel"] = "confidenceLevel";
    })(SortBy = ListSensitiveColumnsRequest.SortBy || (ListSensitiveColumnsRequest.SortBy = {}));
})(ListSensitiveColumnsRequest = exports.ListSensitiveColumnsRequest || (exports.ListSensitiveColumnsRequest = {}));
