// oci-datasafe/lib/request/list-checks-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListChecksRequest = void 0;
var ListChecksRequest;
(function (ListChecksRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListChecksRequest.SortOrder || (ListChecksRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Title"] = "title";
        SortBy["Category"] = "category";
    })(SortBy = ListChecksRequest.SortBy || (ListChecksRequest.SortBy = {}));
    let SuggestedSeverity;
    (function (SuggestedSeverity) {
        SuggestedSeverity["High"] = "HIGH";
        SuggestedSeverity["Medium"] = "MEDIUM";
        SuggestedSeverity["Low"] = "LOW";
        SuggestedSeverity["Evaluate"] = "EVALUATE";
        SuggestedSeverity["Advisory"] = "ADVISORY";
        SuggestedSeverity["Pass"] = "PASS";
        SuggestedSeverity["Deferred"] = "DEFERRED";
    })(SuggestedSeverity = ListChecksRequest.SuggestedSeverity || (ListChecksRequest.SuggestedSeverity = {}));
    let ContainsSeverity;
    (function (ContainsSeverity) {
        ContainsSeverity["High"] = "HIGH";
        ContainsSeverity["Medium"] = "MEDIUM";
        ContainsSeverity["Low"] = "LOW";
        ContainsSeverity["Evaluate"] = "EVALUATE";
        ContainsSeverity["Advisory"] = "ADVISORY";
        ContainsSeverity["Pass"] = "PASS";
        ContainsSeverity["Deferred"] = "DEFERRED";
    })(ContainsSeverity = ListChecksRequest.ContainsSeverity || (ListChecksRequest.ContainsSeverity = {}));
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListChecksRequest.AccessLevel || (ListChecksRequest.AccessLevel = {}));
})(ListChecksRequest = exports.ListChecksRequest || (exports.ListChecksRequest = {}));
