// oci-datasafe/lib/request/list-attribute-sets-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAttributeSetsRequest = void 0;
var ListAttributeSetsRequest;
(function (ListAttributeSetsRequest) {
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListAttributeSetsRequest.AccessLevel || (ListAttributeSetsRequest.AccessLevel = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListAttributeSetsRequest.SortOrder || (ListAttributeSetsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["Timecreated"] = "TIMECREATED";
        SortBy["Displayname"] = "DISPLAYNAME";
    })(SortBy = ListAttributeSetsRequest.SortBy || (ListAttributeSetsRequest.SortBy = {}));
    let InUse;
    (function (InUse) {
        InUse["Yes"] = "YES";
        InUse["No"] = "NO";
    })(InUse = ListAttributeSetsRequest.InUse || (ListAttributeSetsRequest.InUse = {}));
})(ListAttributeSetsRequest = exports.ListAttributeSetsRequest || (exports.ListAttributeSetsRequest = {}));
