// oci-datasafe/lib/request/list-target-database-groups-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181201
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListTargetDatabaseGroupsRequest = void 0;
var ListTargetDatabaseGroupsRequest;
(function (ListTargetDatabaseGroupsRequest) {
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
        SortBy["LifecycleState"] = "lifecycleState";
        SortBy["MembershipUpdateTime"] = "membershipUpdateTime";
    })(SortBy = ListTargetDatabaseGroupsRequest.SortBy || (ListTargetDatabaseGroupsRequest.SortBy = {}));
    let AccessLevel;
    (function (AccessLevel) {
        AccessLevel["Restricted"] = "RESTRICTED";
        AccessLevel["Accessible"] = "ACCESSIBLE";
    })(AccessLevel = ListTargetDatabaseGroupsRequest.AccessLevel || (ListTargetDatabaseGroupsRequest.AccessLevel = {}));
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListTargetDatabaseGroupsRequest.SortOrder || (ListTargetDatabaseGroupsRequest.SortOrder = {}));
})(ListTargetDatabaseGroupsRequest = exports.ListTargetDatabaseGroupsRequest || (exports.ListTargetDatabaseGroupsRequest = {}));
