// oci-iot/lib/request/list-digital-twin-models-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20250531
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListDigitalTwinModelsRequest = void 0;
var ListDigitalTwinModelsRequest;
(function (ListDigitalTwinModelsRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListDigitalTwinModelsRequest.SortOrder || (ListDigitalTwinModelsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
    })(SortBy = ListDigitalTwinModelsRequest.SortBy || (ListDigitalTwinModelsRequest.SortBy = {}));
})(ListDigitalTwinModelsRequest = exports.ListDigitalTwinModelsRequest || (exports.ListDigitalTwinModelsRequest = {}));
