// oci-aidocument/lib/model/driving-license-kv-model-sub-type-details.js
"use strict";
/**
 * Document Understanding API
 * Document AI helps customers perform various analysis on their documents. If a customer has lots of documents, they can process them in batch using asynchronous API endpoints.
 * OpenAPI spec version: 20221109
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DrivingLicenseKvModelSubTypeDetails = void 0;
var DrivingLicenseKvModelSubTypeDetails;
(function (DrivingLicenseKvModelSubTypeDetails) {
    let ModelSubType;
    (function (ModelSubType) {
        ModelSubType["Receipt"] = "RECEIPT";
        ModelSubType["Invoice"] = "INVOICE";
        ModelSubType["Passport"] = "PASSPORT";
        ModelSubType["DriverLicense"] = "DRIVER_LICENSE";
        ModelSubType["HealthInsuranceId"] = "HEALTH_INSURANCE_ID";
    })(ModelSubType = DrivingLicenseKvModelSubTypeDetails.ModelSubType || (DrivingLicenseKvModelSubTypeDetails.ModelSubType = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DrivingLicenseKvModelSubTypeDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DrivingLicenseKvModelSubTypeDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(DrivingLicenseKvModelSubTypeDetails = exports.DrivingLicenseKvModelSubTypeDetails || (exports.DrivingLicenseKvModelSubTypeDetails = {}));
