// @aws-sdk/client-transcribe/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "transcribe",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class TranscribeClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultTranscribeHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

let TranscribeServiceException$1 = class TranscribeServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, TranscribeServiceException.prototype);
    }
};

let BadRequestException$1 = class BadRequestException extends TranscribeServiceException$1 {
    name = "BadRequestException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "BadRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BadRequestException.prototype);
        this.Message = opts.Message;
    }
};
let ConflictException$1 = class ConflictException extends TranscribeServiceException$1 {
    name = "ConflictException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.Message = opts.Message;
    }
};
let InternalFailureException$1 = class InternalFailureException extends TranscribeServiceException$1 {
    name = "InternalFailureException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "InternalFailureException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalFailureException.prototype);
        this.Message = opts.Message;
    }
};
let LimitExceededException$1 = class LimitExceededException extends TranscribeServiceException$1 {
    name = "LimitExceededException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LimitExceededException.prototype);
        this.Message = opts.Message;
    }
};
let NotFoundException$1 = class NotFoundException extends TranscribeServiceException$1 {
    name = "NotFoundException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NotFoundException.prototype);
        this.Message = opts.Message;
    }
};

const _ADE = "AllowDeferredExecution";
const _ATR = "AbsoluteTimeRange";
const _BMN = "BaseModelName";
const _BRE = "BadRequestException";
const _C = "Categories";
const _CAJ = "CallAnalyticsJob";
const _CAJD = "CallAnalyticsJobDetails";
const _CAJN = "CallAnalyticsJobName";
const _CAJS = "CallAnalyticsJobStatus";
const _CAJSa = "CallAnalyticsJobSettings";
const _CAJSal = "CallAnalyticsJobSummary";
const _CAJSall = "CallAnalyticsJobSummaries";
const _CASF = "CallAnalyticsSkippedFeature";
const _CASFL = "CallAnalyticsSkippedFeatureList";
const _CCAC = "CreateCallAnalyticsCategory";
const _CCACR = "CreateCallAnalyticsCategoryRequest";
const _CCACRr = "CreateCallAnalyticsCategoryResponse";
const _CD = "ChannelDefinitions";
const _CDU = "ClinicalDocumentUri";
const _CDh = "ChannelDefinition";
const _CE = "ConflictException";
const _CI = "ChannelId";
const _CIT = "ContentIdentificationType";
const _CIh = "ChannelIdentification";
const _CLM = "CreateLanguageModel";
const _CLMR = "CreateLanguageModelRequest";
const _CLMRr = "CreateLanguageModelResponse";
const _CMV = "CreateMedicalVocabulary";
const _CMVR = "CreateMedicalVocabularyRequest";
const _CMVRr = "CreateMedicalVocabularyResponse";
const _CN = "CategoryName";
const _CNGS = "ClinicalNoteGenerationSettings";
const _CP = "CategoryProperties";
const _CPL = "CategoryPropertiesList";
const _CR = "ContentRedaction";
const _CT = "CreationTime";
const _CTo = "CompletionTime";
const _CTr = "CreateTime";
const _CV = "CreateVocabulary";
const _CVF = "CreateVocabularyFilter";
const _CVFR = "CreateVocabularyFilterRequest";
const _CVFRr = "CreateVocabularyFilterResponse";
const _CVR = "CreateVocabularyRequest";
const _CVRr = "CreateVocabularyResponse";
const _DARA = "DataAccessRoleArn";
const _DCAC = "DeleteCallAnalyticsCategory";
const _DCACR = "DeleteCallAnalyticsCategoryRequest";
const _DCACRe = "DeleteCallAnalyticsCategoryResponse";
const _DCAJ = "DeleteCallAnalyticsJob";
const _DCAJR = "DeleteCallAnalyticsJobRequest";
const _DCAJRe = "DeleteCallAnalyticsJobResponse";
const _DIS = "DurationInSeconds";
const _DLM = "DeleteLanguageModel";
const _DLMR = "DeleteLanguageModelRequest";
const _DLMRe = "DescribeLanguageModelRequest";
const _DLMRes = "DescribeLanguageModelResponse";
const _DLMe = "DescribeLanguageModel";
const _DMSJ = "DeleteMedicalScribeJob";
const _DMSJR = "DeleteMedicalScribeJobRequest";
const _DMTJ = "DeleteMedicalTranscriptionJob";
const _DMTJR = "DeleteMedicalTranscriptionJobRequest";
const _DMV = "DeleteMedicalVocabulary";
const _DMVR = "DeleteMedicalVocabularyRequest";
const _DTJ = "DeleteTranscriptionJob";
const _DTJR = "DeleteTranscriptionJobRequest";
const _DU = "DownloadUri";
const _DV = "DeleteVocabulary";
const _DVF = "DeleteVocabularyFilter";
const _DVFR = "DeleteVocabularyFilterRequest";
const _DVR = "DeleteVocabularyRequest";
const _EP = "EndPercentage";
const _ET = "EndTime";
const _F = "First";
const _FR = "FailureReason";
const _Fe = "Feature";
const _Fo = "Formats";
const _GAS = "GenerateAbstractiveSummary";
const _GCAC = "GetCallAnalyticsCategory";
const _GCACR = "GetCallAnalyticsCategoryRequest";
const _GCACRe = "GetCallAnalyticsCategoryResponse";
const _GCAJ = "GetCallAnalyticsJob";
const _GCAJR = "GetCallAnalyticsJobRequest";
const _GCAJRe = "GetCallAnalyticsJobResponse";
const _GMSJ = "GetMedicalScribeJob";
const _GMSJR = "GetMedicalScribeJobRequest";
const _GMSJRe = "GetMedicalScribeJobResponse";
const _GMTJ = "GetMedicalTranscriptionJob";
const _GMTJR = "GetMedicalTranscriptionJobRequest";
const _GMTJRe = "GetMedicalTranscriptionJobResponse";
const _GMV = "GetMedicalVocabulary";
const _GMVR = "GetMedicalVocabularyRequest";
const _GMVRe = "GetMedicalVocabularyResponse";
const _GTJ = "GetTranscriptionJob";
const _GTJR = "GetTranscriptionJobRequest";
const _GTJRe = "GetTranscriptionJobResponse";
const _GV = "GetVocabulary";
const _GVF = "GetVocabularyFilter";
const _GVFR = "GetVocabularyFilterRequest";
const _GVFRe = "GetVocabularyFilterResponse";
const _GVR = "GetVocabularyRequest";
const _GVRe = "GetVocabularyResponse";
const _IDC = "InputDataConfig";
const _IF = "InterruptionFilter";
const _IFE = "InternalFailureException";
const _IL = "IdentifyLanguage";
const _ILS = "IdentifiedLanguageScore";
const _IML = "IdentifyMultipleLanguages";
const _IT = "InputType";
const _JES = "JobExecutionSettings";
const _JNC = "JobNameContains";
const _K = "Key";
const _KMSEC = "KMSEncryptionContext";
const _L = "Last";
const _LC = "LanguageCode";
const _LCAC = "ListCallAnalyticsCategories";
const _LCACR = "ListCallAnalyticsCategoriesRequest";
const _LCACRi = "ListCallAnalyticsCategoriesResponse";
const _LCAJ = "ListCallAnalyticsJobs";
const _LCAJR = "ListCallAnalyticsJobsRequest";
const _LCAJRi = "ListCallAnalyticsJobsResponse";
const _LCI = "LanguageCodeItem";
const _LCL = "LanguageCodeList";
const _LCa = "LanguageCodes";
const _LEE = "LimitExceededException";
const _LIS = "LanguageIdSettings";
const _LISM = "LanguageIdSettingsMap";
const _LLM = "ListLanguageModels";
const _LLMR = "ListLanguageModelsRequest";
const _LLMRi = "ListLanguageModelsResponse";
const _LM = "LanguageModel";
const _LMN = "LanguageModelName";
const _LMSJ = "ListMedicalScribeJobs";
const _LMSJR = "ListMedicalScribeJobsRequest";
const _LMSJRi = "ListMedicalScribeJobsResponse";
const _LMT = "LastModifiedTime";
const _LMTJ = "ListMedicalTranscriptionJobs";
const _LMTJR = "ListMedicalTranscriptionJobsRequest";
const _LMTJRi = "ListMedicalTranscriptionJobsResponse";
const _LMV = "ListMedicalVocabularies";
const _LMVR = "ListMedicalVocabulariesRequest";
const _LMVRi = "ListMedicalVocabulariesResponse";
const _LO = "LanguageOptions";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LTJ = "ListTranscriptionJobs";
const _LTJR = "ListTranscriptionJobsRequest";
const _LTJRi = "ListTranscriptionJobsResponse";
const _LUT = "LastUpdateTime";
const _LV = "ListVocabularies";
const _LVF = "ListVocabularyFilters";
const _LVFR = "ListVocabularyFiltersRequest";
const _LVFRi = "ListVocabularyFiltersResponse";
const _LVR = "ListVocabulariesRequest";
const _LVRi = "ListVocabulariesResponse";
const _M = "Message";
const _MA = "MaxAlternatives";
const _MF = "MediaFormat";
const _MFU = "MediaFileUri";
const _MN = "ModelName";
const _MR = "MaxResults";
const _MS = "ModelStatus";
const _MSC = "MedicalScribeContext";
const _MSCD = "MedicalScribeChannelDefinition";
const _MSCDe = "MedicalScribeChannelDefinitions";
const _MSCP = "MedicalScribeContextProvided";
const _MSJ = "MedicalScribeJob";
const _MSJN = "MedicalScribeJobName";
const _MSJS = "MedicalScribeJobSummaries";
const _MSJSe = "MedicalScribeJobStatus";
const _MSJSed = "MedicalScribeJobSummary";
const _MSL = "MaxSpeakerLabels";
const _MSO = "MedicalScribeOutput";
const _MSPC = "MedicalScribePatientContext";
const _MSRH = "MediaSampleRateHertz";
const _MSS = "MedicalScribeSettings";
const _MSo = "ModelSettings";
const _MT = "MedicalTranscript";
const _MTJ = "MedicalTranscriptionJob";
const _MTJN = "MedicalTranscriptionJobName";
const _MTJS = "MedicalTranscriptionJobSummaries";
const _MTJSe = "MedicalTranscriptionJobSummary";
const _MTS = "MedicalTranscriptionSetting";
const _Me = "Media";
const _Mo = "Models";
const _N = "Negate";
const _NC = "NameContains";
const _NFE = "NotFoundException";
const _NT = "NoteTemplate";
const _NTTF = "NonTalkTimeFilter";
const _NTe = "NextToken";
const _OBN = "OutputBucketName";
const _OEKMSKI = "OutputEncryptionKMSKeyId";
const _OK = "OutputKey";
const _OL = "OutputLocation";
const _OLT = "OutputLocationType";
const _OSI = "OutputStartIndex";
const _P = "Pronouns";
const _PC = "PatientContext";
const _PET = "PiiEntityTypes";
const _PR = "ParticipantRole";
const _Ph = "Phrases";
const _R = "Rules";
const _RA = "ResourceArn";
const _RC = "ReasonCode";
const _RL = "RuleList";
const _RMFU = "RedactedMediaFileUri";
const _RO = "RedactionOutput";
const _RT = "RedactionType";
const _RTFU = "RedactedTranscriptFileUri";
const _RTR = "RelativeTimeRange";
const _Ru = "Rule";
const _S = "Settings";
const _SA = "ShowAlternatives";
const _SCAJ = "StartCallAnalyticsJob";
const _SCAJR = "StartCallAnalyticsJobRequest";
const _SCAJRt = "StartCallAnalyticsJobResponse";
const _SE = "StatusEquals";
const _SEt = "         StatusEquals";
const _SEta = "StateEquals";
const _SF = "SentimentFilter";
const _SFU = "SubtitleFileUris";
const _SMSJ = "StartMedicalScribeJob";
const _SMSJR = "StartMedicalScribeJobRequest";
const _SMSJRt = "StartMedicalScribeJobResponse";
const _SMTJ = "StartMedicalTranscriptionJob";
const _SMTJR = "StartMedicalTranscriptionJobRequest";
const _SMTJRt = "StartMedicalTranscriptionJobResponse";
const _SO = "SubtitlesOutput";
const _SP = "StartPercentage";
const _SSL = "ShowSpeakerLabels";
const _ST = "StartTime";
const _STJ = "StartTranscriptionJob";
const _STJR = "StartTranscriptionJobRequest";
const _STJRt = "StartTranscriptionJobResponse";
const _SU = "S3Uri";
const _Se = "Sentiments";
const _Sk = "Skipped";
const _Sp = "Specialty";
const _St = "Status";
const _Su = "Summarization";
const _Sub = "Subtitles";
const _T = "Transcript";
const _TC = "ToxicityCategories";
const _TD = "ToxicityDetection";
const _TDS = "ToxicityDetectionSettings";
const _TDSU = "TuningDataS3Uri";
const _TF = "TranscriptFilter";
const _TFT = "TranscriptFilterType";
const _TFU = "TranscriptFileUri";
const _TJ = "TranscriptionJob";
const _TJN = "TranscriptionJobName";
const _TJS = "TranscriptionJobSummaries";
const _TJSr = "TranscriptionJobStatus";
const _TJSra = "TranscriptionJobSummary";
const _TK = "TagKeys";
const _TL = "TagList";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _Ta = "Tags";
const _Tag = "Tag";
const _Tar = "Targets";
const _Th = "Threshold";
const _Ty = "Type";
const _UA = "UpgradeAvailability";
const _UCAC = "UpdateCallAnalyticsCategory";
const _UCACR = "UpdateCallAnalyticsCategoryRequest";
const _UCACRp = "UpdateCallAnalyticsCategoryResponse";
const _UMV = "UpdateMedicalVocabulary";
const _UMVR = "UpdateMedicalVocabularyRequest";
const _UMVRp = "UpdateMedicalVocabularyResponse";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _UV = "UpdateVocabulary";
const _UVF = "UpdateVocabularyFilter";
const _UVFR = "UpdateVocabularyFilterRequest";
const _UVFRp = "UpdateVocabularyFilterResponse";
const _UVR = "UpdateVocabularyRequest";
const _UVRp = "UpdateVocabularyResponse";
const _V = "Vocabularies";
const _VF = "VocabularyFilters";
const _VFFU = "VocabularyFilterFileUri";
const _VFI = "VocabularyFilterInfo";
const _VFM = "VocabularyFilterMethod";
const _VFN = "VocabularyFilterName";
const _VFU = "VocabularyFileUri";
const _VI = "VocabularyInfo";
const _VN = "VocabularyName";
const _VS = "VocabularyState";
const _Va = "Value";
const _W = "Words";
const _c = "client";
const _e = "error";
const _h = "http";
const _hE = "httpError";
const _hQ = "httpQuery";
const _s = "server";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.transcribe";
const _tK = "tagKeys";
const n0 = "com.amazonaws.transcribe";
var Pronouns$1 = [0, n0, _P, 8, 0];
var AbsoluteTimeRange = [3, n0, _ATR, 0, [_ST, _ET, _F, _L], [1, 1, 1, 1]];
var BadRequestException = [
    -3,
    n0,
    _BRE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(BadRequestException, BadRequestException$1);
var CallAnalyticsJob = [
    3,
    n0,
    _CAJ,
    0,
    [_CAJN, _CAJS, _CAJD, _LC, _MSRH, _MF, _Me, _T, _ST, _CT, _CTo, _FR, _DARA, _ILS, _S, _CD, _Ta],
    [
        0,
        0,
        () => CallAnalyticsJobDetails,
        0,
        1,
        0,
        () => Media,
        () => Transcript,
        4,
        4,
        4,
        0,
        0,
        1,
        () => CallAnalyticsJobSettings,
        () => ChannelDefinitions,
        () => TagList,
    ],
];
var CallAnalyticsJobDetails = [
    3,
    n0,
    _CAJD,
    0,
    [_Sk],
    [() => CallAnalyticsSkippedFeatureList],
];
var CallAnalyticsJobSettings = [
    3,
    n0,
    _CAJSa,
    0,
    [_VN, _VFN, _VFM, _LMN, _CR, _LO, _LIS, _Su],
    [0, 0, 0, 0, () => ContentRedaction, 64 | 0, () => LanguageIdSettingsMap, () => Summarization],
];
var CallAnalyticsJobSummary = [
    3,
    n0,
    _CAJSal,
    0,
    [_CAJN, _CT, _ST, _CTo, _LC, _CAJS, _CAJD, _FR],
    [0, 4, 4, 4, 0, 0, () => CallAnalyticsJobDetails, 0],
];
var CallAnalyticsSkippedFeature = [3, n0, _CASF, 0, [_Fe, _RC, _M], [0, 0, 0]];
var CategoryProperties = [
    3,
    n0,
    _CP,
    0,
    [_CN, _R, _CTr, _LUT, _Ta, _IT],
    [0, () => RuleList, 4, 4, () => TagList, 0],
];
var ChannelDefinition = [3, n0, _CDh, 0, [_CI, _PR], [1, 0]];
var ClinicalNoteGenerationSettings = [3, n0, _CNGS, 0, [_NT], [0]];
var ConflictException = [
    -3,
    n0,
    _CE,
    {
        [_e]: _c,
        [_hE]: 409,
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(ConflictException, ConflictException$1);
var ContentRedaction = [3, n0, _CR, 0, [_RT, _RO, _PET], [0, 0, 64 | 0]];
var CreateCallAnalyticsCategoryRequest = [
    3,
    n0,
    _CCACR,
    0,
    [_CN, _R, _Ta, _IT],
    [[0, 1], () => RuleList, () => TagList, 0],
];
var CreateCallAnalyticsCategoryResponse = [
    3,
    n0,
    _CCACRr,
    0,
    [_CP],
    [() => CategoryProperties],
];
var CreateLanguageModelRequest = [
    3,
    n0,
    _CLMR,
    0,
    [_LC, _BMN, _MN, _IDC, _Ta],
    [0, 0, [0, 1], () => InputDataConfig, () => TagList],
];
var CreateLanguageModelResponse = [
    3,
    n0,
    _CLMRr,
    0,
    [_LC, _BMN, _MN, _IDC, _MS],
    [0, 0, 0, () => InputDataConfig, 0],
];
var CreateMedicalVocabularyRequest = [
    3,
    n0,
    _CMVR,
    0,
    [_VN, _LC, _VFU, _Ta],
    [[0, 1], 0, 0, () => TagList],
];
var CreateMedicalVocabularyResponse = [
    3,
    n0,
    _CMVRr,
    0,
    [_VN, _LC, _VS, _LMT, _FR],
    [0, 0, 0, 4, 0],
];
var CreateVocabularyFilterRequest = [
    3,
    n0,
    _CVFR,
    0,
    [_VFN, _LC, _W, _VFFU, _Ta, _DARA],
    [[0, 1], 0, 64 | 0, 0, () => TagList, 0],
];
var CreateVocabularyFilterResponse = [3, n0, _CVFRr, 0, [_VFN, _LC, _LMT], [0, 0, 4]];
var CreateVocabularyRequest = [
    3,
    n0,
    _CVR,
    0,
    [_VN, _LC, _Ph, _VFU, _Ta, _DARA],
    [[0, 1], 0, 64 | 0, 0, () => TagList, 0],
];
var CreateVocabularyResponse = [
    3,
    n0,
    _CVRr,
    0,
    [_VN, _LC, _VS, _LMT, _FR],
    [0, 0, 0, 4, 0],
];
var DeleteCallAnalyticsCategoryRequest = [3, n0, _DCACR, 0, [_CN], [[0, 1]]];
var DeleteCallAnalyticsCategoryResponse = [3, n0, _DCACRe, 0, [], []];
var DeleteCallAnalyticsJobRequest = [3, n0, _DCAJR, 0, [_CAJN], [[0, 1]]];
var DeleteCallAnalyticsJobResponse = [3, n0, _DCAJRe, 0, [], []];
var DeleteLanguageModelRequest = [3, n0, _DLMR, 0, [_MN], [[0, 1]]];
var DeleteMedicalScribeJobRequest = [3, n0, _DMSJR, 0, [_MSJN], [[0, 1]]];
var DeleteMedicalTranscriptionJobRequest = [3, n0, _DMTJR, 0, [_MTJN], [[0, 1]]];
var DeleteMedicalVocabularyRequest = [3, n0, _DMVR, 0, [_VN], [[0, 1]]];
var DeleteTranscriptionJobRequest = [3, n0, _DTJR, 0, [_TJN], [[0, 1]]];
var DeleteVocabularyFilterRequest = [3, n0, _DVFR, 0, [_VFN], [[0, 1]]];
var DeleteVocabularyRequest = [3, n0, _DVR, 0, [_VN], [[0, 1]]];
var DescribeLanguageModelRequest = [3, n0, _DLMRe, 0, [_MN], [[0, 1]]];
var DescribeLanguageModelResponse = [3, n0, _DLMRes, 0, [_LM], [() => LanguageModel]];
var GetCallAnalyticsCategoryRequest = [3, n0, _GCACR, 0, [_CN], [[0, 1]]];
var GetCallAnalyticsCategoryResponse = [
    3,
    n0,
    _GCACRe,
    0,
    [_CP],
    [() => CategoryProperties],
];
var GetCallAnalyticsJobRequest = [3, n0, _GCAJR, 0, [_CAJN], [[0, 1]]];
var GetCallAnalyticsJobResponse = [3, n0, _GCAJRe, 0, [_CAJ], [() => CallAnalyticsJob]];
var GetMedicalScribeJobRequest = [3, n0, _GMSJR, 0, [_MSJN], [[0, 1]]];
var GetMedicalScribeJobResponse = [3, n0, _GMSJRe, 0, [_MSJ], [() => MedicalScribeJob]];
var GetMedicalTranscriptionJobRequest = [3, n0, _GMTJR, 0, [_MTJN], [[0, 1]]];
var GetMedicalTranscriptionJobResponse = [
    3,
    n0,
    _GMTJRe,
    0,
    [_MTJ],
    [() => MedicalTranscriptionJob],
];
var GetMedicalVocabularyRequest = [3, n0, _GMVR, 0, [_VN], [[0, 1]]];
var GetMedicalVocabularyResponse = [
    3,
    n0,
    _GMVRe,
    0,
    [_VN, _LC, _VS, _LMT, _FR, _DU],
    [0, 0, 0, 4, 0, 0],
];
var GetTranscriptionJobRequest = [3, n0, _GTJR, 0, [_TJN], [[0, 1]]];
var GetTranscriptionJobResponse = [3, n0, _GTJRe, 0, [_TJ], [() => TranscriptionJob]];
var GetVocabularyFilterRequest = [3, n0, _GVFR, 0, [_VFN], [[0, 1]]];
var GetVocabularyFilterResponse = [
    3,
    n0,
    _GVFRe,
    0,
    [_VFN, _LC, _LMT, _DU],
    [0, 0, 4, 0],
];
var GetVocabularyRequest = [3, n0, _GVR, 0, [_VN], [[0, 1]]];
var GetVocabularyResponse = [
    3,
    n0,
    _GVRe,
    0,
    [_VN, _LC, _VS, _LMT, _FR, _DU],
    [0, 0, 0, 4, 0, 0],
];
var InputDataConfig = [3, n0, _IDC, 0, [_SU, _TDSU, _DARA], [0, 0, 0]];
var InternalFailureException = [
    -3,
    n0,
    _IFE,
    {
        [_e]: _s,
        [_hE]: 500,
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(InternalFailureException, InternalFailureException$1);
var InterruptionFilter = [
    3,
    n0,
    _IF,
    0,
    [_Th, _PR, _ATR, _RTR, _N],
    [1, 0, () => AbsoluteTimeRange, () => RelativeTimeRange, 2],
];
var JobExecutionSettings = [3, n0, _JES, 0, [_ADE, _DARA], [2, 0]];
var LanguageCodeItem = [3, n0, _LCI, 0, [_LC, _DIS], [0, 1]];
var LanguageIdSettings = [3, n0, _LIS, 0, [_VN, _VFN, _LMN], [0, 0, 0]];
var LanguageModel = [
    3,
    n0,
    _LM,
    0,
    [_MN, _CTr, _LMT, _LC, _BMN, _MS, _UA, _FR, _IDC],
    [0, 4, 4, 0, 0, 0, 2, 0, () => InputDataConfig],
];
var LimitExceededException = [
    -3,
    n0,
    _LEE,
    {
        [_e]: _c,
        [_hE]: 429,
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(LimitExceededException, LimitExceededException$1);
var ListCallAnalyticsCategoriesRequest = [
    3,
    n0,
    _LCACR,
    0,
    [_NTe, _MR],
    [
        [
            0,
            {
                [_hQ]: _NTe,
            },
        ],
        [
            1,
            {
                [_hQ]: _MR,
            },
        ],
    ],
];
var ListCallAnalyticsCategoriesResponse = [
    3,
    n0,
    _LCACRi,
    0,
    [_NTe, _C],
    [0, () => CategoryPropertiesList],
];
var ListCallAnalyticsJobsRequest = [
    3,
    n0,
    _LCAJR,
    0,
    [_St, _JNC, _NTe, _MR],
    [
        [
            0,
            {
                [_hQ]: _St,
            },
        ],
        [
            0,
            {
                [_hQ]: _JNC,
            },
        ],
        [
            0,
            {
                [_hQ]: _NTe,
            },
        ],
        [
            1,
            {
                [_hQ]: _MR,
            },
        ],
    ],
];
var ListCallAnalyticsJobsResponse = [
    3,
    n0,
    _LCAJRi,
    0,
    [_St, _NTe, _CAJSall],
    [0, 0, () => CallAnalyticsJobSummaries],
];
var ListLanguageModelsRequest = [
    3,
    n0,
    _LLMR,
    0,
    [_SE, _NC, _NTe, _MR],
    [
        [
            0,
            {
                [_hQ]: _SEt,
            },
        ],
        [
            0,
            {
                [_hQ]: _NC,
            },
        ],
        [
            0,
            {
                [_hQ]: _NTe,
            },
        ],
        [
            1,
            {
                [_hQ]: _MR,
            },
        ],
    ],
];
var ListLanguageModelsResponse = [3, n0, _LLMRi, 0, [_NTe, _Mo], [0, () => Models]];
var ListMedicalScribeJobsRequest = [
    3,
    n0,
    _LMSJR,
    0,
    [_St, _JNC, _NTe, _MR],
    [
        [
            0,
            {
                [_hQ]: _St,
            },
        ],
        [
            0,
            {
                [_hQ]: _JNC,
            },
        ],
        [
            0,
            {
                [_hQ]: _NTe,
            },
        ],
        [
            1,
            {
                [_hQ]: _MR,
            },
        ],
    ],
];
var ListMedicalScribeJobsResponse = [
    3,
    n0,
    _LMSJRi,
    0,
    [_St, _NTe, _MSJS],
    [0, 0, () => MedicalScribeJobSummaries],
];
var ListMedicalTranscriptionJobsRequest = [
    3,
    n0,
    _LMTJR,
    0,
    [_St, _JNC, _NTe, _MR],
    [
        [
            0,
            {
                [_hQ]: _St,
            },
        ],
        [
            0,
            {
                [_hQ]: _JNC,
            },
        ],
        [
            0,
            {
                [_hQ]: _NTe,
            },
        ],
        [
            1,
            {
                [_hQ]: _MR,
            },
        ],
    ],
];
var ListMedicalTranscriptionJobsResponse = [
    3,
    n0,
    _LMTJRi,
    0,
    [_St, _NTe, _MTJS],
    [0, 0, () => MedicalTranscriptionJobSummaries],
];
var ListMedicalVocabulariesRequest = [
    3,
    n0,
    _LMVR,
    0,
    [_NTe, _MR, _SEta, _NC],
    [
        [
            0,
            {
                [_hQ]: _NTe,
            },
        ],
        [
            1,
            {
                [_hQ]: _MR,
            },
        ],
        [
            0,
            {
                [_hQ]: _SEta,
            },
        ],
        [
            0,
            {
                [_hQ]: _NC,
            },
        ],
    ],
];
var ListMedicalVocabulariesResponse = [
    3,
    n0,
    _LMVRi,
    0,
    [_St, _NTe, _V],
    [0, 0, () => Vocabularies],
];
var ListTagsForResourceRequest = [3, n0, _LTFRR, 0, [_RA], [[0, 1]]];
var ListTagsForResourceResponse = [3, n0, _LTFRRi, 0, [_RA, _Ta], [0, () => TagList]];
var ListTranscriptionJobsRequest = [
    3,
    n0,
    _LTJR,
    0,
    [_St, _JNC, _NTe, _MR],
    [
        [
            0,
            {
                [_hQ]: _St,
            },
        ],
        [
            0,
            {
                [_hQ]: _JNC,
            },
        ],
        [
            0,
            {
                [_hQ]: _NTe,
            },
        ],
        [
            1,
            {
                [_hQ]: _MR,
            },
        ],
    ],
];
var ListTranscriptionJobsResponse = [
    3,
    n0,
    _LTJRi,
    0,
    [_St, _NTe, _TJS],
    [0, 0, () => TranscriptionJobSummaries],
];
var ListVocabulariesRequest = [
    3,
    n0,
    _LVR,
    0,
    [_NTe, _MR, _SEta, _NC],
    [
        [
            0,
            {
                [_hQ]: _NTe,
            },
        ],
        [
            1,
            {
                [_hQ]: _MR,
            },
        ],
        [
            0,
            {
                [_hQ]: _SEta,
            },
        ],
        [
            0,
            {
                [_hQ]: _NC,
            },
        ],
    ],
];
var ListVocabulariesResponse = [
    3,
    n0,
    _LVRi,
    0,
    [_St, _NTe, _V],
    [0, 0, () => Vocabularies],
];
var ListVocabularyFiltersRequest = [
    3,
    n0,
    _LVFR,
    0,
    [_NTe, _MR, _NC],
    [
        [
            0,
            {
                [_hQ]: _NTe,
            },
        ],
        [
            1,
            {
                [_hQ]: _MR,
            },
        ],
        [
            0,
            {
                [_hQ]: _NC,
            },
        ],
    ],
];
var ListVocabularyFiltersResponse = [
    3,
    n0,
    _LVFRi,
    0,
    [_NTe, _VF],
    [0, () => VocabularyFilters],
];
var Media = [3, n0, _Me, 0, [_MFU, _RMFU], [0, 0]];
var MedicalScribeChannelDefinition = [3, n0, _MSCD, 0, [_CI, _PR], [1, 0]];
var MedicalScribeContext = [
    3,
    n0,
    _MSC,
    0,
    [_PC],
    [[() => MedicalScribePatientContext, 0]],
];
var MedicalScribeJob = [
    3,
    n0,
    _MSJ,
    0,
    [_MSJN, _MSJSe, _LC, _Me, _MSO, _ST, _CT, _CTo, _FR, _S, _DARA, _CD, _MSCP, _Ta],
    [
        0,
        0,
        0,
        () => Media,
        () => MedicalScribeOutput,
        4,
        4,
        4,
        0,
        () => MedicalScribeSettings,
        0,
        () => MedicalScribeChannelDefinitions,
        2,
        () => TagList,
    ],
];
var MedicalScribeJobSummary = [
    3,
    n0,
    _MSJSed,
    0,
    [_MSJN, _CT, _ST, _CTo, _LC, _MSJSe, _FR],
    [0, 4, 4, 4, 0, 0, 0],
];
var MedicalScribeOutput = [3, n0, _MSO, 0, [_TFU, _CDU], [0, 0]];
var MedicalScribePatientContext = [3, n0, _MSPC, 0, [_P], [[() => Pronouns$1, 0]]];
var MedicalScribeSettings = [
    3,
    n0,
    _MSS,
    0,
    [_SSL, _MSL, _CIh, _VN, _VFN, _VFM, _CNGS],
    [2, 1, 2, 0, 0, 0, () => ClinicalNoteGenerationSettings],
];
var MedicalTranscript = [3, n0, _MT, 0, [_TFU], [0]];
var MedicalTranscriptionJob = [
    3,
    n0,
    _MTJ,
    0,
    [_MTJN, _TJSr, _LC, _MSRH, _MF, _Me, _T, _ST, _CT, _CTo, _FR, _S, _CIT, _Sp, _Ty, _Ta],
    [
        0,
        0,
        0,
        1,
        0,
        () => Media,
        () => MedicalTranscript,
        4,
        4,
        4,
        0,
        () => MedicalTranscriptionSetting,
        0,
        0,
        0,
        () => TagList,
    ],
];
var MedicalTranscriptionJobSummary = [
    3,
    n0,
    _MTJSe,
    0,
    [_MTJN, _CT, _ST, _CTo, _LC, _TJSr, _FR, _OLT, _Sp, _CIT, _Ty],
    [0, 4, 4, 4, 0, 0, 0, 0, 0, 0, 0],
];
var MedicalTranscriptionSetting = [
    3,
    n0,
    _MTS,
    0,
    [_SSL, _MSL, _CIh, _SA, _MA, _VN],
    [2, 1, 2, 2, 1, 0],
];
var ModelSettings = [3, n0, _MSo, 0, [_LMN], [0]];
var NonTalkTimeFilter = [
    3,
    n0,
    _NTTF,
    0,
    [_Th, _ATR, _RTR, _N],
    [1, () => AbsoluteTimeRange, () => RelativeTimeRange, 2],
];
var NotFoundException = [
    -3,
    n0,
    _NFE,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_M],
    [0],
];
schema.TypeRegistry.for(n0).registerError(NotFoundException, NotFoundException$1);
var RelativeTimeRange = [3, n0, _RTR, 0, [_SP, _EP, _F, _L], [1, 1, 1, 1]];
var SentimentFilter = [
    3,
    n0,
    _SF,
    0,
    [_Se, _ATR, _RTR, _PR, _N],
    [64 | 0, () => AbsoluteTimeRange, () => RelativeTimeRange, 0, 2],
];
var Settings = [
    3,
    n0,
    _S,
    0,
    [_VN, _SSL, _MSL, _CIh, _SA, _MA, _VFN, _VFM],
    [0, 2, 1, 2, 2, 1, 0, 0],
];
var StartCallAnalyticsJobRequest = [
    3,
    n0,
    _SCAJR,
    0,
    [_CAJN, _Me, _OL, _OEKMSKI, _DARA, _S, _Ta, _CD],
    [[0, 1], () => Media, 0, 0, 0, () => CallAnalyticsJobSettings, () => TagList, () => ChannelDefinitions],
];
var StartCallAnalyticsJobResponse = [3, n0, _SCAJRt, 0, [_CAJ], [() => CallAnalyticsJob]];
var StartMedicalScribeJobRequest = [
    3,
    n0,
    _SMSJR,
    0,
    [_MSJN, _Me, _OBN, _OEKMSKI, _KMSEC, _DARA, _S, _CD, _Ta, _MSC],
    [
        [0, 1],
        () => Media,
        0,
        0,
        128 | 0,
        0,
        () => MedicalScribeSettings,
        () => MedicalScribeChannelDefinitions,
        () => TagList,
        [() => MedicalScribeContext, 0],
    ],
];
var StartMedicalScribeJobResponse = [3, n0, _SMSJRt, 0, [_MSJ], [() => MedicalScribeJob]];
var StartMedicalTranscriptionJobRequest = [
    3,
    n0,
    _SMTJR,
    0,
    [_MTJN, _LC, _MSRH, _MF, _Me, _OBN, _OK, _OEKMSKI, _KMSEC, _S, _CIT, _Sp, _Ty, _Ta],
    [[0, 1], 0, 1, 0, () => Media, 0, 0, 0, 128 | 0, () => MedicalTranscriptionSetting, 0, 0, 0, () => TagList],
];
var StartMedicalTranscriptionJobResponse = [
    3,
    n0,
    _SMTJRt,
    0,
    [_MTJ],
    [() => MedicalTranscriptionJob],
];
var StartTranscriptionJobRequest = [
    3,
    n0,
    _STJR,
    0,
    [_TJN, _LC, _MSRH, _MF, _Me, _OBN, _OK, _OEKMSKI, _KMSEC, _S, _MSo, _JES, _CR, _IL, _IML, _LO, _Sub, _Ta, _LIS, _TD],
    [
        [0, 1],
        0,
        1,
        0,
        () => Media,
        0,
        0,
        0,
        128 | 0,
        () => Settings,
        () => ModelSettings,
        () => JobExecutionSettings,
        () => ContentRedaction,
        2,
        2,
        64 | 0,
        () => Subtitles,
        () => TagList,
        () => LanguageIdSettingsMap,
        () => ToxicityDetection,
    ],
];
var StartTranscriptionJobResponse = [3, n0, _STJRt, 0, [_TJ], [() => TranscriptionJob]];
var Subtitles = [3, n0, _Sub, 0, [_Fo, _OSI], [64 | 0, 1]];
var SubtitlesOutput = [3, n0, _SO, 0, [_Fo, _SFU, _OSI], [64 | 0, 64 | 0, 1]];
var Summarization = [3, n0, _Su, 0, [_GAS], [2]];
var Tag = [3, n0, _Tag, 0, [_K, _Va], [0, 0]];
var TagResourceRequest = [3, n0, _TRR, 0, [_RA, _Ta], [[0, 1], () => TagList]];
var TagResourceResponse = [3, n0, _TRRa, 0, [], []];
var ToxicityDetectionSettings = [3, n0, _TDS, 0, [_TC], [64 | 0]];
var Transcript = [3, n0, _T, 0, [_TFU, _RTFU], [0, 0]];
var TranscriptFilter = [
    3,
    n0,
    _TF,
    0,
    [_TFT, _ATR, _RTR, _PR, _N, _Tar],
    [0, () => AbsoluteTimeRange, () => RelativeTimeRange, 0, 2, 64 | 0],
];
var TranscriptionJob = [
    3,
    n0,
    _TJ,
    0,
    [
        _TJN,
        _TJSr,
        _LC,
        _MSRH,
        _MF,
        _Me,
        _T,
        _ST,
        _CT,
        _CTo,
        _FR,
        _S,
        _MSo,
        _JES,
        _CR,
        _IL,
        _IML,
        _LO,
        _ILS,
        _LCa,
        _Ta,
        _Sub,
        _LIS,
        _TD,
    ],
    [
        0,
        0,
        0,
        1,
        0,
        () => Media,
        () => Transcript,
        4,
        4,
        4,
        0,
        () => Settings,
        () => ModelSettings,
        () => JobExecutionSettings,
        () => ContentRedaction,
        2,
        2,
        64 | 0,
        1,
        () => LanguageCodeList,
        () => TagList,
        () => SubtitlesOutput,
        () => LanguageIdSettingsMap,
        () => ToxicityDetection,
    ],
];
var TranscriptionJobSummary = [
    3,
    n0,
    _TJSra,
    0,
    [_TJN, _CT, _ST, _CTo, _LC, _TJSr, _FR, _OLT, _CR, _MSo, _IL, _IML, _ILS, _LCa, _TD],
    [
        0,
        4,
        4,
        4,
        0,
        0,
        0,
        0,
        () => ContentRedaction,
        () => ModelSettings,
        2,
        2,
        1,
        () => LanguageCodeList,
        () => ToxicityDetection,
    ],
];
var UntagResourceRequest = [
    3,
    n0,
    _URR,
    0,
    [_RA, _TK],
    [
        [0, 1],
        [
            64 | 0,
            {
                [_hQ]: _tK,
            },
        ],
    ],
];
var UntagResourceResponse = [3, n0, _URRn, 0, [], []];
var UpdateCallAnalyticsCategoryRequest = [
    3,
    n0,
    _UCACR,
    0,
    [_CN, _R, _IT],
    [[0, 1], () => RuleList, 0],
];
var UpdateCallAnalyticsCategoryResponse = [
    3,
    n0,
    _UCACRp,
    0,
    [_CP],
    [() => CategoryProperties],
];
var UpdateMedicalVocabularyRequest = [3, n0, _UMVR, 0, [_VN, _LC, _VFU], [[0, 1], 0, 0]];
var UpdateMedicalVocabularyResponse = [
    3,
    n0,
    _UMVRp,
    0,
    [_VN, _LC, _LMT, _VS],
    [0, 0, 4, 0],
];
var UpdateVocabularyFilterRequest = [
    3,
    n0,
    _UVFR,
    0,
    [_VFN, _W, _VFFU, _DARA],
    [[0, 1], 64 | 0, 0, 0],
];
var UpdateVocabularyFilterResponse = [3, n0, _UVFRp, 0, [_VFN, _LC, _LMT], [0, 0, 4]];
var UpdateVocabularyRequest = [
    3,
    n0,
    _UVR,
    0,
    [_VN, _LC, _Ph, _VFU, _DARA],
    [[0, 1], 0, 64 | 0, 0, 0],
];
var UpdateVocabularyResponse = [3, n0, _UVRp, 0, [_VN, _LC, _LMT, _VS], [0, 0, 4, 0]];
var VocabularyFilterInfo = [3, n0, _VFI, 0, [_VFN, _LC, _LMT], [0, 0, 4]];
var VocabularyInfo = [3, n0, _VI, 0, [_VN, _LC, _LMT, _VS], [0, 0, 4, 0]];
var __Unit = "unit";
var TranscribeServiceException = [-3, _sm, "TranscribeServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(TranscribeServiceException, TranscribeServiceException$1);
var CallAnalyticsJobSummaries = [1, n0, _CAJSall, 0, () => CallAnalyticsJobSummary];
var CallAnalyticsSkippedFeatureList = [1, n0, _CASFL, 0, () => CallAnalyticsSkippedFeature];
var CategoryPropertiesList = [1, n0, _CPL, 0, () => CategoryProperties];
var ChannelDefinitions = [1, n0, _CD, 0, () => ChannelDefinition];
var LanguageCodeList = [1, n0, _LCL, 0, () => LanguageCodeItem];
var MedicalScribeChannelDefinitions = [1, n0, _MSCDe, 0, () => MedicalScribeChannelDefinition];
var MedicalScribeJobSummaries = [1, n0, _MSJS, 0, () => MedicalScribeJobSummary];
var MedicalTranscriptionJobSummaries = [1, n0, _MTJS, 0, () => MedicalTranscriptionJobSummary];
var Models = [1, n0, _Mo, 0, () => LanguageModel];
var RuleList = [1, n0, _RL, 0, () => Rule];
var TagList = [1, n0, _TL, 0, () => Tag];
var ToxicityDetection = [1, n0, _TD, 0, () => ToxicityDetectionSettings];
var TranscriptionJobSummaries = [1, n0, _TJS, 0, () => TranscriptionJobSummary];
var Vocabularies = [1, n0, _V, 0, () => VocabularyInfo];
var VocabularyFilters = [1, n0, _VF, 0, () => VocabularyFilterInfo];
var LanguageIdSettingsMap = [2, n0, _LISM, 0, 0, () => LanguageIdSettings];
var Rule = [
    3,
    n0,
    _Ru,
    0,
    [_NTTF, _IF, _TF, _SF],
    [() => NonTalkTimeFilter, () => InterruptionFilter, () => TranscriptFilter, () => SentimentFilter],
];
var CreateCallAnalyticsCategory = [
    9,
    n0,
    _CCAC,
    {
        [_h]: ["PUT", "/callanalyticscategories/{CategoryName}", 200],
    },
    () => CreateCallAnalyticsCategoryRequest,
    () => CreateCallAnalyticsCategoryResponse,
];
var CreateLanguageModel = [
    9,
    n0,
    _CLM,
    {
        [_h]: ["PUT", "/languagemodels/{ModelName}", 200],
    },
    () => CreateLanguageModelRequest,
    () => CreateLanguageModelResponse,
];
var CreateMedicalVocabulary = [
    9,
    n0,
    _CMV,
    {
        [_h]: ["PUT", "/medicalvocabularies/{VocabularyName}", 200],
    },
    () => CreateMedicalVocabularyRequest,
    () => CreateMedicalVocabularyResponse,
];
var CreateVocabulary = [
    9,
    n0,
    _CV,
    {
        [_h]: ["PUT", "/vocabularies/{VocabularyName}", 200],
    },
    () => CreateVocabularyRequest,
    () => CreateVocabularyResponse,
];
var CreateVocabularyFilter = [
    9,
    n0,
    _CVF,
    {
        [_h]: ["POST", "/vocabularyFilters/{VocabularyFilterName}", 201],
    },
    () => CreateVocabularyFilterRequest,
    () => CreateVocabularyFilterResponse,
];
var DeleteCallAnalyticsCategory = [
    9,
    n0,
    _DCAC,
    {
        [_h]: ["DELETE", "/callanalyticscategories/{CategoryName}", 204],
    },
    () => DeleteCallAnalyticsCategoryRequest,
    () => DeleteCallAnalyticsCategoryResponse,
];
var DeleteCallAnalyticsJob = [
    9,
    n0,
    _DCAJ,
    {
        [_h]: ["DELETE", "/callanalyticsjobs/{CallAnalyticsJobName}", 204],
    },
    () => DeleteCallAnalyticsJobRequest,
    () => DeleteCallAnalyticsJobResponse,
];
var DeleteLanguageModel = [
    9,
    n0,
    _DLM,
    {
        [_h]: ["DELETE", "/languagemodels/{ModelName}", 204],
    },
    () => DeleteLanguageModelRequest,
    () => __Unit,
];
var DeleteMedicalScribeJob = [
    9,
    n0,
    _DMSJ,
    {
        [_h]: ["DELETE", "/medicalscribejobs/{MedicalScribeJobName}", 204],
    },
    () => DeleteMedicalScribeJobRequest,
    () => __Unit,
];
var DeleteMedicalTranscriptionJob = [
    9,
    n0,
    _DMTJ,
    {
        [_h]: ["DELETE", "/medicaltranscriptionjobs/{MedicalTranscriptionJobName}", 204],
    },
    () => DeleteMedicalTranscriptionJobRequest,
    () => __Unit,
];
var DeleteMedicalVocabulary = [
    9,
    n0,
    _DMV,
    {
        [_h]: ["DELETE", "/medicalvocabularies/{VocabularyName}", 204],
    },
    () => DeleteMedicalVocabularyRequest,
    () => __Unit,
];
var DeleteTranscriptionJob = [
    9,
    n0,
    _DTJ,
    {
        [_h]: ["DELETE", "/transcriptionjobs/{TranscriptionJobName}", 204],
    },
    () => DeleteTranscriptionJobRequest,
    () => __Unit,
];
var DeleteVocabulary = [
    9,
    n0,
    _DV,
    {
        [_h]: ["DELETE", "/vocabularies/{VocabularyName}", 204],
    },
    () => DeleteVocabularyRequest,
    () => __Unit,
];
var DeleteVocabularyFilter = [
    9,
    n0,
    _DVF,
    {
        [_h]: ["DELETE", "/vocabularyFilters/{VocabularyFilterName}", 204],
    },
    () => DeleteVocabularyFilterRequest,
    () => __Unit,
];
var DescribeLanguageModel = [
    9,
    n0,
    _DLMe,
    {
        [_h]: ["GET", "/languagemodels/{ModelName}", 200],
    },
    () => DescribeLanguageModelRequest,
    () => DescribeLanguageModelResponse,
];
var GetCallAnalyticsCategory = [
    9,
    n0,
    _GCAC,
    {
        [_h]: ["GET", "/callanalyticscategories/{CategoryName}", 200],
    },
    () => GetCallAnalyticsCategoryRequest,
    () => GetCallAnalyticsCategoryResponse,
];
var GetCallAnalyticsJob = [
    9,
    n0,
    _GCAJ,
    {
        [_h]: ["GET", "/callanalyticsjobs/{CallAnalyticsJobName}", 200],
    },
    () => GetCallAnalyticsJobRequest,
    () => GetCallAnalyticsJobResponse,
];
var GetMedicalScribeJob = [
    9,
    n0,
    _GMSJ,
    {
        [_h]: ["GET", "/medicalscribejobs/{MedicalScribeJobName}", 200],
    },
    () => GetMedicalScribeJobRequest,
    () => GetMedicalScribeJobResponse,
];
var GetMedicalTranscriptionJob = [
    9,
    n0,
    _GMTJ,
    {
        [_h]: ["GET", "/medicaltranscriptionjobs/{MedicalTranscriptionJobName}", 200],
    },
    () => GetMedicalTranscriptionJobRequest,
    () => GetMedicalTranscriptionJobResponse,
];
var GetMedicalVocabulary = [
    9,
    n0,
    _GMV,
    {
        [_h]: ["GET", "/medicalvocabularies/{VocabularyName}", 200],
    },
    () => GetMedicalVocabularyRequest,
    () => GetMedicalVocabularyResponse,
];
var GetTranscriptionJob = [
    9,
    n0,
    _GTJ,
    {
        [_h]: ["GET", "/transcriptionjobs/{TranscriptionJobName}", 200],
    },
    () => GetTranscriptionJobRequest,
    () => GetTranscriptionJobResponse,
];
var GetVocabulary = [
    9,
    n0,
    _GV,
    {
        [_h]: ["GET", "/vocabularies/{VocabularyName}", 200],
    },
    () => GetVocabularyRequest,
    () => GetVocabularyResponse,
];
var GetVocabularyFilter = [
    9,
    n0,
    _GVF,
    {
        [_h]: ["GET", "/vocabularyFilters/{VocabularyFilterName}", 200],
    },
    () => GetVocabularyFilterRequest,
    () => GetVocabularyFilterResponse,
];
var ListCallAnalyticsCategories = [
    9,
    n0,
    _LCAC,
    {
        [_h]: ["GET", "/callanalyticscategories", 200],
    },
    () => ListCallAnalyticsCategoriesRequest,
    () => ListCallAnalyticsCategoriesResponse,
];
var ListCallAnalyticsJobs = [
    9,
    n0,
    _LCAJ,
    {
        [_h]: ["GET", "/callanalyticsjobs", 200],
    },
    () => ListCallAnalyticsJobsRequest,
    () => ListCallAnalyticsJobsResponse,
];
var ListLanguageModels = [
    9,
    n0,
    _LLM,
    {
        [_h]: ["GET", "/languagemodels", 200],
    },
    () => ListLanguageModelsRequest,
    () => ListLanguageModelsResponse,
];
var ListMedicalScribeJobs = [
    9,
    n0,
    _LMSJ,
    {
        [_h]: ["GET", "/medicalscribejobs", 200],
    },
    () => ListMedicalScribeJobsRequest,
    () => ListMedicalScribeJobsResponse,
];
var ListMedicalTranscriptionJobs = [
    9,
    n0,
    _LMTJ,
    {
        [_h]: ["GET", "/medicaltranscriptionjobs", 200],
    },
    () => ListMedicalTranscriptionJobsRequest,
    () => ListMedicalTranscriptionJobsResponse,
];
var ListMedicalVocabularies = [
    9,
    n0,
    _LMV,
    {
        [_h]: ["GET", "/medicalvocabularies", 200],
    },
    () => ListMedicalVocabulariesRequest,
    () => ListMedicalVocabulariesResponse,
];
var ListTagsForResource = [
    9,
    n0,
    _LTFR,
    {
        [_h]: ["GET", "/tags/{ResourceArn}", 200],
    },
    () => ListTagsForResourceRequest,
    () => ListTagsForResourceResponse,
];
var ListTranscriptionJobs = [
    9,
    n0,
    _LTJ,
    {
        [_h]: ["GET", "/transcriptionjobs", 200],
    },
    () => ListTranscriptionJobsRequest,
    () => ListTranscriptionJobsResponse,
];
var ListVocabularies = [
    9,
    n0,
    _LV,
    {
        [_h]: ["GET", "/vocabularies", 200],
    },
    () => ListVocabulariesRequest,
    () => ListVocabulariesResponse,
];
var ListVocabularyFilters = [
    9,
    n0,
    _LVF,
    {
        [_h]: ["GET", "/vocabularyFilters", 200],
    },
    () => ListVocabularyFiltersRequest,
    () => ListVocabularyFiltersResponse,
];
var StartCallAnalyticsJob = [
    9,
    n0,
    _SCAJ,
    {
        [_h]: ["PUT", "/callanalyticsjobs/{CallAnalyticsJobName}", 200],
    },
    () => StartCallAnalyticsJobRequest,
    () => StartCallAnalyticsJobResponse,
];
var StartMedicalScribeJob = [
    9,
    n0,
    _SMSJ,
    {
        [_h]: ["PUT", "/medicalscribejobs/{MedicalScribeJobName}", 200],
    },
    () => StartMedicalScribeJobRequest,
    () => StartMedicalScribeJobResponse,
];
var StartMedicalTranscriptionJob = [
    9,
    n0,
    _SMTJ,
    {
        [_h]: ["PUT", "/medicaltranscriptionjobs/{MedicalTranscriptionJobName}", 200],
    },
    () => StartMedicalTranscriptionJobRequest,
    () => StartMedicalTranscriptionJobResponse,
];
var StartTranscriptionJob = [
    9,
    n0,
    _STJ,
    {
        [_h]: ["PUT", "/transcriptionjobs/{TranscriptionJobName}", 200],
    },
    () => StartTranscriptionJobRequest,
    () => StartTranscriptionJobResponse,
];
var TagResource = [
    9,
    n0,
    _TR,
    {
        [_h]: ["PUT", "/tags/{ResourceArn}", 200],
    },
    () => TagResourceRequest,
    () => TagResourceResponse,
];
var UntagResource = [
    9,
    n0,
    _UR,
    {
        [_h]: ["DELETE", "/tags/{ResourceArn}", 204],
    },
    () => UntagResourceRequest,
    () => UntagResourceResponse,
];
var UpdateCallAnalyticsCategory = [
    9,
    n0,
    _UCAC,
    {
        [_h]: ["PATCH", "/callanalyticscategories/{CategoryName}", 200],
    },
    () => UpdateCallAnalyticsCategoryRequest,
    () => UpdateCallAnalyticsCategoryResponse,
];
var UpdateMedicalVocabulary = [
    9,
    n0,
    _UMV,
    {
        [_h]: ["PATCH", "/medicalvocabularies/{VocabularyName}", 200],
    },
    () => UpdateMedicalVocabularyRequest,
    () => UpdateMedicalVocabularyResponse,
];
var UpdateVocabulary = [
    9,
    n0,
    _UV,
    {
        [_h]: ["PATCH", "/vocabularies/{VocabularyName}", 200],
    },
    () => UpdateVocabularyRequest,
    () => UpdateVocabularyResponse,
];
var UpdateVocabularyFilter = [
    9,
    n0,
    _UVF,
    {
        [_h]: ["PUT", "/vocabularyFilters/{VocabularyFilterName}", 200],
    },
    () => UpdateVocabularyFilterRequest,
    () => UpdateVocabularyFilterResponse,
];

class CreateCallAnalyticsCategoryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "CreateCallAnalyticsCategory", {})
    .n("TranscribeClient", "CreateCallAnalyticsCategoryCommand")
    .sc(CreateCallAnalyticsCategory)
    .build() {
}

class CreateLanguageModelCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "CreateLanguageModel", {})
    .n("TranscribeClient", "CreateLanguageModelCommand")
    .sc(CreateLanguageModel)
    .build() {
}

class CreateMedicalVocabularyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "CreateMedicalVocabulary", {})
    .n("TranscribeClient", "CreateMedicalVocabularyCommand")
    .sc(CreateMedicalVocabulary)
    .build() {
}

class CreateVocabularyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "CreateVocabulary", {})
    .n("TranscribeClient", "CreateVocabularyCommand")
    .sc(CreateVocabulary)
    .build() {
}

class CreateVocabularyFilterCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "CreateVocabularyFilter", {})
    .n("TranscribeClient", "CreateVocabularyFilterCommand")
    .sc(CreateVocabularyFilter)
    .build() {
}

class DeleteCallAnalyticsCategoryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "DeleteCallAnalyticsCategory", {})
    .n("TranscribeClient", "DeleteCallAnalyticsCategoryCommand")
    .sc(DeleteCallAnalyticsCategory)
    .build() {
}

class DeleteCallAnalyticsJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "DeleteCallAnalyticsJob", {})
    .n("TranscribeClient", "DeleteCallAnalyticsJobCommand")
    .sc(DeleteCallAnalyticsJob)
    .build() {
}

class DeleteLanguageModelCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "DeleteLanguageModel", {})
    .n("TranscribeClient", "DeleteLanguageModelCommand")
    .sc(DeleteLanguageModel)
    .build() {
}

class DeleteMedicalScribeJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "DeleteMedicalScribeJob", {})
    .n("TranscribeClient", "DeleteMedicalScribeJobCommand")
    .sc(DeleteMedicalScribeJob)
    .build() {
}

class DeleteMedicalTranscriptionJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "DeleteMedicalTranscriptionJob", {})
    .n("TranscribeClient", "DeleteMedicalTranscriptionJobCommand")
    .sc(DeleteMedicalTranscriptionJob)
    .build() {
}

class DeleteMedicalVocabularyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "DeleteMedicalVocabulary", {})
    .n("TranscribeClient", "DeleteMedicalVocabularyCommand")
    .sc(DeleteMedicalVocabulary)
    .build() {
}

class DeleteTranscriptionJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "DeleteTranscriptionJob", {})
    .n("TranscribeClient", "DeleteTranscriptionJobCommand")
    .sc(DeleteTranscriptionJob)
    .build() {
}

class DeleteVocabularyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "DeleteVocabulary", {})
    .n("TranscribeClient", "DeleteVocabularyCommand")
    .sc(DeleteVocabulary)
    .build() {
}

class DeleteVocabularyFilterCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "DeleteVocabularyFilter", {})
    .n("TranscribeClient", "DeleteVocabularyFilterCommand")
    .sc(DeleteVocabularyFilter)
    .build() {
}

class DescribeLanguageModelCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "DescribeLanguageModel", {})
    .n("TranscribeClient", "DescribeLanguageModelCommand")
    .sc(DescribeLanguageModel)
    .build() {
}

class GetCallAnalyticsCategoryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "GetCallAnalyticsCategory", {})
    .n("TranscribeClient", "GetCallAnalyticsCategoryCommand")
    .sc(GetCallAnalyticsCategory)
    .build() {
}

class GetCallAnalyticsJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "GetCallAnalyticsJob", {})
    .n("TranscribeClient", "GetCallAnalyticsJobCommand")
    .sc(GetCallAnalyticsJob)
    .build() {
}

class GetMedicalScribeJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "GetMedicalScribeJob", {})
    .n("TranscribeClient", "GetMedicalScribeJobCommand")
    .sc(GetMedicalScribeJob)
    .build() {
}

class GetMedicalTranscriptionJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "GetMedicalTranscriptionJob", {})
    .n("TranscribeClient", "GetMedicalTranscriptionJobCommand")
    .sc(GetMedicalTranscriptionJob)
    .build() {
}

class GetMedicalVocabularyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "GetMedicalVocabulary", {})
    .n("TranscribeClient", "GetMedicalVocabularyCommand")
    .sc(GetMedicalVocabulary)
    .build() {
}

class GetTranscriptionJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "GetTranscriptionJob", {})
    .n("TranscribeClient", "GetTranscriptionJobCommand")
    .sc(GetTranscriptionJob)
    .build() {
}

class GetVocabularyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "GetVocabulary", {})
    .n("TranscribeClient", "GetVocabularyCommand")
    .sc(GetVocabulary)
    .build() {
}

class GetVocabularyFilterCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "GetVocabularyFilter", {})
    .n("TranscribeClient", "GetVocabularyFilterCommand")
    .sc(GetVocabularyFilter)
    .build() {
}

class ListCallAnalyticsCategoriesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "ListCallAnalyticsCategories", {})
    .n("TranscribeClient", "ListCallAnalyticsCategoriesCommand")
    .sc(ListCallAnalyticsCategories)
    .build() {
}

class ListCallAnalyticsJobsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "ListCallAnalyticsJobs", {})
    .n("TranscribeClient", "ListCallAnalyticsJobsCommand")
    .sc(ListCallAnalyticsJobs)
    .build() {
}

class ListLanguageModelsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "ListLanguageModels", {})
    .n("TranscribeClient", "ListLanguageModelsCommand")
    .sc(ListLanguageModels)
    .build() {
}

class ListMedicalScribeJobsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "ListMedicalScribeJobs", {})
    .n("TranscribeClient", "ListMedicalScribeJobsCommand")
    .sc(ListMedicalScribeJobs)
    .build() {
}

class ListMedicalTranscriptionJobsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "ListMedicalTranscriptionJobs", {})
    .n("TranscribeClient", "ListMedicalTranscriptionJobsCommand")
    .sc(ListMedicalTranscriptionJobs)
    .build() {
}

class ListMedicalVocabulariesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "ListMedicalVocabularies", {})
    .n("TranscribeClient", "ListMedicalVocabulariesCommand")
    .sc(ListMedicalVocabularies)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "ListTagsForResource", {})
    .n("TranscribeClient", "ListTagsForResourceCommand")
    .sc(ListTagsForResource)
    .build() {
}

class ListTranscriptionJobsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "ListTranscriptionJobs", {})
    .n("TranscribeClient", "ListTranscriptionJobsCommand")
    .sc(ListTranscriptionJobs)
    .build() {
}

class ListVocabulariesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "ListVocabularies", {})
    .n("TranscribeClient", "ListVocabulariesCommand")
    .sc(ListVocabularies)
    .build() {
}

class ListVocabularyFiltersCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "ListVocabularyFilters", {})
    .n("TranscribeClient", "ListVocabularyFiltersCommand")
    .sc(ListVocabularyFilters)
    .build() {
}

class StartCallAnalyticsJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "StartCallAnalyticsJob", {})
    .n("TranscribeClient", "StartCallAnalyticsJobCommand")
    .sc(StartCallAnalyticsJob)
    .build() {
}

class StartMedicalScribeJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "StartMedicalScribeJob", {})
    .n("TranscribeClient", "StartMedicalScribeJobCommand")
    .sc(StartMedicalScribeJob)
    .build() {
}

class StartMedicalTranscriptionJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "StartMedicalTranscriptionJob", {})
    .n("TranscribeClient", "StartMedicalTranscriptionJobCommand")
    .sc(StartMedicalTranscriptionJob)
    .build() {
}

class StartTranscriptionJobCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "StartTranscriptionJob", {})
    .n("TranscribeClient", "StartTranscriptionJobCommand")
    .sc(StartTranscriptionJob)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "TagResource", {})
    .n("TranscribeClient", "TagResourceCommand")
    .sc(TagResource)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "UntagResource", {})
    .n("TranscribeClient", "UntagResourceCommand")
    .sc(UntagResource)
    .build() {
}

class UpdateCallAnalyticsCategoryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "UpdateCallAnalyticsCategory", {})
    .n("TranscribeClient", "UpdateCallAnalyticsCategoryCommand")
    .sc(UpdateCallAnalyticsCategory)
    .build() {
}

class UpdateMedicalVocabularyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "UpdateMedicalVocabulary", {})
    .n("TranscribeClient", "UpdateMedicalVocabularyCommand")
    .sc(UpdateMedicalVocabulary)
    .build() {
}

class UpdateVocabularyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "UpdateVocabulary", {})
    .n("TranscribeClient", "UpdateVocabularyCommand")
    .sc(UpdateVocabulary)
    .build() {
}

class UpdateVocabularyFilterCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("Transcribe", "UpdateVocabularyFilter", {})
    .n("TranscribeClient", "UpdateVocabularyFilterCommand")
    .sc(UpdateVocabularyFilter)
    .build() {
}

const commands = {
    CreateCallAnalyticsCategoryCommand,
    CreateLanguageModelCommand,
    CreateMedicalVocabularyCommand,
    CreateVocabularyCommand,
    CreateVocabularyFilterCommand,
    DeleteCallAnalyticsCategoryCommand,
    DeleteCallAnalyticsJobCommand,
    DeleteLanguageModelCommand,
    DeleteMedicalScribeJobCommand,
    DeleteMedicalTranscriptionJobCommand,
    DeleteMedicalVocabularyCommand,
    DeleteTranscriptionJobCommand,
    DeleteVocabularyCommand,
    DeleteVocabularyFilterCommand,
    DescribeLanguageModelCommand,
    GetCallAnalyticsCategoryCommand,
    GetCallAnalyticsJobCommand,
    GetMedicalScribeJobCommand,
    GetMedicalTranscriptionJobCommand,
    GetMedicalVocabularyCommand,
    GetTranscriptionJobCommand,
    GetVocabularyCommand,
    GetVocabularyFilterCommand,
    ListCallAnalyticsCategoriesCommand,
    ListCallAnalyticsJobsCommand,
    ListLanguageModelsCommand,
    ListMedicalScribeJobsCommand,
    ListMedicalTranscriptionJobsCommand,
    ListMedicalVocabulariesCommand,
    ListTagsForResourceCommand,
    ListTranscriptionJobsCommand,
    ListVocabulariesCommand,
    ListVocabularyFiltersCommand,
    StartCallAnalyticsJobCommand,
    StartMedicalScribeJobCommand,
    StartMedicalTranscriptionJobCommand,
    StartTranscriptionJobCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateCallAnalyticsCategoryCommand,
    UpdateMedicalVocabularyCommand,
    UpdateVocabularyCommand,
    UpdateVocabularyFilterCommand,
};
class Transcribe extends TranscribeClient {
}
smithyClient.createAggregatedClient(commands, Transcribe);

const paginateListCallAnalyticsCategories = core.createPaginator(TranscribeClient, ListCallAnalyticsCategoriesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListCallAnalyticsJobs = core.createPaginator(TranscribeClient, ListCallAnalyticsJobsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListLanguageModels = core.createPaginator(TranscribeClient, ListLanguageModelsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListMedicalScribeJobs = core.createPaginator(TranscribeClient, ListMedicalScribeJobsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListMedicalTranscriptionJobs = core.createPaginator(TranscribeClient, ListMedicalTranscriptionJobsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListMedicalVocabularies = core.createPaginator(TranscribeClient, ListMedicalVocabulariesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListTranscriptionJobs = core.createPaginator(TranscribeClient, ListTranscriptionJobsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListVocabularies = core.createPaginator(TranscribeClient, ListVocabulariesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListVocabularyFilters = core.createPaginator(TranscribeClient, ListVocabularyFiltersCommand, "NextToken", "NextToken", "MaxResults");

const BaseModelName = {
    NARROW_BAND: "NarrowBand",
    WIDE_BAND: "WideBand",
};
const CallAnalyticsFeature = {
    GENERATIVE_SUMMARIZATION: "GENERATIVE_SUMMARIZATION",
};
const CallAnalyticsSkippedReasonCode = {
    FAILED_SAFETY_GUIDELINES: "FAILED_SAFETY_GUIDELINES",
    INSUFFICIENT_CONVERSATION_CONTENT: "INSUFFICIENT_CONVERSATION_CONTENT",
};
const CallAnalyticsJobStatus = {
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    QUEUED: "QUEUED",
};
const ParticipantRole = {
    AGENT: "AGENT",
    CUSTOMER: "CUSTOMER",
};
const LanguageCode = {
    AB_GE: "ab-GE",
    AF_ZA: "af-ZA",
    AR_AE: "ar-AE",
    AR_SA: "ar-SA",
    AST_ES: "ast-ES",
    AZ_AZ: "az-AZ",
    BA_RU: "ba-RU",
    BE_BY: "be-BY",
    BG_BG: "bg-BG",
    BN_IN: "bn-IN",
    BS_BA: "bs-BA",
    CA_ES: "ca-ES",
    CKB_IQ: "ckb-IQ",
    CKB_IR: "ckb-IR",
    CS_CZ: "cs-CZ",
    CY_WL: "cy-WL",
    DA_DK: "da-DK",
    DE_CH: "de-CH",
    DE_DE: "de-DE",
    EL_GR: "el-GR",
    EN_AB: "en-AB",
    EN_AU: "en-AU",
    EN_GB: "en-GB",
    EN_IE: "en-IE",
    EN_IN: "en-IN",
    EN_NZ: "en-NZ",
    EN_US: "en-US",
    EN_WL: "en-WL",
    EN_ZA: "en-ZA",
    ES_ES: "es-ES",
    ES_US: "es-US",
    ET_EE: "et-EE",
    ET_ET: "et-ET",
    EU_ES: "eu-ES",
    FA_IR: "fa-IR",
    FI_FI: "fi-FI",
    FR_CA: "fr-CA",
    FR_FR: "fr-FR",
    GL_ES: "gl-ES",
    GU_IN: "gu-IN",
    HA_NG: "ha-NG",
    HE_IL: "he-IL",
    HI_IN: "hi-IN",
    HR_HR: "hr-HR",
    HU_HU: "hu-HU",
    HY_AM: "hy-AM",
    ID_ID: "id-ID",
    IS_IS: "is-IS",
    IT_IT: "it-IT",
    JA_JP: "ja-JP",
    KAB_DZ: "kab-DZ",
    KA_GE: "ka-GE",
    KK_KZ: "kk-KZ",
    KN_IN: "kn-IN",
    KO_KR: "ko-KR",
    KY_KG: "ky-KG",
    LG_IN: "lg-IN",
    LT_LT: "lt-LT",
    LV_LV: "lv-LV",
    MHR_RU: "mhr-RU",
    MI_NZ: "mi-NZ",
    MK_MK: "mk-MK",
    ML_IN: "ml-IN",
    MN_MN: "mn-MN",
    MR_IN: "mr-IN",
    MS_MY: "ms-MY",
    MT_MT: "mt-MT",
    NL_NL: "nl-NL",
    NO_NO: "no-NO",
    OR_IN: "or-IN",
    PA_IN: "pa-IN",
    PL_PL: "pl-PL",
    PS_AF: "ps-AF",
    PT_BR: "pt-BR",
    PT_PT: "pt-PT",
    RO_RO: "ro-RO",
    RU_RU: "ru-RU",
    RW_RW: "rw-RW",
    SI_LK: "si-LK",
    SK_SK: "sk-SK",
    SL_SI: "sl-SI",
    SO_SO: "so-SO",
    SR_RS: "sr-RS",
    SU_ID: "su-ID",
    SV_SE: "sv-SE",
    SW_BI: "sw-BI",
    SW_KE: "sw-KE",
    SW_RW: "sw-RW",
    SW_TZ: "sw-TZ",
    SW_UG: "sw-UG",
    TA_IN: "ta-IN",
    TE_IN: "te-IN",
    TH_TH: "th-TH",
    TL_PH: "tl-PH",
    TR_TR: "tr-TR",
    TT_RU: "tt-RU",
    UG_CN: "ug-CN",
    UK_UA: "uk-UA",
    UZ_UZ: "uz-UZ",
    VI_VN: "vi-VN",
    WO_SN: "wo-SN",
    ZH_CN: "zh-CN",
    ZH_HK: "zh-HK",
    ZH_TW: "zh-TW",
    ZU_ZA: "zu-ZA",
};
const MediaFormat = {
    AMR: "amr",
    FLAC: "flac",
    M4A: "m4a",
    MP3: "mp3",
    MP4: "mp4",
    OGG: "ogg",
    WAV: "wav",
    WEBM: "webm",
};
const PiiEntityType = {
    ADDRESS: "ADDRESS",
    ALL: "ALL",
    BANK_ACCOUNT_NUMBER: "BANK_ACCOUNT_NUMBER",
    BANK_ROUTING: "BANK_ROUTING",
    CREDIT_DEBIT_CVV: "CREDIT_DEBIT_CVV",
    CREDIT_DEBIT_EXPIRY: "CREDIT_DEBIT_EXPIRY",
    CREDIT_DEBIT_NUMBER: "CREDIT_DEBIT_NUMBER",
    EMAIL: "EMAIL",
    NAME: "NAME",
    PHONE: "PHONE",
    PIN: "PIN",
    SSN: "SSN",
};
const RedactionOutput = {
    REDACTED: "redacted",
    REDACTED_AND_UNREDACTED: "redacted_and_unredacted",
};
const RedactionType = {
    PII: "PII",
};
const VocabularyFilterMethod = {
    MASK: "mask",
    REMOVE: "remove",
    TAG: "tag",
};
const InputType = {
    POST_CALL: "POST_CALL",
    REAL_TIME: "REAL_TIME",
};
const SentimentValue = {
    MIXED: "MIXED",
    NEGATIVE: "NEGATIVE",
    NEUTRAL: "NEUTRAL",
    POSITIVE: "POSITIVE",
};
const TranscriptFilterType = {
    EXACT: "EXACT",
};
const MedicalScribeNoteTemplate = {
    BEHAVIORAL_SOAP: "BEHAVIORAL_SOAP",
    BIRP: "BIRP",
    DAP: "DAP",
    GIRPP: "GIRPP",
    HISTORY_AND_PHYSICAL: "HISTORY_AND_PHYSICAL",
    PHYSICAL_SOAP: "PHYSICAL_SOAP",
    SIRP: "SIRP",
};
const CLMLanguageCode = {
    DE_DE: "de-DE",
    EN_AU: "en-AU",
    EN_GB: "en-GB",
    EN_US: "en-US",
    ES_US: "es-US",
    HI_IN: "hi-IN",
    JA_JP: "ja-JP",
};
const ModelStatus = {
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
};
const VocabularyState = {
    FAILED: "FAILED",
    PENDING: "PENDING",
    READY: "READY",
};
const MedicalScribeParticipantRole = {
    CLINICIAN: "CLINICIAN",
    PATIENT: "PATIENT",
};
const MedicalScribeLanguageCode = {
    EN_US: "en-US",
};
const MedicalScribeJobStatus = {
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    QUEUED: "QUEUED",
};
const MedicalContentIdentificationType = {
    PHI: "PHI",
};
const Specialty = {
    PRIMARYCARE: "PRIMARYCARE",
};
const TranscriptionJobStatus = {
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    QUEUED: "QUEUED",
};
const Type = {
    CONVERSATION: "CONVERSATION",
    DICTATION: "DICTATION",
};
const SubtitleFormat = {
    SRT: "srt",
    VTT: "vtt",
};
const ToxicityCategory = {
    ALL: "ALL",
};
const OutputLocationType = {
    CUSTOMER_BUCKET: "CUSTOMER_BUCKET",
    SERVICE_BUCKET: "SERVICE_BUCKET",
};
const Pronouns = {
    HE_HIM: "HE_HIM",
    SHE_HER: "SHE_HER",
    THEY_THEM: "THEY_THEM",
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.BadRequestException = BadRequestException$1;
exports.BaseModelName = BaseModelName;
exports.CLMLanguageCode = CLMLanguageCode;
exports.CallAnalyticsFeature = CallAnalyticsFeature;
exports.CallAnalyticsJobStatus = CallAnalyticsJobStatus;
exports.CallAnalyticsSkippedReasonCode = CallAnalyticsSkippedReasonCode;
exports.ConflictException = ConflictException$1;
exports.CreateCallAnalyticsCategoryCommand = CreateCallAnalyticsCategoryCommand;
exports.CreateLanguageModelCommand = CreateLanguageModelCommand;
exports.CreateMedicalVocabularyCommand = CreateMedicalVocabularyCommand;
exports.CreateVocabularyCommand = CreateVocabularyCommand;
exports.CreateVocabularyFilterCommand = CreateVocabularyFilterCommand;
exports.DeleteCallAnalyticsCategoryCommand = DeleteCallAnalyticsCategoryCommand;
exports.DeleteCallAnalyticsJobCommand = DeleteCallAnalyticsJobCommand;
exports.DeleteLanguageModelCommand = DeleteLanguageModelCommand;
exports.DeleteMedicalScribeJobCommand = DeleteMedicalScribeJobCommand;
exports.DeleteMedicalTranscriptionJobCommand = DeleteMedicalTranscriptionJobCommand;
exports.DeleteMedicalVocabularyCommand = DeleteMedicalVocabularyCommand;
exports.DeleteTranscriptionJobCommand = DeleteTranscriptionJobCommand;
exports.DeleteVocabularyCommand = DeleteVocabularyCommand;
exports.DeleteVocabularyFilterCommand = DeleteVocabularyFilterCommand;
exports.DescribeLanguageModelCommand = DescribeLanguageModelCommand;
exports.GetCallAnalyticsCategoryCommand = GetCallAnalyticsCategoryCommand;
exports.GetCallAnalyticsJobCommand = GetCallAnalyticsJobCommand;
exports.GetMedicalScribeJobCommand = GetMedicalScribeJobCommand;
exports.GetMedicalTranscriptionJobCommand = GetMedicalTranscriptionJobCommand;
exports.GetMedicalVocabularyCommand = GetMedicalVocabularyCommand;
exports.GetTranscriptionJobCommand = GetTranscriptionJobCommand;
exports.GetVocabularyCommand = GetVocabularyCommand;
exports.GetVocabularyFilterCommand = GetVocabularyFilterCommand;
exports.InputType = InputType;
exports.InternalFailureException = InternalFailureException$1;
exports.LanguageCode = LanguageCode;
exports.LimitExceededException = LimitExceededException$1;
exports.ListCallAnalyticsCategoriesCommand = ListCallAnalyticsCategoriesCommand;
exports.ListCallAnalyticsJobsCommand = ListCallAnalyticsJobsCommand;
exports.ListLanguageModelsCommand = ListLanguageModelsCommand;
exports.ListMedicalScribeJobsCommand = ListMedicalScribeJobsCommand;
exports.ListMedicalTranscriptionJobsCommand = ListMedicalTranscriptionJobsCommand;
exports.ListMedicalVocabulariesCommand = ListMedicalVocabulariesCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTranscriptionJobsCommand = ListTranscriptionJobsCommand;
exports.ListVocabulariesCommand = ListVocabulariesCommand;
exports.ListVocabularyFiltersCommand = ListVocabularyFiltersCommand;
exports.MediaFormat = MediaFormat;
exports.MedicalContentIdentificationType = MedicalContentIdentificationType;
exports.MedicalScribeJobStatus = MedicalScribeJobStatus;
exports.MedicalScribeLanguageCode = MedicalScribeLanguageCode;
exports.MedicalScribeNoteTemplate = MedicalScribeNoteTemplate;
exports.MedicalScribeParticipantRole = MedicalScribeParticipantRole;
exports.ModelStatus = ModelStatus;
exports.NotFoundException = NotFoundException$1;
exports.OutputLocationType = OutputLocationType;
exports.ParticipantRole = ParticipantRole;
exports.PiiEntityType = PiiEntityType;
exports.Pronouns = Pronouns;
exports.RedactionOutput = RedactionOutput;
exports.RedactionType = RedactionType;
exports.SentimentValue = SentimentValue;
exports.Specialty = Specialty;
exports.StartCallAnalyticsJobCommand = StartCallAnalyticsJobCommand;
exports.StartMedicalScribeJobCommand = StartMedicalScribeJobCommand;
exports.StartMedicalTranscriptionJobCommand = StartMedicalTranscriptionJobCommand;
exports.StartTranscriptionJobCommand = StartTranscriptionJobCommand;
exports.SubtitleFormat = SubtitleFormat;
exports.TagResourceCommand = TagResourceCommand;
exports.ToxicityCategory = ToxicityCategory;
exports.Transcribe = Transcribe;
exports.TranscribeClient = TranscribeClient;
exports.TranscribeServiceException = TranscribeServiceException$1;
exports.TranscriptFilterType = TranscriptFilterType;
exports.TranscriptionJobStatus = TranscriptionJobStatus;
exports.Type = Type;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateCallAnalyticsCategoryCommand = UpdateCallAnalyticsCategoryCommand;
exports.UpdateMedicalVocabularyCommand = UpdateMedicalVocabularyCommand;
exports.UpdateVocabularyCommand = UpdateVocabularyCommand;
exports.UpdateVocabularyFilterCommand = UpdateVocabularyFilterCommand;
exports.VocabularyFilterMethod = VocabularyFilterMethod;
exports.VocabularyState = VocabularyState;
exports.paginateListCallAnalyticsCategories = paginateListCallAnalyticsCategories;
exports.paginateListCallAnalyticsJobs = paginateListCallAnalyticsJobs;
exports.paginateListLanguageModels = paginateListLanguageModels;
exports.paginateListMedicalScribeJobs = paginateListMedicalScribeJobs;
exports.paginateListMedicalTranscriptionJobs = paginateListMedicalTranscriptionJobs;
exports.paginateListMedicalVocabularies = paginateListMedicalVocabularies;
exports.paginateListTranscriptionJobs = paginateListTranscriptionJobs;
exports.paginateListVocabularies = paginateListVocabularies;
exports.paginateListVocabularyFilters = paginateListVocabularyFilters;
