// oci-lustrefilestorage/lib/model/object-storage-link.js
"use strict";
/**
 * File Storage with Lustre API
 * Use the File Storage with Lustre API to manage Lustre file systems and related resources. For more information, see [File Storage with Lustre](https://docs.oracle.com/iaas/Content/lustre/home.htm).
 * OpenAPI spec version: 20250228
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObjectStorageLink = void 0;
var ObjectStorageLink;
(function (ObjectStorageLink) {
    let LifecycleState;
    (function (LifecycleState) {
        LifecycleState["Creating"] = "CREATING";
        LifecycleState["Active"] = "ACTIVE";
        LifecycleState["Deleting"] = "DELETING";
        LifecycleState["Deleted"] = "DELETED";
        LifecycleState["Failed"] = "FAILED";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        LifecycleState["UnknownValue"] = "UNKNOWN_VALUE";
    })(LifecycleState = ObjectStorageLink.LifecycleState || (ObjectStorageLink.LifecycleState = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ObjectStorageLink.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    ObjectStorageLink.getDeserializedJsonObj = getDeserializedJsonObj;
})(ObjectStorageLink = exports.ObjectStorageLink || (exports.ObjectStorageLink = {}));
