// oci-servicecatalog/lib/model/service-catalog-status-enum.js
"use strict";
/**
 * Service Catalog API
 * Use the Service Catalog API to manage solutions in Oracle Cloud Infrastructure Service Catalog.
For more information, see [Overview of Service Catalog](https://docs.oracle.com/iaas/Content/service-catalog/overview_of_service_catalog.htm).

 * OpenAPI spec version: 20210527
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceCatalogStatusEnum = void 0;
/**
 * Possible values for service catalog status
 **/
var ServiceCatalogStatusEnum;
(function (ServiceCatalogStatusEnum) {
    ServiceCatalogStatusEnum["Active"] = "ACTIVE";
    ServiceCatalogStatusEnum["Inactive"] = "INACTIVE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ServiceCatalogStatusEnum["UnknownValue"] = "UNKNOWN_VALUE";
})(ServiceCatalogStatusEnum = exports.ServiceCatalogStatusEnum || (exports.ServiceCatalogStatusEnum = {}));
(function (ServiceCatalogStatusEnum) {
    function getJsonObj(obj) {
        return obj;
    }
    ServiceCatalogStatusEnum.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ServiceCatalogStatusEnum.getDeserializedJsonObj = getDeserializedJsonObj;
})(ServiceCatalogStatusEnum = exports.ServiceCatalogStatusEnum || (exports.ServiceCatalogStatusEnum = {}));
