// oci-goldengate/lib/model/pause-pipeline-type.js
"use strict";
/**
 * GoldenGate API
 * Use the Oracle Cloud Infrastructure GoldenGate APIs to perform data replication operations.

 * OpenAPI spec version: 20200407
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PausePipelineType = void 0;
/**
 * The pipeline pause type.
 *
 **/
var PausePipelineType;
(function (PausePipelineType) {
    PausePipelineType["Default"] = "DEFAULT";
})(PausePipelineType = exports.PausePipelineType || (exports.PausePipelineType = {}));
(function (PausePipelineType) {
    function getJsonObj(obj) {
        return obj;
    }
    PausePipelineType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    PausePipelineType.getDeserializedJsonObj = getDeserializedJsonObj;
})(PausePipelineType = exports.PausePipelineType || (exports.PausePipelineType = {}));
