// oci-marketplace/lib/request/list-marketplace-metadata-public-keys-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20181001
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListMarketplaceMetadataPublicKeysRequest = void 0;
var ListMarketplaceMetadataPublicKeysRequest;
(function (ListMarketplaceMetadataPublicKeysRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListMarketplaceMetadataPublicKeysRequest.SortOrder || (ListMarketplaceMetadataPublicKeysRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["KeyId"] = "keyId";
    })(SortBy = ListMarketplaceMetadataPublicKeysRequest.SortBy || (ListMarketplaceMetadataPublicKeysRequest.SortBy = {}));
})(ListMarketplaceMetadataPublicKeysRequest = exports.ListMarketplaceMetadataPublicKeysRequest || (exports.ListMarketplaceMetadataPublicKeysRequest = {}));
