// oci-distributeddatabase/lib/model/distributed-db-transport-type.js
"use strict";
/**
 * Globally Distributed Database
 * Use the Globally Distributed Database service APIs to create and manage the Globally distributed databases.
 * OpenAPI spec version: 20250101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DistributedDbTransportType = void 0;
/**
 * The redo transport type to use for this Data Guard association.
 **/
var DistributedDbTransportType;
(function (DistributedDbTransportType) {
    DistributedDbTransportType["Sync"] = "SYNC";
    DistributedDbTransportType["Async"] = "ASYNC";
    DistributedDbTransportType["Fastsync"] = "FASTSYNC";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DistributedDbTransportType["UnknownValue"] = "UNKNOWN_VALUE";
})(DistributedDbTransportType = exports.DistributedDbTransportType || (exports.DistributedDbTransportType = {}));
(function (DistributedDbTransportType) {
    function getJsonObj(obj) {
        return obj;
    }
    DistributedDbTransportType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DistributedDbTransportType.getDeserializedJsonObj = getDeserializedJsonObj;
})(DistributedDbTransportType = exports.DistributedDbTransportType || (exports.DistributedDbTransportType = {}));
