// oci-distributeddatabase/lib/model/distributed-autonomous-db-protection-mode.js
"use strict";
/**
 * Globally Distributed Database
 * Use the Globally Distributed Database service APIs to create and manage the Globally distributed databases.
 * OpenAPI spec version: 20250101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DistributedAutonomousDbProtectionMode = void 0;
/**
 * The protection mode of this Autonomous Data Guard association. For more information, see
 * [Oracle Data Guard Protection Modes](http://docs.oracle.com/database/122/SBYDB/oracle-data-guard-protection-modes.htm#SBYDB02000)
 * in the Oracle Data Guard documentation.
 *
 **/
var DistributedAutonomousDbProtectionMode;
(function (DistributedAutonomousDbProtectionMode) {
    DistributedAutonomousDbProtectionMode["MaximumAvailability"] = "MAXIMUM_AVAILABILITY";
    DistributedAutonomousDbProtectionMode["MaximumPerformance"] = "MAXIMUM_PERFORMANCE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DistributedAutonomousDbProtectionMode["UnknownValue"] = "UNKNOWN_VALUE";
})(DistributedAutonomousDbProtectionMode = exports.DistributedAutonomousDbProtectionMode || (exports.DistributedAutonomousDbProtectionMode = {}));
(function (DistributedAutonomousDbProtectionMode) {
    function getJsonObj(obj) {
        return obj;
    }
    DistributedAutonomousDbProtectionMode.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DistributedAutonomousDbProtectionMode.getDeserializedJsonObj = getDeserializedJsonObj;
})(DistributedAutonomousDbProtectionMode = exports.DistributedAutonomousDbProtectionMode || (exports.DistributedAutonomousDbProtectionMode = {}));
