// oci-distributeddatabase/lib/model/distributed-autonomous-db-backup-destination.js
"use strict";
/**
 * Globally Distributed Database
 * Use the Globally Distributed Database service APIs to create and manage the Globally distributed databases.
 * OpenAPI spec version: 20250101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DistributedAutonomousDbBackupDestination = void 0;
var DistributedAutonomousDbBackupDestination;
(function (DistributedAutonomousDbBackupDestination) {
    let Type;
    (function (Type) {
        Type["Nfs"] = "NFS";
        Type["RecoveryAppliance"] = "RECOVERY_APPLIANCE";
        Type["ObjectStore"] = "OBJECT_STORE";
        Type["Local"] = "LOCAL";
        Type["Dbrs"] = "DBRS";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        Type["UnknownValue"] = "UNKNOWN_VALUE";
    })(Type = DistributedAutonomousDbBackupDestination.Type || (DistributedAutonomousDbBackupDestination.Type = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DistributedAutonomousDbBackupDestination.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    DistributedAutonomousDbBackupDestination.getDeserializedJsonObj = getDeserializedJsonObj;
})(DistributedAutonomousDbBackupDestination = exports.DistributedAutonomousDbBackupDestination || (exports.DistributedAutonomousDbBackupDestination = {}));
