// oci-distributeddatabase/lib/model/distributed-db-protection-mode.js
"use strict";
/**
 * Globally Distributed Database
 * Use the Globally Distributed Database service APIs to create and manage the Globally distributed databases.
 * OpenAPI spec version: 20250101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DistributedDbProtectionMode = void 0;
/**
 * The protection mode to set up between the primary and standby databases.
 **/
var DistributedDbProtectionMode;
(function (DistributedDbProtectionMode) {
    DistributedDbProtectionMode["MaximumAvailability"] = "MAXIMUM_AVAILABILITY";
    DistributedDbProtectionMode["MaximumPerformance"] = "MAXIMUM_PERFORMANCE";
    DistributedDbProtectionMode["MaximumProtection"] = "MAXIMUM_PROTECTION";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    DistributedDbProtectionMode["UnknownValue"] = "UNKNOWN_VALUE";
})(DistributedDbProtectionMode = exports.DistributedDbProtectionMode || (exports.DistributedDbProtectionMode = {}));
(function (DistributedDbProtectionMode) {
    function getJsonObj(obj) {
        return obj;
    }
    DistributedDbProtectionMode.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    DistributedDbProtectionMode.getDeserializedJsonObj = getDeserializedJsonObj;
})(DistributedDbProtectionMode = exports.DistributedDbProtectionMode || (exports.DistributedDbProtectionMode = {}));
