// oci-managementagent/lib/model/named-credential-field-definition.js
"use strict";
/**
 * Management Agent API
 * Use the Management Agent API to manage your infrastructure's management agents, including their plugins and install keys.
For more information, see [Management Agent](https://docs.oracle.com/iaas/management-agents/index.html).

 * OpenAPI spec version: 20200202
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NamedCredentialFieldDefinition = void 0;
var NamedCredentialFieldDefinition;
(function (NamedCredentialFieldDefinition) {
    let ValueCategory;
    (function (ValueCategory) {
        ValueCategory["ClearText"] = "CLEAR_TEXT";
        ValueCategory["SecretIdentifier"] = "SECRET_IDENTIFIER";
        ValueCategory["AdbIdentifier"] = "ADB_IDENTIFIER";
        ValueCategory["AllowedValue"] = "ALLOWED_VALUE";
        /**
         * This value is used if a service returns a value for this enum that is not recognized by this
         * version of the SDK.
         */
        ValueCategory["UnknownValue"] = "UNKNOWN_VALUE";
    })(ValueCategory = NamedCredentialFieldDefinition.ValueCategory || (NamedCredentialFieldDefinition.ValueCategory = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    NamedCredentialFieldDefinition.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    NamedCredentialFieldDefinition.getDeserializedJsonObj = getDeserializedJsonObj;
})(NamedCredentialFieldDefinition = exports.NamedCredentialFieldDefinition || (exports.NamedCredentialFieldDefinition = {}));
