// oci-managementagent/lib/model/value-category-type.js
"use strict";
/**
 * Management Agent API
 * Use the Management Agent API to manage your infrastructure's management agents, including their plugins and install keys.
For more information, see [Management Agent](https://docs.oracle.com/iaas/management-agents/index.html).

 * OpenAPI spec version: 20200202
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ValueCategoryType = void 0;
/**
 * The category of the Named credential property value.
 * CLEAR_TEXT indicates the value field contains a clear text value.
 * SECRET_IDENTIFIER indicates the value field contains a vault secret ocid identifier.
 * ADB_IDENTIFIER indicates the value field contains an Autonomous database ocid identifier.
 * ALLOWED_VALUE indicates the value should be selected from the options in the allowedValues field.
 *
 **/
var ValueCategoryType;
(function (ValueCategoryType) {
    ValueCategoryType["ClearText"] = "CLEAR_TEXT";
    ValueCategoryType["SecretIdentifier"] = "SECRET_IDENTIFIER";
    ValueCategoryType["AdbIdentifier"] = "ADB_IDENTIFIER";
    ValueCategoryType["AllowedValue"] = "ALLOWED_VALUE";
    /**
     * This value is used if a service returns a value for this enum that is not recognized by this
     * version of the SDK.
     */
    ValueCategoryType["UnknownValue"] = "UNKNOWN_VALUE";
})(ValueCategoryType = exports.ValueCategoryType || (exports.ValueCategoryType = {}));
(function (ValueCategoryType) {
    function getJsonObj(obj) {
        return obj;
    }
    ValueCategoryType.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        return obj;
    }
    ValueCategoryType.getDeserializedJsonObj = getDeserializedJsonObj;
})(ValueCategoryType = exports.ValueCategoryType || (exports.ValueCategoryType = {}));
