// oci-datascience/lib/request/list-model-groups-request.js
"use strict";
/**
 *
 *
 * OpenAPI spec version: 20190101
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListModelGroupsRequest = void 0;
var ListModelGroupsRequest;
(function (ListModelGroupsRequest) {
    let SortOrder;
    (function (SortOrder) {
        SortOrder["Asc"] = "ASC";
        SortOrder["Desc"] = "DESC";
    })(SortOrder = ListModelGroupsRequest.SortOrder || (ListModelGroupsRequest.SortOrder = {}));
    let SortBy;
    (function (SortBy) {
        SortBy["TimeCreated"] = "timeCreated";
        SortBy["DisplayName"] = "displayName";
        SortBy["LifecycleState"] = "lifecycleState";
    })(SortBy = ListModelGroupsRequest.SortBy || (ListModelGroupsRequest.SortBy = {}));
})(ListModelGroupsRequest = exports.ListModelGroupsRequest || (exports.ListModelGroupsRequest = {}));
