// oci-artifacts/lib/model/create-container-image-signature-details.js
"use strict";
/**
 * Artifacts and Container Images API
 * Use the Artifacts and Container Images API to manage container images and non-container generic artifacts.

- For container images such as Docker images, use the {@link ContainerImage} resource. Save the images in a {@link ContainerRepository}.

- For non-container generic artifacts or blobs, use the {@link GenericArtifact} resource. Save the artifacts in an {@link Repository}.
- To upload and download non-container generic artifacts, instead of the Artifacts and Container Images API, use the Generic Artifacts Content API.
For more information, see the user guides for [Container Registry](https://docs.oracle.com/iaas/Content/Registry/home.htm) and [Artifact Registry](https://docs.oracle.com/iaas/Content/artifacts/home.htm).

 * OpenAPI spec version: 20160918
 *
 *
 * NOTE: This class is auto generated by OracleSDKGenerator.
 * Do not edit the class manually.
 *
 * Copyright (c) 2020, 2025, Oracle and/or its affiliates.  All rights reserved.
 * This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateContainerImageSignatureDetails = void 0;
var CreateContainerImageSignatureDetails;
(function (CreateContainerImageSignatureDetails) {
    let SigningAlgorithm;
    (function (SigningAlgorithm) {
        SigningAlgorithm["Sha224RsaPkcsPss"] = "SHA_224_RSA_PKCS_PSS";
        SigningAlgorithm["Sha256RsaPkcsPss"] = "SHA_256_RSA_PKCS_PSS";
        SigningAlgorithm["Sha384RsaPkcsPss"] = "SHA_384_RSA_PKCS_PSS";
        SigningAlgorithm["Sha512RsaPkcsPss"] = "SHA_512_RSA_PKCS_PSS";
    })(SigningAlgorithm = CreateContainerImageSignatureDetails.SigningAlgorithm || (CreateContainerImageSignatureDetails.SigningAlgorithm = {}));
    function getJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateContainerImageSignatureDetails.getJsonObj = getJsonObj;
    function getDeserializedJsonObj(obj) {
        const jsonObj = Object.assign(Object.assign({}, obj), {});
        return jsonObj;
    }
    CreateContainerImageSignatureDetails.getDeserializedJsonObj = getDeserializedJsonObj;
})(CreateContainerImageSignatureDetails = exports.CreateContainerImageSignatureDetails || (exports.CreateContainerImageSignatureDetails = {}));
